{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "initial_id",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-13T15:38:18.029744Z",
     "start_time": "2024-05-13T15:38:17.807980Z"
    },
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "from api.data import (store_database_for_eys_gene,\n",
    "                      parse_lovd,\n",
    "                      LOVD_PATH,\n",
    "                      set_lovd_dtypes)\n",
    "from api.data import save_lovd_as_vcf\n",
    "pd.options.display.max_columns = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f49f7691a27aa7b4",
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-08-11T16:16:57.305309Z",
     "start_time": "2024-08-11T16:16:56.668571Z"
    }
   },
   "outputs": [],
   "source": [
    "store_database_for_eys_gene(\"lovd\", override=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "cf5c45c0f7b9de0f",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-13T15:38:24.591752Z",
     "start_time": "2024-05-13T15:38:19.498594Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "data = parse_lovd(LOVD_PATH + \"/lovd_data.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8a089e29bfc8c119",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-13T15:12:07.510712Z",
     "start_time": "2024-05-13T15:12:07.366319Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chrom_band</th>\n",
       "      <th>imprinting</th>\n",
       "      <th>refseq_genomic</th>\n",
       "      <th>refseq_UD</th>\n",
       "      <th>reference</th>\n",
       "      <th>url_homepage</th>\n",
       "      <th>url_external</th>\n",
       "      <th>allow_download</th>\n",
       "      <th>id_hgnc</th>\n",
       "      <th>id_entrez</th>\n",
       "      <th>id_omim</th>\n",
       "      <th>show_hgmd</th>\n",
       "      <th>show_genecards</th>\n",
       "      <th>show_genetests</th>\n",
       "      <th>show_orphanet</th>\n",
       "      <th>note_index</th>\n",
       "      <th>note_listing</th>\n",
       "      <th>refseq</th>\n",
       "      <th>refseq_url</th>\n",
       "      <th>disclaimer</th>\n",
       "      <th>disclaimer_text</th>\n",
       "      <th>header</th>\n",
       "      <th>header_align</th>\n",
       "      <th>footer</th>\n",
       "      <th>footer_align</th>\n",
       "      <th>created_by</th>\n",
       "      <th>created_date</th>\n",
       "      <th>edited_by</th>\n",
       "      <th>edited_date</th>\n",
       "      <th>updated_by</th>\n",
       "      <th>updated_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EYS</td>\n",
       "      <td>eyes shut homolog (Drosophila)</td>\n",
       "      <td>6</td>\n",
       "      <td>q12</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NG_023443.2</td>\n",
       "      <td>UD_132085377375</td>\n",
       "      <td></td>\n",
       "      <td>http://www.LOVD.nl/EYS</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>21555</td>\n",
       "      <td>346007</td>\n",
       "      <td>612424</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;font color=\\\"#FF0000\\\"&gt;This database is one o...</td>\n",
       "      <td></td>\n",
       "      <td>g</td>\n",
       "      <td>http://databases.lovd.nl/shared/refseq/EYS_NM_...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>&lt;font color=\\\"#FF0000\\\"&gt;This database is one o...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>2012-02-13</td>\n",
       "      <td>6</td>\n",
       "      <td>2023-08-30 13:08:19</td>\n",
       "      <td>0</td>\n",
       "      <td>2024-04-19 20:27:30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    id                            name  ...  updated_by        updated_date\n",
       "0  EYS  eyes shut homolog (Drosophila)  ...           0 2024-04-19 20:27:30\n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transcripts\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>geneid</th>\n",
       "      <th>name</th>\n",
       "      <th>id_mutalyzer</th>\n",
       "      <th>id_ncbi</th>\n",
       "      <th>id_ensembl</th>\n",
       "      <th>id_protein_ncbi</th>\n",
       "      <th>id_protein_ensembl</th>\n",
       "      <th>id_protein_uniprot</th>\n",
       "      <th>remarks</th>\n",
       "      <th>position_c_mrna_start</th>\n",
       "      <th>position_c_mrna_end</th>\n",
       "      <th>position_c_cds_end</th>\n",
       "      <th>position_g_mrna_start</th>\n",
       "      <th>position_g_mrna_end</th>\n",
       "      <th>created_by</th>\n",
       "      <th>created_date</th>\n",
       "      <th>edited_by</th>\n",
       "      <th>edited_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7329</td>\n",
       "      <td>EYS</td>\n",
       "      <td>transcript variant 1</td>\n",
       "      <td>1</td>\n",
       "      <td>NM_001142800.1</td>\n",
       "      <td></td>\n",
       "      <td>NP_001136272.1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-538</td>\n",
       "      <td>10051</td>\n",
       "      <td>9435</td>\n",
       "      <td>66417118</td>\n",
       "      <td>64429876</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     id geneid                  name  ...  created_date edited_by edited_date\n",
       "0  7329    EYS  transcript variant 1  ...           NaT      <NA>         NaT\n",
       "\n",
       "[1 rows x 19 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Diseases\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>symbol</th>\n",
       "      <th>name</th>\n",
       "      <th>inheritance</th>\n",
       "      <th>id_omim</th>\n",
       "      <th>tissues</th>\n",
       "      <th>features</th>\n",
       "      <th>remarks</th>\n",
       "      <th>created_by</th>\n",
       "      <th>created_date</th>\n",
       "      <th>edited_by</th>\n",
       "      <th>edited_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12</td>\n",
       "      <td>PSORS</td>\n",
       "      <td>psoriasis, pustular, generalized (PSORS)</td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2012-07-06 21:50:32</td>\n",
       "      <td>6</td>\n",
       "      <td>2019-08-12 13:38:21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>58</td>\n",
       "      <td>CORD</td>\n",
       "      <td>dystrophy, cone-rod (CORD)</td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2012-09-22 11:31:25</td>\n",
       "      <td>6</td>\n",
       "      <td>2020-08-30 09:43:59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>112</td>\n",
       "      <td>RP</td>\n",
       "      <td>retinitis pigmentosa (RP)</td>\n",
       "      <td></td>\n",
       "      <td>268000</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>2013-02-21 17:12:36</td>\n",
       "      <td>6</td>\n",
       "      <td>2021-01-18 09:53:26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>139</td>\n",
       "      <td>ID</td>\n",
       "      <td>intellectual disability (ID)</td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>84</td>\n",
       "      <td>2013-06-04 18:18:07</td>\n",
       "      <td>6</td>\n",
       "      <td>2015-02-09 10:02:49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>173</td>\n",
       "      <td>SLOS</td>\n",
       "      <td>Smith-Lemli-Opitz syndrome (SLOS)</td>\n",
       "      <td>AR</td>\n",
       "      <td>270400</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2013-08-01 11:16:14</td>\n",
       "      <td>6</td>\n",
       "      <td>2021-12-10 21:51:32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>198</td>\n",
       "      <td>?</td>\n",
       "      <td>unclassified / mixed</td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2013-09-13 14:21:47</td>\n",
       "      <td>6</td>\n",
       "      <td>2016-10-22 17:54:40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2156</td>\n",
       "      <td>-</td>\n",
       "      <td>retinitis pigmentosa, X-linked, and sinorespir...</td>\n",
       "      <td></td>\n",
       "      <td>300455</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2014-09-25 23:29:40</td>\n",
       "      <td>6</td>\n",
       "      <td>2021-12-10 21:51:32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2440</td>\n",
       "      <td>RP25</td>\n",
       "      <td>retinitis pigmentosa, type 25 (RP25)</td>\n",
       "      <td>AR</td>\n",
       "      <td>602772</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2014-09-25 23:29:40</td>\n",
       "      <td>6</td>\n",
       "      <td>2021-12-10 21:51:32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4211</td>\n",
       "      <td>RPar</td>\n",
       "      <td>retinitis pigmentosa, autosomal recessive (RPar)</td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2015-02-27 18:58:57</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4214</td>\n",
       "      <td>-</td>\n",
       "      <td>retinal disease</td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2015-02-27 19:48:07</td>\n",
       "      <td>1</td>\n",
       "      <td>2023-03-09 14:26:26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>4249</td>\n",
       "      <td>macular dystrophy</td>\n",
       "      <td>dystrophy, macular</td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2015-05-04 22:10:58</td>\n",
       "      <td>6</td>\n",
       "      <td>2024-02-15 21:18:39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5086</td>\n",
       "      <td>HL</td>\n",
       "      <td>hearing loss (HL)</td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2015-10-23 11:41:05</td>\n",
       "      <td>6</td>\n",
       "      <td>2015-10-23 11:43:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5415</td>\n",
       "      <td>USH</td>\n",
       "      <td>Usher syndrome (USH)</td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2018-04-02 16:40:44</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5468</td>\n",
       "      <td>uveitis</td>\n",
       "      <td>uveitis</td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2018-08-22 09:47:04</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>6906</td>\n",
       "      <td>DEE</td>\n",
       "      <td>encephalopathy, developmental and epileptic</td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>2022-04-07 09:24:23</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      id             symbol  ... edited_by         edited_date\n",
       "0     12              PSORS  ...         6 2019-08-12 13:38:21\n",
       "1     58               CORD  ...         6 2020-08-30 09:43:59\n",
       "2    112                 RP  ...         6 2021-01-18 09:53:26\n",
       "3    139                 ID  ...         6 2015-02-09 10:02:49\n",
       "4    173               SLOS  ...         6 2021-12-10 21:51:32\n",
       "5    198                  ?  ...         6 2016-10-22 17:54:40\n",
       "6   2156                  -  ...         6 2021-12-10 21:51:32\n",
       "7   2440               RP25  ...         6 2021-12-10 21:51:32\n",
       "8   4211               RPar  ...      <NA>                 NaT\n",
       "9   4214                  -  ...         1 2023-03-09 14:26:26\n",
       "10  4249  macular dystrophy  ...         6 2024-02-15 21:18:39\n",
       "11  5086                 HL  ...         6 2015-10-23 11:43:00\n",
       "12  5415                USH  ...      <NA>                 NaT\n",
       "13  5468            uveitis  ...      <NA>                 NaT\n",
       "14  6906                DEE  ...      <NA>                 NaT\n",
       "\n",
       "[15 rows x 12 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genes_To_Diseases\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneid</th>\n",
       "      <th>diseaseid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EYS</td>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EYS</td>\n",
       "      <td>2440</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  geneid  diseaseid\n",
       "0    EYS        112\n",
       "1    EYS       2440"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>fatherid</th>\n",
       "      <th>motherid</th>\n",
       "      <th>panelid</th>\n",
       "      <th>panel_size</th>\n",
       "      <th>license</th>\n",
       "      <th>owned_by</th>\n",
       "      <th>Individual/Reference</th>\n",
       "      <th>Individual/Remarks</th>\n",
       "      <th>Individual/Gender</th>\n",
       "      <th>Individual/Consanguinity</th>\n",
       "      <th>Individual/Origin/Geographic</th>\n",
       "      <th>Individual/Age_of_death</th>\n",
       "      <th>Individual/VIP</th>\n",
       "      <th>Individual/Data_av</th>\n",
       "      <th>Individual/Treatment</th>\n",
       "      <th>Individual/Origin/Population</th>\n",
       "      <th>Individual/Individual_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>135</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>{PMID:Marrakchi 2011:21848462}</td>\n",
       "      <td>5-generation family, 3 affecteds (M)</td>\n",
       "      <td>M</td>\n",
       "      <td>yes</td>\n",
       "      <td>Tunisia</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>210</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>39</td>\n",
       "      <td>{PMID:Abu-Safieh-2013:23105016}</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>(Saudi Arabia)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>25</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>M</td>\n",
       "      <td>?</td>\n",
       "      <td>Germany</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>white</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>16605</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>552</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>33096</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>229</td>\n",
       "      <td>{PMID:Neveling 2012:22334370}</td>\n",
       "      <td></td>\n",
       "      <td>M</td>\n",
       "      <td>no</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1445</th>\n",
       "      <td>447702</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>{PMID:Weisschuh 2024:37734845}</td>\n",
       "      <td>patient, no family history</td>\n",
       "      <td>F</td>\n",
       "      <td></td>\n",
       "      <td>Germany</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SRP-1105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1446</th>\n",
       "      <td>447707</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>{PMID:Weisschuh 2024:37734845}</td>\n",
       "      <td>patient, no family history</td>\n",
       "      <td>M</td>\n",
       "      <td></td>\n",
       "      <td>Germany</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SRP-1167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1447</th>\n",
       "      <td>447716</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>{PMID:Weisschuh 2024:37734845}</td>\n",
       "      <td>patient, no family history</td>\n",
       "      <td>F</td>\n",
       "      <td></td>\n",
       "      <td>Germany</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SRP-1249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1448</th>\n",
       "      <td>447718</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>{PMID:Weisschuh 2024:37734845}</td>\n",
       "      <td>patient, no family history</td>\n",
       "      <td>M</td>\n",
       "      <td></td>\n",
       "      <td>Germany</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SRP-1274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1449</th>\n",
       "      <td>447720</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>6</td>\n",
       "      <td>{PMID:Weisschuh 2024:37734845}</td>\n",
       "      <td>patient</td>\n",
       "      <td>M</td>\n",
       "      <td></td>\n",
       "      <td>Germany</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SRP-1299</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1450 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          id fatherid  ... Individual/Origin/Population  Individual/Individual_ID\n",
       "0        135           ...                                                       \n",
       "1        210           ...                                                       \n",
       "2       1962           ...                        white                          \n",
       "3      16605           ...                                                       \n",
       "4      33096           ...                                                       \n",
       "...      ...      ...  ...                          ...                       ...\n",
       "1445  447702           ...                                               SRP-1105\n",
       "1446  447707           ...                                               SRP-1167\n",
       "1447  447716           ...                                               SRP-1249\n",
       "1448  447718           ...                                               SRP-1274\n",
       "1449  447720           ...                                               SRP-1299\n",
       "\n",
       "[1450 rows x 18 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals_To_Diseases\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualid</th>\n",
       "      <th>diseaseid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>135</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>210</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1962</td>\n",
       "      <td>173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>33096</td>\n",
       "      <td>4214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>33109</td>\n",
       "      <td>4214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1444</th>\n",
       "      <td>447702</td>\n",
       "      <td>198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1445</th>\n",
       "      <td>447707</td>\n",
       "      <td>198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1446</th>\n",
       "      <td>447716</td>\n",
       "      <td>198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1447</th>\n",
       "      <td>447718</td>\n",
       "      <td>198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1448</th>\n",
       "      <td>447720</td>\n",
       "      <td>198</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1449 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      individualid  diseaseid\n",
       "0              135         12\n",
       "1              210         58\n",
       "2             1962        173\n",
       "3            33096       4214\n",
       "4            33109       4214\n",
       "...            ...        ...\n",
       "1444        447702        198\n",
       "1445        447707        198\n",
       "1446        447716        198\n",
       "1447        447718        198\n",
       "1448        447720        198\n",
       "\n",
       "[1449 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Phenotypes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>diseaseid</th>\n",
       "      <th>individualid</th>\n",
       "      <th>owned_by</th>\n",
       "      <th>Phenotype/Inheritance</th>\n",
       "      <th>Phenotype/Age</th>\n",
       "      <th>Phenotype/Additional</th>\n",
       "      <th>Phenotype/Biochem_param</th>\n",
       "      <th>Phenotype/Age/Onset</th>\n",
       "      <th>Phenotype/Age/Diagnosis</th>\n",
       "      <th>Phenotype/Severity_score</th>\n",
       "      <th>Phenotype/Onset</th>\n",
       "      <th>Phenotype/Protein</th>\n",
       "      <th>Phenotype/Tumor/MSI</th>\n",
       "      <th>Phenotype/Enzyme/CPK</th>\n",
       "      <th>Phenotype/Heart/Myocardium</th>\n",
       "      <th>Phenotype/Lung</th>\n",
       "      <th>Phenotype/Diagnosis/Definite</th>\n",
       "      <th>Phenotype/Diagnosis/Initial</th>\n",
       "      <th>Phenotype/Diagnosis/Criteria</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>135</td>\n",
       "      <td>6</td>\n",
       "      <td>Familial, autosomal recessive</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26</td>\n",
       "      <td>58</td>\n",
       "      <td>210</td>\n",
       "      <td>39</td>\n",
       "      <td>Familial, autosomal recessive</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>941</td>\n",
       "      <td>173</td>\n",
       "      <td>1962</td>\n",
       "      <td>25</td>\n",
       "      <td>Familial</td>\n",
       "      <td></td>\n",
       "      <td>2-3 toe syndactyly</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26525</td>\n",
       "      <td>4214</td>\n",
       "      <td>33096</td>\n",
       "      <td>229</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26538</td>\n",
       "      <td>4214</td>\n",
       "      <td>33109</td>\n",
       "      <td>229</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1266</th>\n",
       "      <td>336901</td>\n",
       "      <td>198</td>\n",
       "      <td>447702</td>\n",
       "      <td>6</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa, simplex</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1267</th>\n",
       "      <td>336906</td>\n",
       "      <td>198</td>\n",
       "      <td>447707</td>\n",
       "      <td>6</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa, simplex</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1268</th>\n",
       "      <td>336915</td>\n",
       "      <td>198</td>\n",
       "      <td>447716</td>\n",
       "      <td>6</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa, simplex</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269</th>\n",
       "      <td>336917</td>\n",
       "      <td>198</td>\n",
       "      <td>447718</td>\n",
       "      <td>6</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa, simplex</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1270</th>\n",
       "      <td>336919</td>\n",
       "      <td>198</td>\n",
       "      <td>447720</td>\n",
       "      <td>6</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa, simplex</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1271 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          id  ...  Phenotype/Diagnosis/Criteria\n",
       "0          8  ...                              \n",
       "1         26  ...                              \n",
       "2        941  ...                              \n",
       "3      26525  ...                              \n",
       "4      26538  ...                              \n",
       "...      ...  ...                           ...\n",
       "1266  336901  ...                              \n",
       "1267  336906  ...                              \n",
       "1268  336915  ...                              \n",
       "1269  336917  ...                              \n",
       "1270  336919  ...                              \n",
       "\n",
       "[1271 rows x 20 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screenings\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>individualid</th>\n",
       "      <th>variants_found</th>\n",
       "      <th>owned_by</th>\n",
       "      <th>created_by</th>\n",
       "      <th>created_date</th>\n",
       "      <th>edited_by</th>\n",
       "      <th>edited_date</th>\n",
       "      <th>Screening/Technique</th>\n",
       "      <th>Screening/Template</th>\n",
       "      <th>Screening/Tissue</th>\n",
       "      <th>Screening/Remarks</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>126</td>\n",
       "      <td>135</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>2012-07-07 19:04:19</td>\n",
       "      <td>6</td>\n",
       "      <td>2012-07-07 19:12:08</td>\n",
       "      <td>RT-PCR;SEQ</td>\n",
       "      <td>DNA;RNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>211</td>\n",
       "      <td>210</td>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "      <td>6</td>\n",
       "      <td>2012-09-22 11:36:24</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "      <td>SEQ</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1640</td>\n",
       "      <td>1962</td>\n",
       "      <td>1</td>\n",
       "      <td>25</td>\n",
       "      <td>6</td>\n",
       "      <td>2010-03-11 16:36:41</td>\n",
       "      <td>25</td>\n",
       "      <td>2012-04-13 15:18:00</td>\n",
       "      <td>SEQ</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>16557</td>\n",
       "      <td>16605</td>\n",
       "      <td>1</td>\n",
       "      <td>552</td>\n",
       "      <td>552</td>\n",
       "      <td>2014-05-23 13:12:43</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "      <td>SEQ-NG-I</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>33164</td>\n",
       "      <td>33096</td>\n",
       "      <td>1</td>\n",
       "      <td>229</td>\n",
       "      <td>229</td>\n",
       "      <td>2012-02-04 15:20:01</td>\n",
       "      <td>6</td>\n",
       "      <td>2012-05-18 13:59:33</td>\n",
       "      <td>SEQ;SEQ-NG-S</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1445</th>\n",
       "      <td>449279</td>\n",
       "      <td>447702</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>2024-01-26 10:23:59</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "      <td>SEQ-NG</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td>WGS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1446</th>\n",
       "      <td>449284</td>\n",
       "      <td>447707</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>2024-01-26 10:23:59</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "      <td>SEQ-NG</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td>WGS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1447</th>\n",
       "      <td>449293</td>\n",
       "      <td>447716</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>2024-01-26 10:23:59</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "      <td>SEQ-NG</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td>WGS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1448</th>\n",
       "      <td>449295</td>\n",
       "      <td>447718</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>2024-01-26 10:23:59</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "      <td>SEQ-NG</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td>WGS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1449</th>\n",
       "      <td>449297</td>\n",
       "      <td>447720</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>2024-01-26 10:23:59</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaT</td>\n",
       "      <td>SEQ-NG</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td>WGS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1450 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          id  individualid  ...  Screening/Tissue  Screening/Remarks\n",
       "0        126           135  ...                                     \n",
       "1        211           210  ...                                     \n",
       "2       1640          1962  ...                                     \n",
       "3      16557         16605  ...                                     \n",
       "4      33164         33096  ...                                     \n",
       "...      ...           ...  ...               ...                ...\n",
       "1445  449279        447702  ...                                  WGS\n",
       "1446  449284        447707  ...                                  WGS\n",
       "1447  449293        447716  ...                                  WGS\n",
       "1448  449295        447718  ...                                  WGS\n",
       "1449  449297        447720  ...                                  WGS\n",
       "\n",
       "[1450 rows x 12 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screenings_To_Genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>screeningid</th>\n",
       "      <th>geneid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>126</td>\n",
       "      <td>IL36RN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>211</td>\n",
       "      <td>MKS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1640</td>\n",
       "      <td>DHCR7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>33164</td>\n",
       "      <td>AHI1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>33164</td>\n",
       "      <td>EYS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1311</th>\n",
       "      <td>437646</td>\n",
       "      <td>EYS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1312</th>\n",
       "      <td>437902</td>\n",
       "      <td>EYS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1313</th>\n",
       "      <td>437922</td>\n",
       "      <td>EYS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1314</th>\n",
       "      <td>443144</td>\n",
       "      <td>EYS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1315</th>\n",
       "      <td>443145</td>\n",
       "      <td>EYS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1316 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      screeningid  geneid\n",
       "0             126  IL36RN\n",
       "1             211    MKS1\n",
       "2            1640   DHCR7\n",
       "3           33164    AHI1\n",
       "4           33164     EYS\n",
       "...           ...     ...\n",
       "1311       437646     EYS\n",
       "1312       437902     EYS\n",
       "1313       437922     EYS\n",
       "1314       443144     EYS\n",
       "1315       443145     EYS\n",
       "\n",
       "[1316 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants_On_Genome\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>allele</th>\n",
       "      <th>effectid</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position_g_start</th>\n",
       "      <th>position_g_end</th>\n",
       "      <th>type</th>\n",
       "      <th>average_frequency</th>\n",
       "      <th>owned_by</th>\n",
       "      <th>VariantOnGenome/DBID</th>\n",
       "      <th>VariantOnGenome/DNA</th>\n",
       "      <th>VariantOnGenome/Frequency</th>\n",
       "      <th>VariantOnGenome/Reference</th>\n",
       "      <th>VariantOnGenome/Restriction_site</th>\n",
       "      <th>VariantOnGenome/Published_as</th>\n",
       "      <th>VariantOnGenome/Remarks</th>\n",
       "      <th>VariantOnGenome/Genetic_origin</th>\n",
       "      <th>VariantOnGenome/Segregation</th>\n",
       "      <th>VariantOnGenome/dbSNP</th>\n",
       "      <th>VariantOnGenome/VIP</th>\n",
       "      <th>VariantOnGenome/Methylation</th>\n",
       "      <th>VariantOnGenome/ISCN</th>\n",
       "      <th>VariantOnGenome/DNA/hg38</th>\n",
       "      <th>VariantOnGenome/ClinVar</th>\n",
       "      <th>VariantOnGenome/ClinicalClassification</th>\n",
       "      <th>VariantOnGenome/ClinicalClassification/Method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>36426</td>\n",
       "      <td>3</td>\n",
       "      <td>50</td>\n",
       "      <td>6</td>\n",
       "      <td>64498971</td>\n",
       "      <td>64498971</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.000743</td>\n",
       "      <td>552</td>\n",
       "      <td>EYS_000007</td>\n",
       "      <td>g.64498971A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>Germline</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>g.63789078A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>VUS</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>59881</td>\n",
       "      <td>3</td>\n",
       "      <td>55</td>\n",
       "      <td>6</td>\n",
       "      <td>65655758</td>\n",
       "      <td>65655758</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.001153</td>\n",
       "      <td>229</td>\n",
       "      <td>EYS_000001</td>\n",
       "      <td>g.65655758T&gt;G</td>\n",
       "      <td>ExAC: 60, 19750, 0, 0.003038</td>\n",
       "      <td>{PMID:Neveling 2012:22334370}</td>\n",
       "      <td></td>\n",
       "      <td>Q770P</td>\n",
       "      <td></td>\n",
       "      <td>Germline</td>\n",
       "      <td>yes</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>g.64945865T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>VUS</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>59883</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>65336143</td>\n",
       "      <td>65336143</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.224189</td>\n",
       "      <td>229</td>\n",
       "      <td>EYS_000002</td>\n",
       "      <td>g.65336143G&gt;A</td>\n",
       "      <td>ExAC: 3936, 19366, 441, 0.2032</td>\n",
       "      <td>{PMID:Neveling 2012:22334370}</td>\n",
       "      <td></td>\n",
       "      <td>p.?</td>\n",
       "      <td>unaffected brother also this variant homozygous</td>\n",
       "      <td>Germline</td>\n",
       "      <td>no</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>g.64626250G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>59884</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>6</td>\n",
       "      <td>65300869</td>\n",
       "      <td>65300869</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.000838</td>\n",
       "      <td>229</td>\n",
       "      <td>EYS_000003</td>\n",
       "      <td>g.65300869G&gt;A</td>\n",
       "      <td>ExAC: 12, 19406, 0, 0.0006184</td>\n",
       "      <td>{PMID:Neveling 2012:22334370}</td>\n",
       "      <td></td>\n",
       "      <td>(P1631S)</td>\n",
       "      <td>predicted benign, disease-related variant in o...</td>\n",
       "      <td>Germline</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>g.64590976G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>59885</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>65016998</td>\n",
       "      <td>65016999</td>\n",
       "      <td>del</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>229</td>\n",
       "      <td>EYS_000004</td>\n",
       "      <td>g.65016998_65016999del</td>\n",
       "      <td>ExAC: 9866, 18292, 921, 0.5394</td>\n",
       "      <td>{PMID:Neveling 2012:22334370}</td>\n",
       "      <td></td>\n",
       "      <td>6045-4_6045-3del</td>\n",
       "      <td>predicted benign</td>\n",
       "      <td>Germline</td>\n",
       "      <td>yes</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>g.64307105_64307106del</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2536</th>\n",
       "      <td>964211</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "      <td>65767634</td>\n",
       "      <td>65767634</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.243022</td>\n",
       "      <td>2330</td>\n",
       "      <td>EYS_000248</td>\n",
       "      <td>g.65767634G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>EYS(NM_001292009.2):c.2024-15_2024-14delTCinsTT</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>CLASSIFICATION record</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>likely benign</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2537</th>\n",
       "      <td>964212</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "      <td>65767643</td>\n",
       "      <td>65767643</td>\n",
       "      <td>del</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2330</td>\n",
       "      <td>EYS_000926</td>\n",
       "      <td>g.65767643del</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>EYS(NM_001292009.2):c.2024-15delT</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>CLASSIFICATION record</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>likely benign</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2538</th>\n",
       "      <td>964215</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>6</td>\n",
       "      <td>66005927</td>\n",
       "      <td>66005927</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.000112</td>\n",
       "      <td>2327</td>\n",
       "      <td>EYS_000253</td>\n",
       "      <td>g.66005927C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>EYS(NM_001142800.1):c.1852G&gt;A (p.G618S), EYS(N...</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>CLASSIFICATION record</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VUS</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2539</th>\n",
       "      <td>964216</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>6</td>\n",
       "      <td>66044874</td>\n",
       "      <td>66044874</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.000082</td>\n",
       "      <td>2327</td>\n",
       "      <td>EYS_000256</td>\n",
       "      <td>g.66044874T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>EYS(NM_001292009.2):c.1765A&gt;G (p.R589G)</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>CLASSIFICATION record</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VUS</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2540</th>\n",
       "      <td>977314</td>\n",
       "      <td>0</td>\n",
       "      <td>90</td>\n",
       "      <td>6</td>\n",
       "      <td>64430943</td>\n",
       "      <td>64430943</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>1804</td>\n",
       "      <td>EYS_000060</td>\n",
       "      <td>g.64430943A&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>EYS(NM_001142800.2):c.8984T&gt;A (p.(Ile2995Asn))...</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>CLASSIFICATION record</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>pathogenic</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2541 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          id  ...  VariantOnGenome/ClinicalClassification/Method\n",
       "0      36426  ...                                               \n",
       "1      59881  ...                                               \n",
       "2      59883  ...                                               \n",
       "3      59884  ...                                               \n",
       "4      59885  ...                                               \n",
       "...      ...  ...                                            ...\n",
       "2536  964211  ...                                               \n",
       "2537  964212  ...                                               \n",
       "2538  964215  ...                                               \n",
       "2539  964216  ...                                               \n",
       "2540  977314  ...                                               \n",
       "\n",
       "[2541 rows x 26 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants_On_Transcripts\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>transcriptid</th>\n",
       "      <th>effectid</th>\n",
       "      <th>position_c_start</th>\n",
       "      <th>position_c_start_intron</th>\n",
       "      <th>position_c_end</th>\n",
       "      <th>position_c_end_intron</th>\n",
       "      <th>VariantOnTranscript/DNA</th>\n",
       "      <th>VariantOnTranscript/RNA</th>\n",
       "      <th>VariantOnTranscript/Protein</th>\n",
       "      <th>VariantOnTranscript/Exon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>36426</td>\n",
       "      <td>7329</td>\n",
       "      <td>50</td>\n",
       "      <td>7558</td>\n",
       "      <td>0</td>\n",
       "      <td>7558</td>\n",
       "      <td>0</td>\n",
       "      <td>c.7558T&gt;C</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Phe2520Leu)</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>59881</td>\n",
       "      <td>7329</td>\n",
       "      <td>55</td>\n",
       "      <td>2309</td>\n",
       "      <td>0</td>\n",
       "      <td>2309</td>\n",
       "      <td>0</td>\n",
       "      <td>c.2309A&gt;C</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Gln770Pro)</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>59883</td>\n",
       "      <td>7329</td>\n",
       "      <td>11</td>\n",
       "      <td>3444</td>\n",
       "      <td>-5</td>\n",
       "      <td>3444</td>\n",
       "      <td>-5</td>\n",
       "      <td>c.3444-5C&gt;T</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>22i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>59884</td>\n",
       "      <td>7329</td>\n",
       "      <td>15</td>\n",
       "      <td>4891</td>\n",
       "      <td>0</td>\n",
       "      <td>4891</td>\n",
       "      <td>0</td>\n",
       "      <td>c.4891C&gt;T</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Pro1631Ser)</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>59885</td>\n",
       "      <td>7329</td>\n",
       "      <td>11</td>\n",
       "      <td>6079</td>\n",
       "      <td>-4</td>\n",
       "      <td>6079</td>\n",
       "      <td>-3</td>\n",
       "      <td>c.6079-4_6079-3del</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>29i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2536</th>\n",
       "      <td>964211</td>\n",
       "      <td>7329</td>\n",
       "      <td>30</td>\n",
       "      <td>2024</td>\n",
       "      <td>-14</td>\n",
       "      <td>2024</td>\n",
       "      <td>-14</td>\n",
       "      <td>c.2024-14C&gt;T</td>\n",
       "      <td>r.(=)</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2537</th>\n",
       "      <td>964212</td>\n",
       "      <td>7329</td>\n",
       "      <td>30</td>\n",
       "      <td>2024</td>\n",
       "      <td>-15</td>\n",
       "      <td>2024</td>\n",
       "      <td>-15</td>\n",
       "      <td>c.2024-15del</td>\n",
       "      <td>r.(=)</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2538</th>\n",
       "      <td>964215</td>\n",
       "      <td>7329</td>\n",
       "      <td>50</td>\n",
       "      <td>1852</td>\n",
       "      <td>0</td>\n",
       "      <td>1852</td>\n",
       "      <td>0</td>\n",
       "      <td>c.1852G&gt;A</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Gly618Ser)</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2539</th>\n",
       "      <td>964216</td>\n",
       "      <td>7329</td>\n",
       "      <td>50</td>\n",
       "      <td>1765</td>\n",
       "      <td>0</td>\n",
       "      <td>1765</td>\n",
       "      <td>0</td>\n",
       "      <td>c.1765A&gt;G</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Arg589Gly)</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2540</th>\n",
       "      <td>977314</td>\n",
       "      <td>7329</td>\n",
       "      <td>90</td>\n",
       "      <td>8984</td>\n",
       "      <td>0</td>\n",
       "      <td>8984</td>\n",
       "      <td>0</td>\n",
       "      <td>c.8984T&gt;A</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Ile2995Asn)</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2541 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          id  transcriptid  ...  VariantOnTranscript/Protein  VariantOnTranscript/Exon\n",
       "0      36426          7329  ...               p.(Phe2520Leu)                        38\n",
       "1      59881          7329  ...                p.(Gln770Pro)                        15\n",
       "2      59883          7329  ...                        p.(=)                       22i\n",
       "3      59884          7329  ...               p.(Pro1631Ser)                        26\n",
       "4      59885          7329  ...                        p.(=)                       29i\n",
       "...      ...           ...  ...                          ...                       ...\n",
       "2536  964211          7329  ...                        p.(=)                          \n",
       "2537  964212          7329  ...                        p.(=)                          \n",
       "2538  964215          7329  ...                p.(Gly618Ser)                          \n",
       "2539  964216          7329  ...                p.(Arg589Gly)                          \n",
       "2540  977314          7329  ...               p.(Ile2995Asn)                          \n",
       "\n",
       "[2541 rows x 11 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screenings_To_Variants\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>screeningid</th>\n",
       "      <th>variantid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>126</td>\n",
       "      <td>783293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>211</td>\n",
       "      <td>790459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1640</td>\n",
       "      <td>235838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>16557</td>\n",
       "      <td>36426</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>33164</td>\n",
       "      <td>59884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2144</th>\n",
       "      <td>449279</td>\n",
       "      <td>959046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2145</th>\n",
       "      <td>449284</td>\n",
       "      <td>959051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2146</th>\n",
       "      <td>449293</td>\n",
       "      <td>959060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2147</th>\n",
       "      <td>449295</td>\n",
       "      <td>959474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2148</th>\n",
       "      <td>449297</td>\n",
       "      <td>959064</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2149 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      screeningid  variantid\n",
       "0             126     783293\n",
       "1             211     790459\n",
       "2            1640     235838\n",
       "3           16557      36426\n",
       "4           33164      59884\n",
       "...           ...        ...\n",
       "2144       449279     959046\n",
       "2145       449284     959051\n",
       "2146       449293     959060\n",
       "2147       449295     959474\n",
       "2148       449297     959064\n",
       "\n",
       "[2149 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for i in data:\n",
    "    print(i)\n",
    "    display(data[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef07740b2fa63e42",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "set_lovd_dtypes(data)\n",
    "for i in data:\n",
    "    print(i)\n",
    "    display(data[i].info())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c968af1617be40db",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-13T15:38:25.149624Z",
     "start_time": "2024-05-13T15:38:24.807199Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:root:Skipping variant g.64307105_64307106del\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64902422_64902438del\n",
      "WARNING:root:Skipping variant g.64902422_64902438del\n",
      "WARNING:root:Skipping variant g.64840707_64997105del\n",
      "WARNING:root:Skipping variant g.64840707_64997105del\n",
      "WARNING:root:Skipping variant g.64840707_64997105del\n",
      "WARNING:root:Skipping variant g.65295915del\n",
      "WARNING:root:Skipping variant g.65295915del\n",
      "WARNING:root:Skipping variant g.65295915del\n",
      "WARNING:root:Skipping variant g.65057728_65320715del\n",
      "WARNING:root:Skipping variant g.65057728_65320715del\n",
      "WARNING:root:Skipping variant g.65057728_65320715del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65384425del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64439165del\n",
      "WARNING:root:Skipping variant g.64439165del\n",
      "WARNING:root:Skipping variant g.64626122del\n",
      "WARNING:root:Skipping variant g.65494867del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65494867del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.63720919_63720920del\n",
      "WARNING:root:Skipping variant g.63720668dup\n",
      "WARNING:root:Skipping variant g.63720919_63720920del\n",
      "WARNING:root:Skipping variant g.63720668dup\n",
      "WARNING:root:Skipping variant g.63999116del\n",
      "WARNING:root:Skipping variant g.64591039_64591042del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64822643dup\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.64591505_64591506delinsCT\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65384480dup\n",
      "WARNING:root:Skipping variant g.63726599_63726600del\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.63726599_63726600del\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.63726599_63726600del\n",
      "WARNING:root:Skipping variant g.65335102del\n",
      "WARNING:root:Skipping variant g.65335102del\n",
      "WARNING:root:Skipping variant g.65335102del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63999116del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63999110_63999111del\n",
      "WARNING:root:Skipping variant g.63999110_63999111del\n",
      "WARNING:root:Skipping variant g.65321830_65370656del\n",
      "WARNING:root:Skipping variant g.63720850_63720853del\n",
      "WARNING:root:Skipping variant g.65321830_65370656del\n",
      "WARNING:root:Skipping variant g.63720850_63720853del\n",
      "WARNING:root:Skipping variant g.65321830_65370656del\n",
      "WARNING:root:Skipping variant g.63720850_63720853del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63806240del\n",
      "WARNING:root:Skipping variant g.63806240del\n",
      "WARNING:root:Skipping variant g.65495332_65495333dup\n",
      "WARNING:root:Skipping variant g.65324960_65416038del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65295915del\n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.65384480dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.65344144_65344152delinsCTTTTCG\n",
      "WARNING:root:Skipping variant g.63984409_63984410delinsACGAT\n",
      "WARNING:root:Skipping variant g.63788163_63788164del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64912603dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590700dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590700dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64591845del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63984390del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63721651_63721652insCA\n",
      "WARNING:root:Skipping variant g.64590665_64590666del\n",
      "WARNING:root:Skipping variant g.63721651_63721652insCA\n",
      "WARNING:root:Skipping variant g.64590665_64590666del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64591466dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64591480del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63788136del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63720730_63720733del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64912603dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590665_64590666del\n",
      "WARNING:root:Skipping variant g.63721652_63721655dup\n",
      "WARNING:root:Skipping variant g.64590665_64590666del\n",
      "WARNING:root:Skipping variant g.63721652_63721655dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64912603dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64591313del\n",
      "WARNING:root:Skipping variant g.64439355_64439356insA\n",
      "WARNING:root:Skipping variant g.63984543_63984570del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63984537_63984542dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64614795_64939832del\n",
      "WARNING:root:Skipping variant g.64614795_64939832del\n",
      "WARNING:root:Skipping variant g.64614795_64939832del\n",
      "WARNING:root:Skipping variant g.63721576del\n",
      "WARNING:root:Skipping variant g.64902132_64902133del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720889dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63721625dup\n",
      "WARNING:root:Skipping variant g.65405300dup\n",
      "WARNING:root:Skipping variant g.64813506del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63720753_63720754dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.65707136_65707226del\n",
      "WARNING:root:Skipping variant g.65495348del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63720845_63720846del\n",
      "WARNING:root:Skipping variant g.63726607del\n",
      "WARNING:root:Skipping variant g.63721786dup\n",
      "WARNING:root:Skipping variant g.63721640del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65344138dup\n",
      "WARNING:root:Skipping variant g.65402503del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590699_64590700del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590699_64590700del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65384384_65384387del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65384480dup\n",
      "WARNING:root:Skipping variant g.63806228dup\n",
      "WARNING:root:Skipping variant g.63999116del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.65384384_65384387del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63999110_63999111del\n",
      "WARNING:root:Skipping variant g.63999110_63999111del\n",
      "WARNING:root:Skipping variant g.63999110_63999111del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64439331del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65494957dup\n",
      "WARNING:root:Skipping variant g.63720995del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63721771_63721776del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.65494957dup\n",
      "WARNING:root:Skipping variant g.64591977del\n",
      "WARNING:root:Skipping variant g.64886728_64886736del\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant g.63721138del\n",
      "WARNING:root:Skipping variant g.65353541del\n",
      "WARNING:root:Skipping variant g.65353541del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65295857dup\n",
      "WARNING:root:Skipping variant g.65295856_65295857dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64307105_64307106del\n",
      "WARNING:root:Skipping variant g.64307103_64307106del\n",
      "WARNING:root:Skipping variant g.63726618_63726622del\n",
      "WARNING:root:Skipping variant g.63720642_63720644del\n",
      "WARNING:root:Skipping variant g.65295857dup\n",
      "WARNING:root:Skipping variant g.64307105_64307106del\n",
      "WARNING:root:Skipping variant g.65353541del\n",
      "WARNING:root:Skipping variant g.64591918_64591919del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64590525_64590548del\n",
      "WARNING:root:Skipping variant g.64388841_64388843del\n",
      "WARNING:root:Skipping variant g.64307084_64307085del\n",
      "WARNING:root:Skipping variant g.64307105_64307106del\n",
      "WARNING:root:Skipping variant g.64307103_64307106del\n",
      "WARNING:root:Skipping variant g.64307105_64307106del\n",
      "WARNING:root:Skipping variant g.64593097_64593101del\n",
      "WARNING:root:Skipping variant g.65384384_65384387del\n",
      "WARNING:root:Skipping variant g.63999110_63999111del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64439213del\n",
      "WARNING:root:Skipping variant g.63721771_63721776del\n",
      "WARNING:root:Skipping variant g.65494957dup\n",
      "WARNING:root:Skipping variant g.65494988_65495008delinsAAAAG\n",
      "WARNING:root:Skipping variant g.63720799_63720808del\n",
      "WARNING:root:Skipping variant g.63726584del\n",
      "WARNING:root:Skipping variant g.63726599_63726600del\n",
      "WARNING:root:Skipping variant g.63726648del\n",
      "WARNING:root:Skipping variant g.64349976_64426764del\n",
      "WARNING:root:Skipping variant g.64591505_64591506delinsCT\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720728_63720729del\n",
      "WARNING:root:Skipping variant g.63720845_63720846del\n",
      "WARNING:root:Skipping variant g.63721162del\n",
      "WARNING:root:Skipping variant g.63721237_63721240del\n",
      "WARNING:root:Skipping variant g.63721343del\n",
      "WARNING:root:Skipping variant g.63721730_63721733del\n",
      "WARNING:root:Skipping variant g.63721763_63721767del\n",
      "WARNING:root:Skipping variant g.63721787dup\n",
      "WARNING:root:Skipping variant g.63726618_63726622del\n",
      "WARNING:root:Skipping variant g.63788163_63788164del\n",
      "WARNING:root:Skipping variant g.63984389del\n",
      "WARNING:root:Skipping variant g.64066348del\n",
      "WARNING:root:Skipping variant g.64590909dup\n",
      "WARNING:root:Skipping variant g.64591309del\n",
      "WARNING:root:Skipping variant g.64591858_64591859del\n",
      "WARNING:root:Skipping variant g.64886728_64886736del\n",
      "WARNING:root:Skipping variant g.64945814del\n",
      "WARNING:root:Skipping variant g.65344143_65344144insCTTT\n",
      "WARNING:root:Skipping variant g.65344146_65344151del\n",
      "WARNING:root:Skipping variant g.65344181dup\n",
      "WARNING:root:Skipping variant g.65384473dup\n",
      "WARNING:root:Skipping variant g.65405287del\n",
      "WARNING:root:Skipping variant g.65494885_65494887del\n",
      "WARNING:root:Skipping variant g.65494961del\n",
      "WARNING:root:Skipping variant g.65495205del\n",
      "WARNING:root:Skipping variant g.65495296_65495297del\n",
      "WARNING:root:Skipping variant g.64590909dup\n",
      "WARNING:root:Skipping variant g.65384473dup\n",
      "WARNING:root:Skipping variant g.65405287del\n",
      "WARNING:root:Skipping variant g.65494885_65494887del\n",
      "WARNING:root:Skipping variant g.63762589del\n",
      "WARNING:root:Skipping variant g.63720874del\n",
      "WARNING:root:Skipping variant g.63721619_63721620insGT\n",
      "WARNING:root:Skipping variant g.63788268dup\n",
      "WARNING:root:Skipping variant g.63984369_63984392del\n",
      "WARNING:root:Skipping variant g.65057740_65057741insA\n",
      "WARNING:root:Skipping variant g.65057740_65057741insAA\n",
      "WARNING:root:Skipping variant g.65057750dup\n",
      "WARNING:root:Skipping variant g.65384384_65384387del\n",
      "WARNING:root:Skipping variant g.65402624dup\n",
      "WARNING:root:Skipping variant g.65405377dup\n",
      "WARNING:root:Skipping variant g.65494957dup\n",
      "WARNING:root:Skipping variant g.65494957dup\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant g.63721625dup\n",
      "WARNING:root:Skipping variant g.63721704dup\n",
      "WARNING:root:Skipping variant g.65353537dup\n",
      "WARNING:root:Skipping variant g.65335105_65335108del\n",
      "WARNING:root:Skipping variant g.63721385_63721386del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64439195del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63720867_63720868del\n",
      "WARNING:root:Skipping variant g.63721314_63721321del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63765706_63791377del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63781919_63803805del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64892926_64948294del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64591845del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64945857dup\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63721557_63721558insA\n",
      "WARNING:root:Skipping variant g.64439319_64439323del\n",
      "WARNING:root:Skipping variant g.65296021del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65384425del\n",
      "WARNING:root:Skipping variant g.64591039_64591042del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63721599_63721604del\n",
      "WARNING:root:Skipping variant g.64902132_64902133del\n",
      "WARNING:root:Skipping variant g.65494988_65495008delinsAAAAG\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65495236del\n",
      "WARNING:root:Skipping variant g.64590908_64590909insT\n",
      "WARNING:root:Skipping variant g.64617409_64617411dup\n",
      "WARNING:root:Skipping variant g.65295857dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63720845_63720846del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720668dup\n",
      "WARNING:root:Skipping variant g.(63741975_63762461)_(63778180_63788105)dup\n",
      "WARNING:root:Skipping variant g.65296051_65296052del\n",
      "WARNING:root:Skipping variant g.65296051_65296052del\n",
      "WARNING:root:Skipping variant g.64439200dup\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.65494885_65494887del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63984390del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.65295897del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65494885_65494887del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590615dup\n",
      "WARNING:root:Skipping variant g.63721705del\n",
      "WARNING:root:Skipping variant g.64081884del\n",
      "WARNING:root:Skipping variant g.65405342_65405355delinsAAA\n",
      "WARNING:root:Skipping variant g.65490640_65490643dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65296058del\n",
      "WARNING:root:Skipping variant g.65405345del\n",
      "WARNING:root:Skipping variant g.(65405368_65490593)_(65495411_?)del\n",
      "WARNING:root:Skipping variant g.(65353618_65384385)_(65405368_65490593)del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64081885_64081888del\n",
      "WARNING:root:Skipping variant g.64591401_64591408dup\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63721237_63721240del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64591256_64591272del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.63721436del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64591039_64591042del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.63720695_63720714del\n",
      "WARNING:root:Skipping variant g.65384480dup\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.65495064del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64066470_64066473del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64066470_64066473del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.63720695_63720714del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720668del\n",
      "WARNING:root:Skipping variant g.64912603dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63720730_63720733del\n",
      "WARNING:root:Skipping variant g.64902132_64902133del\n",
      "WARNING:root:Skipping variant g.63720954dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64066335del\n",
      "WARNING:root:Skipping variant g.64886711del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.63778166_63778172del\n",
      "WARNING:root:Skipping variant g.64590914dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63726618_63726622del\n",
      "WARNING:root:Skipping variant g.64886728_64886736del\n",
      "WARNING:root:Skipping variant g.64893157_64947352del\n",
      "WARNING:root:Skipping variant g.64797009_64846087del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.63726519del\n",
      "WARNING:root:Skipping variant g.64997275_64998015del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64081605_64082252del\n",
      "WARNING:root:Skipping variant g.63777755_63789474dup\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.64081605_64082252del\n",
      "WARNING:root:Skipping variant g.65353208_65353867del\n",
      "WARNING:root:Skipping variant g.63984134_63984854del\n",
      "WARNING:root:Skipping variant g.63743557_63907234del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.64662532_64979780del\n",
      "WARNING:root:Skipping variant g.64839119_64970113del\n",
      "WARNING:root:Skipping variant g.63720872_63720873del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64081885_64081888del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720682dup\n",
      "WARNING:root:Skipping variant g.63721619dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64388841del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63720649_63720653del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63720695_63720714del\n",
      "WARNING:root:Skipping variant g.65384384_65384387del\n",
      "WARNING:root:Skipping variant g.63720730_63720733del\n",
      "WARNING:root:Skipping variant g.65335105_65335108del\n",
      "WARNING:root:Skipping variant g.64439165del\n",
      "WARNING:root:Skipping variant g.63999116del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63998527_64002156del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63720730_63720733del\n",
      "WARNING:root:Skipping variant g.63720730_63720733del\n",
      "WARNING:root:Skipping variant g.63721237_63721240del\n",
      "WARNING:root:Skipping variant g.63721237_63721240del\n",
      "WARNING:root:Skipping variant g.63721237_63721240del\n",
      "WARNING:root:Skipping variant g.64081885_64081888del\n",
      "WARNING:root:Skipping variant g.65495348dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64439331del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64912705del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64439200dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63721625dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.65384480del\n",
      "WARNING:root:Skipping variant g.64388841_64388843del\n",
      "WARNING:root:Skipping variant g.63998527_64002156del\n",
      "WARNING:root:Skipping variant g.64590525_64590550delinsTA\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64081884del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.216327637C>T\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.215879068C>T\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.215878931del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.65384480dup\n",
      "WARNING:root:Skipping variant g.216073265G>A\n",
      "WARNING:root:Skipping variant g.65495206_65495207insTGCCAGTTTA\n",
      "WARNING:root:Skipping variant g.63721227dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63720990_63720991insATAT\n",
      "WARNING:root:Skipping variant g.63720728_63720729del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720947_63720948insT\n",
      "WARNING:root:Skipping variant g.64590875_64590876insTCTT\n",
      "WARNING:root:Skipping variant g.63721432del\n",
      "WARNING:root:Skipping variant g.64591501_64591502insAGAA\n",
      "WARNING:root:Skipping variant g.64590556_64590566del\n",
      "WARNING:root:Skipping variant g.63999116del\n",
      "WARNING:root:Skipping variant g.63720753_63720754dup\n",
      "WARNING:root:Skipping variant g.(?_64945792)_(64945915_64997581)del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720730_63720733del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63726524del\n",
      "WARNING:root:Skipping variant g.64423168_64798957delinsATGA\n",
      "WARNING:root:Skipping variant g.63942752_64337822delinsATTATG\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64423168_64798957delinsATGA\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63942752_64337822delinsATTATG\n",
      "WARNING:root:Skipping variant g.63957115_63958454del\n",
      "WARNING:root:Skipping variant g.65001113_65005820del\n",
      "WARNING:root:Skipping variant g.65550144_65552138del\n",
      "WARNING:root:Skipping variant g.65689153_65694794del\n",
      "WARNING:root:Skipping variant g.65454073_65454074insN[305]\n",
      "WARNING:root:Skipping variant g.64296539_64296632del\n",
      "WARNING:root:Skipping variant g.65204982_65205044del\n",
      "WARNING:root:Skipping variant g.65564961_65565284del\n",
      "WARNING:root:Skipping variant :g.64295412_64295413insN[118]\n",
      "WARNING:root:Skipping variant g.65278328_65278329insN[59]\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590875_64590876insTCTT\n",
      "WARNING:root:Skipping variant g.63720947_63720948insT\n",
      "WARNING:root:Skipping variant g.64591501_64591502insAGAA\n",
      "WARNING:root:Skipping variant g.63721432del\n",
      "WARNING:root:Skipping variant g.64590556_64590566del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65658176_65718924del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant NM_001142800.2:c.6079-2A>G\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63999116del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64986218_65013355del\n",
      "WARNING:root:Skipping variant g.64388690_64388840del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.(?_63719980)_(63726681_63762460)dup\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65274506_65316845delinsAGATCA\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.65494100_65508832del\n",
      "WARNING:root:Skipping variant g.65213025_65296862delinsGTTTTCTTTTTA\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64602159_64657461dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63762589del\n",
      "WARNING:root:Skipping variant g.65383303_65441305delinsAACTTTTACT\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.65284957_66872862delinsT\n",
      "WARNING:root:Skipping variant g.64122444_64129159delins64204448_64235506inv\n",
      "WARNING:root:Skipping variant g.64937848_64948401delins[64944099_64944163inv;CAATTTTGTAT]\n",
      "WARNING:root:Skipping variant g.63721385_63721386del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63720654_63720657del\n",
      "WARNING:root:Skipping variant g.64591069_64591081del\n",
      "WARNING:root:Skipping variant g.64886841del\n",
      "WARNING:root:Skipping variant g.64790603_64977512del\n",
      "WARNING:root:Skipping variant g.63721625dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63720649_63720653del\n",
      "WARNING:root:Skipping variant g.65479942_67131267inv\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n"
     ]
    }
   ],
   "source": [
    "save_lovd_as_vcf(data[\"Variants_On_Genome\"], \"./lovd.vcf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c7ff16903e0c52bd",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-13T15:58:47.868055Z",
     "start_time": "2024-05-13T15:58:41.380466Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-05-13 18:58:41.794056: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.\n",
      "2024-05-13 18:58:41.794769: I external/local_tsl/tsl/cuda/cudart_stub.cc:32] Could not find cuda drivers on your machine, GPU will not be used.\n",
      "2024-05-13 18:58:41.797917: I external/local_tsl/tsl/cuda/cudart_stub.cc:32] Could not find cuda drivers on your machine, GPU will not be used.\n",
      "2024-05-13 18:58:41.857361: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: AVX2 AVX_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2024-05-13 18:58:42.410244: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT\n",
      "2024-05-13 18:58:42.957291: I external/local_xla/xla/stream_executor/cuda/cuda_executor.cc:998] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-05-13 18:58:42.957684: W tensorflow/core/common_runtime/gpu/gpu_device.cc:2251] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.\n",
      "Skipping registering GPU devices...\n",
      "WARNING:absl:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.\n",
      "WARNING:absl:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.\n",
      "WARNING:absl:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.\n",
      "WARNING:absl:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.\n",
      "WARNING:absl:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001B[1m1/1\u001B[0m \u001B[32m━━━━━━━━━━━━━━━━━━━━\u001B[0m\u001B[37m\u001B[0m \u001B[1m1s\u001B[0m 595ms/step\n",
      "\u001B[1m1/1\u001B[0m \u001B[32m━━━━━━━━━━━━━━━━━━━━\u001B[0m\u001B[37m\u001B[0m \u001B[1m1s\u001B[0m 554ms/step\n",
      "\u001B[1m1/1\u001B[0m \u001B[32m━━━━━━━━━━━━━━━━━━━━\u001B[0m\u001B[37m\u001B[0m \u001B[1m1s\u001B[0m 553ms/step\n",
      "\u001B[1m1/1\u001B[0m \u001B[32m━━━━━━━━━━━━━━━━━━━━\u001B[0m\u001B[37m\u001B[0m \u001B[1m1s\u001B[0m 548ms/step\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:5 out of the last 5 calls to <function TensorFlowTrainer.make_predict_function.<locals>.one_step_on_data_distributed at 0x7dee981abf40> triggered tf.function retracing. Tracing is expensive and the excessive number of tracings could be due to (1) creating @tf.function repeatedly in a loop, (2) passing tensors with different shapes, (3) passing Python objects instead of tensors. For (1), please define your @tf.function outside of the loop. For (2), @tf.function has reduce_retracing=True option that can avoid unnecessary retracing. For (3), please refer to https://www.tensorflow.org/guide/function#controlling_retracing and https://www.tensorflow.org/api_docs/python/tf/function for  more details.\n",
      "WARNING:tensorflow:5 out of the last 5 calls to <function TensorFlowTrainer.make_predict_function.<locals>.one_step_on_data_distributed at 0x7dee981abf40> triggered tf.function retracing. Tracing is expensive and the excessive number of tracings could be due to (1) creating @tf.function repeatedly in a loop, (2) passing tensors with different shapes, (3) passing Python objects instead of tensors. For (1), please define your @tf.function outside of the loop. For (2), @tf.function has reduce_retracing=True option that can avoid unnecessary retracing. For (3), please refer to https://www.tensorflow.org/guide/function#controlling_retracing and https://www.tensorflow.org/api_docs/python/tf/function for  more details.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001B[1m1/1\u001B[0m \u001B[32m━━━━━━━━━━━━━━━━━━━━\u001B[0m\u001B[37m\u001B[0m \u001B[1m1s\u001B[0m 548ms/step\n",
      "\u001B[1m1/1\u001B[0m \u001B[32m━━━━━━━━━━━━━━━━━━━━\u001B[0m\u001B[37m\u001B[0m \u001B[1m0s\u001B[0m 49ms/step\n",
      "\u001B[1m1/1\u001B[0m \u001B[32m━━━━━━━━━━━━━━━━━━━━\u001B[0m\u001B[37m\u001B[0m \u001B[1m0s\u001B[0m 51ms/step\n",
      "\u001B[1m1/1\u001B[0m \u001B[32m━━━━━━━━━━━━━━━━━━━━\u001B[0m\u001B[37m\u001B[0m \u001B[1m0s\u001B[0m 51ms/step\n",
      "\u001B[1m1/1\u001B[0m \u001B[32m━━━━━━━━━━━━━━━━━━━━\u001B[0m\u001B[37m\u001B[0m \u001B[1m0s\u001B[0m 59ms/step\n",
      "\u001B[1m1/1\u001B[0m \u001B[32m━━━━━━━━━━━━━━━━━━━━\u001B[0m\u001B[37m\u001B[0m \u001B[1m0s\u001B[0m 53ms/step\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:root:dede['G|EYS|0.00|0.00|0.00|0.00|3|9|-20|9']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from subprocess import Popen\n",
    "\n",
    "\n",
    "process = Popen(\"spliceai -I ./lovd.vcf -O ./lovd_output.vcf -R ../tools/spliceai/hg38.fa -A grch38\".split())\n",
    "process.wait()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0514ccc3-5c91-41ad-ab15-f4158030ea14",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/nojsaj/kath/tools/revel/revel.py:9: DtypeWarning: Columns (0,2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  revel_data = pd.read_csv(revel_file)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>hg19_pos</th>\n",
       "      <th>grch38_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>aaref</th>\n",
       "      <th>aaalt</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>Ensembl_transcriptid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>28756127</th>\n",
       "      <td>6</td>\n",
       "      <td>65655758</td>\n",
       "      <td>64945865</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Q</td>\n",
       "      <td>L</td>\n",
       "      <td>0.188</td>\n",
       "      <td>ENST00000503581;ENST00000370621;ENST00000370616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28756128</th>\n",
       "      <td>6</td>\n",
       "      <td>65655758</td>\n",
       "      <td>64945865</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Q</td>\n",
       "      <td>R</td>\n",
       "      <td>0.111</td>\n",
       "      <td>ENST00000503581;ENST00000370621;ENST00000370616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28756129</th>\n",
       "      <td>6</td>\n",
       "      <td>65655758</td>\n",
       "      <td>64945865</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Q</td>\n",
       "      <td>P</td>\n",
       "      <td>0.344</td>\n",
       "      <td>ENST00000503581;ENST00000370621;ENST00000370616</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr  hg19_pos grch38_pos ref alt aaref aaalt  REVEL  \\\n",
       "28756127   6  65655758   64945865   T   A     Q     L  0.188   \n",
       "28756128   6  65655758   64945865   T   C     Q     R  0.111   \n",
       "28756129   6  65655758   64945865   T   G     Q     P  0.344   \n",
       "\n",
       "                                     Ensembl_transcriptid  \n",
       "28756127  ENST00000503581;ENST00000370621;ENST00000370616  \n",
       "28756128  ENST00000503581;ENST00000370621;ENST00000370616  \n",
       "28756129  ENST00000503581;ENST00000370621;ENST00000370616  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from tools import get_revel_scores\n",
    "\n",
    "chromosome = 6\n",
    "position = 65655758\n",
    "\n",
    "results = get_revel_scores(chromosome, position)\n",
    "\n",
    "display(results)"
   ]
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [
    "from api.data.collection import store_database_for_eys_gene\n",
    "store_database_for_eys_gene(\"clinvar\", override=False)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-08-12T16:46:07.973915Z",
     "start_time": "2024-08-12T16:46:07.970874Z"
    }
   },
   "id": "b80a1049abe7596e",
   "execution_count": 12
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The file at ../data/lovd/lovd_data.txt already exists.\n"
     ]
    }
   ],
   "source": [
    "from api.data.collection import store_database_for_eys_gene\n",
    "store_database_for_eys_gene(\"lovd\", override=False)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-08-12T16:47:56.094297Z",
     "start_time": "2024-08-12T16:47:56.090300Z"
    }
   },
   "id": "a1e10fc8175753a0",
   "execution_count": 4
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "data": {
      "text/plain": "              gnomAD ID Chromosome  Position         rsIDs Reference  \\\n0        6-63720525-A-G          6  63720525  rs1768331164         A   \n1        6-63720525-A-T          6  63720525  rs1768331164         A   \n2        6-63720526-T-A          6  63720526          <NA>         T   \n3      6-63720531-C-CAA          6  63720531          <NA>         C   \n4        6-63720531-C-G          6  63720531   rs927390284         C   \n...                 ...        ...       ...           ...       ...   \n11083    6-65495478-G-A          6  65495478   rs530118054         G   \n11084    6-65495479-G-A          6  65495479  rs1766225632         G   \n11085    6-65495482-A-G          6  65495482  rs1766225707         A   \n11086    6-65495484-T-G          6  65495484  rs1766225807         T   \n11087    6-65495485-T-C          6  65495485          <NA>         T   \n\n      Alternate                        Source Filters - exomes  \\\n0             G  gnomAD Exomes,gnomAD Genomes             PASS   \n1             T                gnomAD Genomes             <NA>   \n2             A                 gnomAD Exomes             PASS   \n3           CAA                 gnomAD Exomes             PASS   \n4             G  gnomAD Exomes,gnomAD Genomes             PASS   \n...         ...                           ...              ...   \n11083         A  gnomAD Exomes,gnomAD Genomes             PASS   \n11084         A                 gnomAD Exomes             PASS   \n11085         G  gnomAD Exomes,gnomAD Genomes             PASS   \n11086         G                 gnomAD Exomes             PASS   \n11087         C                 gnomAD Exomes             PASS   \n\n      Filters - genomes         Transcript  ... Homozygote Count Amish  \\\n0                  PASS  ENST00000503581.6  ...                      0   \n1                  PASS  ENST00000503581.6  ...                      0   \n2                  <NA>  ENST00000503581.6  ...                      0   \n3                  <NA>  ENST00000503581.6  ...                      0   \n4                  PASS  ENST00000503581.6  ...                      0   \n...                 ...                ...  ...                    ...   \n11083              PASS  ENST00000503581.6  ...                      0   \n11084              <NA>  ENST00000503581.6  ...                      0   \n11085              PASS  ENST00000503581.6  ...                      0   \n11086              <NA>  ENST00000503581.6  ...                      0   \n11087              <NA>  ENST00000503581.6  ...                      0   \n\n      Hemizygote Count Amish Allele Count South Asian  \\\n0                          0                        0   \n1                          0                        0   \n2                          0                        1   \n3                          0                        0   \n4                          0                        0   \n...                      ...                      ...   \n11083                      0                        4   \n11084                      0                        0   \n11085                      0                        0   \n11086                      0                        5   \n11087                      0                        1   \n\n      Allele Number South Asian Homozygote Count South Asian  \\\n0                         55362                            0   \n1                         55362                            0   \n2                         55360                            0   \n3                         57520                            0   \n4                         57524                            0   \n...                         ...                          ...   \n11083                     88352                            0   \n11084                     88316                            0   \n11085                     87828                            0   \n11086                     87818                            0   \n11087                     87636                            0   \n\n       Hemizygote Count South Asian Allele Count Remaining  \\\n0                                 0                      0   \n1                                 0                      0   \n2                                 0                      0   \n3                                 0                      1   \n4                                 0                      0   \n...                             ...                    ...   \n11083                             0                      8   \n11084                             0                      0   \n11085                             0                      0   \n11086                             0                      0   \n11087                             0                      0   \n\n       Allele Number Remaining  Homozygote Count Remaining  \\\n0                        44082                           0   \n1                        44082                           0   \n2                        44162                           0   \n3                        47700                           0   \n4                        47678                           0   \n...                        ...                         ...   \n11083                    60164                           0   \n11084                    60142                           0   \n11085                    59586                           0   \n11086                    59748                           0   \n11087                    59608                           0   \n\n       Hemizygote Count Remaining  \n0                               0  \n1                               0  \n2                               0  \n3                               0  \n4                               0  \n...                           ...  \n11083                           0  \n11084                           0  \n11085                           0  \n11086                           0  \n11087                           0  \n\n[11088 rows x 72 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>gnomAD ID</th>\n      <th>Chromosome</th>\n      <th>Position</th>\n      <th>rsIDs</th>\n      <th>Reference</th>\n      <th>Alternate</th>\n      <th>Source</th>\n      <th>Filters - exomes</th>\n      <th>Filters - genomes</th>\n      <th>Transcript</th>\n      <th>...</th>\n      <th>Homozygote Count Amish</th>\n      <th>Hemizygote Count Amish</th>\n      <th>Allele Count South Asian</th>\n      <th>Allele Number South Asian</th>\n      <th>Homozygote Count South Asian</th>\n      <th>Hemizygote Count South Asian</th>\n      <th>Allele Count Remaining</th>\n      <th>Allele Number Remaining</th>\n      <th>Homozygote Count Remaining</th>\n      <th>Hemizygote Count Remaining</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>6-63720525-A-G</td>\n      <td>6</td>\n      <td>63720525</td>\n      <td>rs1768331164</td>\n      <td>A</td>\n      <td>G</td>\n      <td>gnomAD Exomes,gnomAD Genomes</td>\n      <td>PASS</td>\n      <td>PASS</td>\n      <td>ENST00000503581.6</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>55362</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>44082</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>6-63720525-A-T</td>\n      <td>6</td>\n      <td>63720525</td>\n      <td>rs1768331164</td>\n      <td>A</td>\n      <td>T</td>\n      <td>gnomAD Genomes</td>\n      <td>&lt;NA&gt;</td>\n      <td>PASS</td>\n      <td>ENST00000503581.6</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>55362</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>44082</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>6-63720526-T-A</td>\n      <td>6</td>\n      <td>63720526</td>\n      <td>&lt;NA&gt;</td>\n      <td>T</td>\n      <td>A</td>\n      <td>gnomAD Exomes</td>\n      <td>PASS</td>\n      <td>&lt;NA&gt;</td>\n      <td>ENST00000503581.6</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>1</td>\n      <td>55360</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>44162</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>6-63720531-C-CAA</td>\n      <td>6</td>\n      <td>63720531</td>\n      <td>&lt;NA&gt;</td>\n      <td>C</td>\n      <td>CAA</td>\n      <td>gnomAD Exomes</td>\n      <td>PASS</td>\n      <td>&lt;NA&gt;</td>\n      <td>ENST00000503581.6</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>57520</td>\n      <td>0</td>\n      <td>0</td>\n      <td>1</td>\n      <td>47700</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>6-63720531-C-G</td>\n      <td>6</td>\n      <td>63720531</td>\n      <td>rs927390284</td>\n      <td>C</td>\n      <td>G</td>\n      <td>gnomAD Exomes,gnomAD Genomes</td>\n      <td>PASS</td>\n      <td>PASS</td>\n      <td>ENST00000503581.6</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>57524</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>47678</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>11083</th>\n      <td>6-65495478-G-A</td>\n      <td>6</td>\n      <td>65495478</td>\n      <td>rs530118054</td>\n      <td>G</td>\n      <td>A</td>\n      <td>gnomAD Exomes,gnomAD Genomes</td>\n      <td>PASS</td>\n      <td>PASS</td>\n      <td>ENST00000503581.6</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>4</td>\n      <td>88352</td>\n      <td>0</td>\n      <td>0</td>\n      <td>8</td>\n      <td>60164</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>11084</th>\n      <td>6-65495479-G-A</td>\n      <td>6</td>\n      <td>65495479</td>\n      <td>rs1766225632</td>\n      <td>G</td>\n      <td>A</td>\n      <td>gnomAD Exomes</td>\n      <td>PASS</td>\n      <td>&lt;NA&gt;</td>\n      <td>ENST00000503581.6</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>88316</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>60142</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>11085</th>\n      <td>6-65495482-A-G</td>\n      <td>6</td>\n      <td>65495482</td>\n      <td>rs1766225707</td>\n      <td>A</td>\n      <td>G</td>\n      <td>gnomAD Exomes,gnomAD Genomes</td>\n      <td>PASS</td>\n      <td>PASS</td>\n      <td>ENST00000503581.6</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>87828</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>59586</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>11086</th>\n      <td>6-65495484-T-G</td>\n      <td>6</td>\n      <td>65495484</td>\n      <td>rs1766225807</td>\n      <td>T</td>\n      <td>G</td>\n      <td>gnomAD Exomes</td>\n      <td>PASS</td>\n      <td>&lt;NA&gt;</td>\n      <td>ENST00000503581.6</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>5</td>\n      <td>87818</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>59748</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>11087</th>\n      <td>6-65495485-T-C</td>\n      <td>6</td>\n      <td>65495485</td>\n      <td>&lt;NA&gt;</td>\n      <td>T</td>\n      <td>C</td>\n      <td>gnomAD Exomes</td>\n      <td>PASS</td>\n      <td>&lt;NA&gt;</td>\n      <td>ENST00000503581.6</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>1</td>\n      <td>87636</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>59608</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n  </tbody>\n</table>\n<p>11088 rows × 72 columns</p>\n</div>"
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from api.data.refactoring import parse_gnomad, set_gnomad_dtypes\n",
    "from api import (store_database_for_eys_gene,GNOMAD_PATH)\n",
    "\n",
    "store_database_for_eys_gene('gnomad', False)\n",
    "\n",
    "gnomad_data = parse_gnomad(GNOMAD_PATH+'/gnomad_data.csv')\n",
    "\n",
    "set_gnomad_dtypes(gnomad_data)\n",
    "\n",
    "gnomad_data"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-08-20T18:43:31.996838Z",
     "start_time": "2024-08-20T18:43:30.905943Z"
    }
   },
   "id": "4ba7fd02a60f5693",
   "execution_count": 1
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The file at ../data/lovd/lovd_data.txt already exists.\n"
     ]
    },
    {
     "data": {
      "text/plain": "id                                                                       59881\nallele                                                                       3\neffectid                                                                    55\nchromosome                                                                   6\nposition_g_start                                                      65655758\nposition_g_end                                                        65655758\ntype                                                                     subst\naverage_frequency                                                     0.001153\nowned_by                                                                   229\nVariantOnGenome/DBID                                                EYS_000001\nVariantOnGenome/DNA                                              g.65655758T>G\nVariantOnGenome/Frequency                         ExAC: 60, 19750, 0, 0.003038\nVariantOnGenome/Reference                        {PMID:Neveling 2012:22334370}\nVariantOnGenome/Restriction_site                                              \nVariantOnGenome/Published_as                                             Q770P\nVariantOnGenome/Remarks                                                       \nVariantOnGenome/Genetic_origin                                        Germline\nVariantOnGenome/Segregation                                                yes\nVariantOnGenome/dbSNP                                                         \nVariantOnGenome/VIP                                                          0\nVariantOnGenome/Methylation                                                   \nVariantOnGenome/ISCN                                                          \nVariantOnGenome/DNA/hg38                                         g.64945865T>G\nVariantOnGenome/ClinVar                                                       \nVariantOnGenome/ClinicalClassification                                     VUS\nVariantOnGenome/ClinicalClassification/Method                                 \nName: 1, dtype: object"
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from api.data.refactoring import merge_gnomad_lovd, parse_gnomad, set_gnomad_dtypes\n",
    "import pandas as pd\n",
    "from api import (store_database_for_eys_gene,\n",
    "                 parse_lovd,\n",
    "                 set_lovd_dtypes,\n",
    "                 LOVD_PATH,\n",
    "                 GNOMAD_PATH)\n",
    "\n",
    "store_database_for_eys_gene('lovd', False)\n",
    "store_database_for_eys_gene('gnomad', False)\n",
    "\n",
    "lovd_data = parse_lovd(LOVD_PATH + \"/lovd_data.txt\")\n",
    "gnomad_data = parse_gnomad(GNOMAD_PATH+'/gnomad_data.csv')\n",
    "\n",
    "set_lovd_dtypes(lovd_data)\n",
    "set_gnomad_dtypes(gnomad_data)\n",
    "\n",
    "variants_on_genome = lovd_data[\"Variants_On_Genome\"].copy()\n",
    "\n",
    "lovd_data = pd.merge(lovd_data[\"Variants_On_Transcripts\"],\n",
    "                       variants_on_genome[['id','VariantOnGenome/DNA/hg38']],\n",
    "                       on='id',\n",
    "                       how='left')\n",
    "\n",
    "gnomad_data = gnomad_data.copy()\n",
    "final_data = merge_gnomad_lovd(lovd_data, gnomad_data)\n",
    "final_data"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-08-22T16:59:49.087139Z",
     "start_time": "2024-08-22T16:59:41.826682Z"
    }
   },
   "id": "dd9b17623f26a07c",
   "execution_count": 5
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'Variants_On_Genome'",
     "output_type": "error",
     "traceback": [
      "\u001B[1;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[1;31mKeyError\u001B[0m                                  Traceback (most recent call last)",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\pandas\\core\\indexes\\base.py:3805\u001B[0m, in \u001B[0;36mIndex.get_loc\u001B[1;34m(self, key)\u001B[0m\n\u001B[0;32m   3804\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m-> 3805\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_engine\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mget_loc\u001B[49m\u001B[43m(\u001B[49m\u001B[43mcasted_key\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m   3806\u001B[0m \u001B[38;5;28;01mexcept\u001B[39;00m \u001B[38;5;167;01mKeyError\u001B[39;00m \u001B[38;5;28;01mas\u001B[39;00m err:\n",
      "File \u001B[1;32mindex.pyx:167\u001B[0m, in \u001B[0;36mpandas._libs.index.IndexEngine.get_loc\u001B[1;34m()\u001B[0m\n",
      "File \u001B[1;32mindex.pyx:196\u001B[0m, in \u001B[0;36mpandas._libs.index.IndexEngine.get_loc\u001B[1;34m()\u001B[0m\n",
      "File \u001B[1;32mpandas\\\\_libs\\\\hashtable_class_helper.pxi:7081\u001B[0m, in \u001B[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001B[1;34m()\u001B[0m\n",
      "File \u001B[1;32mpandas\\\\_libs\\\\hashtable_class_helper.pxi:7089\u001B[0m, in \u001B[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001B[1;34m()\u001B[0m\n",
      "\u001B[1;31mKeyError\u001B[0m: 'Variants_On_Genome'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001B[1;31mKeyError\u001B[0m                                  Traceback (most recent call last)",
      "Cell \u001B[1;32mIn[6], line 5\u001B[0m\n\u001B[0;32m      2\u001B[0m \u001B[38;5;28;01mimport\u001B[39;00m \u001B[38;5;21;01mpandas\u001B[39;00m \u001B[38;5;28;01mas\u001B[39;00m \u001B[38;5;21;01mpd\u001B[39;00m\n\u001B[0;32m      4\u001B[0m row_idx \u001B[38;5;241m=\u001B[39m \u001B[38;5;241m1\u001B[39m\n\u001B[1;32m----> 5\u001B[0m selected_row \u001B[38;5;241m=\u001B[39m \u001B[43mdata\u001B[49m\u001B[43m[\u001B[49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[38;5;124;43mVariants_On_Genome\u001B[39;49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[43m]\u001B[49m\u001B[38;5;241m.\u001B[39miloc[row_idx]\n\u001B[0;32m      7\u001B[0m selected_row_df \u001B[38;5;241m=\u001B[39m pd\u001B[38;5;241m.\u001B[39mDataFrame([selected_row])\n\u001B[0;32m      8\u001B[0m headers \u001B[38;5;241m=\u001B[39m pd\u001B[38;5;241m.\u001B[39mDataFrame(columns\u001B[38;5;241m=\u001B[39mselected_row\u001B[38;5;241m.\u001B[39mindex)\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\pandas\\core\\frame.py:4102\u001B[0m, in \u001B[0;36mDataFrame.__getitem__\u001B[1;34m(self, key)\u001B[0m\n\u001B[0;32m   4100\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mcolumns\u001B[38;5;241m.\u001B[39mnlevels \u001B[38;5;241m>\u001B[39m \u001B[38;5;241m1\u001B[39m:\n\u001B[0;32m   4101\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m_getitem_multilevel(key)\n\u001B[1;32m-> 4102\u001B[0m indexer \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mcolumns\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mget_loc\u001B[49m\u001B[43m(\u001B[49m\u001B[43mkey\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m   4103\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m is_integer(indexer):\n\u001B[0;32m   4104\u001B[0m     indexer \u001B[38;5;241m=\u001B[39m [indexer]\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\pandas\\core\\indexes\\base.py:3812\u001B[0m, in \u001B[0;36mIndex.get_loc\u001B[1;34m(self, key)\u001B[0m\n\u001B[0;32m   3807\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28misinstance\u001B[39m(casted_key, \u001B[38;5;28mslice\u001B[39m) \u001B[38;5;129;01mor\u001B[39;00m (\n\u001B[0;32m   3808\u001B[0m         \u001B[38;5;28misinstance\u001B[39m(casted_key, abc\u001B[38;5;241m.\u001B[39mIterable)\n\u001B[0;32m   3809\u001B[0m         \u001B[38;5;129;01mand\u001B[39;00m \u001B[38;5;28many\u001B[39m(\u001B[38;5;28misinstance\u001B[39m(x, \u001B[38;5;28mslice\u001B[39m) \u001B[38;5;28;01mfor\u001B[39;00m x \u001B[38;5;129;01min\u001B[39;00m casted_key)\n\u001B[0;32m   3810\u001B[0m     ):\n\u001B[0;32m   3811\u001B[0m         \u001B[38;5;28;01mraise\u001B[39;00m InvalidIndexError(key)\n\u001B[1;32m-> 3812\u001B[0m     \u001B[38;5;28;01mraise\u001B[39;00m \u001B[38;5;167;01mKeyError\u001B[39;00m(key) \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01merr\u001B[39;00m\n\u001B[0;32m   3813\u001B[0m \u001B[38;5;28;01mexcept\u001B[39;00m \u001B[38;5;167;01mTypeError\u001B[39;00m:\n\u001B[0;32m   3814\u001B[0m     \u001B[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001B[39;00m\n\u001B[0;32m   3815\u001B[0m     \u001B[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001B[39;00m\n\u001B[0;32m   3816\u001B[0m     \u001B[38;5;66;03m#  the TypeError.\u001B[39;00m\n\u001B[0;32m   3817\u001B[0m     \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m_check_indexing_error(key)\n",
      "\u001B[1;31mKeyError\u001B[0m: 'Variants_On_Genome'"
     ]
    }
   ],
   "source": [
    "from tools.cadd.cadd import fetch_cadd_scores\n",
    "import pandas as pd\n",
    "\n",
    "row_idx = 1\n",
    "selected_row = data[\"Variants_On_Genome\"].iloc[row_idx]\n",
    "\n",
    "selected_row_df = pd.DataFrame([selected_row])\n",
    "headers = pd.DataFrame(columns=selected_row.index)\n",
    "result_df = pd.concat([headers, selected_row_df])\n",
    "display(result_df)\n",
    "\n",
    "chromosome = selected_row[\"chromosome\"]\n",
    "pos_start = selected_row[\"position_g_start\"]\n",
    "pos_end = selected_row[\"position_g_end\"]\n",
    "\n",
    "display(\"Variant scoring:\")\n",
    "score = fetch_cadd_scores(\"GRCh38-v1.7\", chromosome, pos_start, pos_end)\n",
    "score_df = pd.DataFrame(score[1:], columns=score[0])\n",
    "display(score_df)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-08-22T18:10:26.482236Z",
     "start_time": "2024-08-22T18:10:26.178176Z"
    }
   },
   "id": "1a3b6e41853817ca",
   "execution_count": 6
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The file at ../data/lovd/lovd_data.txt already exists.\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001B[1;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[1;31mKeyboardInterrupt\u001B[0m                         Traceback (most recent call last)",
      "Cell \u001B[1;32mIn[2], line 30\u001B[0m\n\u001B[0;32m     28\u001B[0m data \u001B[38;5;241m=\u001B[39m merge_gnomad_lovd(lovd_data, gnomad_data)\n\u001B[0;32m     29\u001B[0m first_100_rows \u001B[38;5;241m=\u001B[39m data\u001B[38;5;241m.\u001B[39mhead(\u001B[38;5;241m100\u001B[39m)\u001B[38;5;241m.\u001B[39mcopy()\n\u001B[1;32m---> 30\u001B[0m result_data \u001B[38;5;241m=\u001B[39m \u001B[43madd_cadd_eval_column\u001B[49m\u001B[43m(\u001B[49m\u001B[43mfirst_100_rows\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m     32\u001B[0m \u001B[38;5;66;03m# # Display the updated DataFrame with the new 'cadd_eval' column\u001B[39;00m\n\u001B[0;32m     33\u001B[0m \u001B[38;5;66;03m# display(result_data)\u001B[39;00m\n\u001B[0;32m     34\u001B[0m \u001B[38;5;66;03m# row_idx=612\u001B[39;00m\n\u001B[1;32m   (...)\u001B[0m\n\u001B[0;32m     55\u001B[0m \u001B[38;5;66;03m# display(highest_score_row)\u001B[39;00m\n\u001B[0;32m     56\u001B[0m \u001B[38;5;66;03m# result_df\u001B[39;00m\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\tests\\tools\\cadd\\cadd.py:100\u001B[0m, in \u001B[0;36madd_cadd_eval_column\u001B[1;34m(data, cadd_version)\u001B[0m\n\u001B[0;32m     92\u001B[0m \u001B[38;5;28;01mdef\u001B[39;00m \u001B[38;5;21madd_cadd_eval_column\u001B[39m(data, cadd_version\u001B[38;5;241m=\u001B[39m\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mGRCh38-v1.7\u001B[39m\u001B[38;5;124m\"\u001B[39m):\n\u001B[0;32m     93\u001B[0m \u001B[38;5;250m    \u001B[39m\u001B[38;5;124;03m\"\"\"\u001B[39;00m\n\u001B[0;32m     94\u001B[0m \u001B[38;5;124;03m    Adds a column 'cadd_eval' to the DataFrame based on CADD score evaluations for each row.\u001B[39;00m\n\u001B[0;32m     95\u001B[0m \n\u001B[1;32m   (...)\u001B[0m\n\u001B[0;32m     98\u001B[0m \u001B[38;5;124;03m    :return: The updated DataFrame with the 'cadd_eval' column.\u001B[39;00m\n\u001B[0;32m     99\u001B[0m \u001B[38;5;124;03m    \"\"\"\u001B[39;00m\n\u001B[1;32m--> 100\u001B[0m     data[\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mcadd_eval\u001B[39m\u001B[38;5;124m\"\u001B[39m] \u001B[38;5;241m=\u001B[39m \u001B[43mdata\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mapply\u001B[49m\u001B[43m(\u001B[49m\u001B[43mevaluate_cadd_score\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43maxis\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;241;43m1\u001B[39;49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mcadd_version\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mcadd_version\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    101\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m data\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\pandas\\core\\frame.py:10374\u001B[0m, in \u001B[0;36mDataFrame.apply\u001B[1;34m(self, func, axis, raw, result_type, args, by_row, engine, engine_kwargs, **kwargs)\u001B[0m\n\u001B[0;32m  10360\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01mpandas\u001B[39;00m\u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mcore\u001B[39;00m\u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mapply\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m frame_apply\n\u001B[0;32m  10362\u001B[0m op \u001B[38;5;241m=\u001B[39m frame_apply(\n\u001B[0;32m  10363\u001B[0m     \u001B[38;5;28mself\u001B[39m,\n\u001B[0;32m  10364\u001B[0m     func\u001B[38;5;241m=\u001B[39mfunc,\n\u001B[1;32m   (...)\u001B[0m\n\u001B[0;32m  10372\u001B[0m     kwargs\u001B[38;5;241m=\u001B[39mkwargs,\n\u001B[0;32m  10373\u001B[0m )\n\u001B[1;32m> 10374\u001B[0m \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[43mop\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mapply\u001B[49m\u001B[43m(\u001B[49m\u001B[43m)\u001B[49m\u001B[38;5;241m.\u001B[39m__finalize__(\u001B[38;5;28mself\u001B[39m, method\u001B[38;5;241m=\u001B[39m\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mapply\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\pandas\\core\\apply.py:916\u001B[0m, in \u001B[0;36mFrameApply.apply\u001B[1;34m(self)\u001B[0m\n\u001B[0;32m    913\u001B[0m \u001B[38;5;28;01melif\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mraw:\n\u001B[0;32m    914\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mapply_raw(engine\u001B[38;5;241m=\u001B[39m\u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mengine, engine_kwargs\u001B[38;5;241m=\u001B[39m\u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mengine_kwargs)\n\u001B[1;32m--> 916\u001B[0m \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mapply_standard\u001B[49m\u001B[43m(\u001B[49m\u001B[43m)\u001B[49m\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\pandas\\core\\apply.py:1063\u001B[0m, in \u001B[0;36mFrameApply.apply_standard\u001B[1;34m(self)\u001B[0m\n\u001B[0;32m   1061\u001B[0m \u001B[38;5;28;01mdef\u001B[39;00m \u001B[38;5;21mapply_standard\u001B[39m(\u001B[38;5;28mself\u001B[39m):\n\u001B[0;32m   1062\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mengine \u001B[38;5;241m==\u001B[39m \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mpython\u001B[39m\u001B[38;5;124m\"\u001B[39m:\n\u001B[1;32m-> 1063\u001B[0m         results, res_index \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mapply_series_generator\u001B[49m\u001B[43m(\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m   1064\u001B[0m     \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[0;32m   1065\u001B[0m         results, res_index \u001B[38;5;241m=\u001B[39m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mapply_series_numba()\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\pandas\\core\\apply.py:1081\u001B[0m, in \u001B[0;36mFrameApply.apply_series_generator\u001B[1;34m(self)\u001B[0m\n\u001B[0;32m   1078\u001B[0m \u001B[38;5;28;01mwith\u001B[39;00m option_context(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mmode.chained_assignment\u001B[39m\u001B[38;5;124m\"\u001B[39m, \u001B[38;5;28;01mNone\u001B[39;00m):\n\u001B[0;32m   1079\u001B[0m     \u001B[38;5;28;01mfor\u001B[39;00m i, v \u001B[38;5;129;01min\u001B[39;00m \u001B[38;5;28menumerate\u001B[39m(series_gen):\n\u001B[0;32m   1080\u001B[0m         \u001B[38;5;66;03m# ignore SettingWithCopy here in case the user mutates\u001B[39;00m\n\u001B[1;32m-> 1081\u001B[0m         results[i] \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mfunc\u001B[49m\u001B[43m(\u001B[49m\u001B[43mv\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43margs\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mkwargs\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m   1082\u001B[0m         \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28misinstance\u001B[39m(results[i], ABCSeries):\n\u001B[0;32m   1083\u001B[0m             \u001B[38;5;66;03m# If we have a view on v, we need to make a copy because\u001B[39;00m\n\u001B[0;32m   1084\u001B[0m             \u001B[38;5;66;03m#  series_generator will swap out the underlying data\u001B[39;00m\n\u001B[0;32m   1085\u001B[0m             results[i] \u001B[38;5;241m=\u001B[39m results[i]\u001B[38;5;241m.\u001B[39mcopy(deep\u001B[38;5;241m=\u001B[39m\u001B[38;5;28;01mFalse\u001B[39;00m)\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\tests\\tools\\cadd\\cadd.py:68\u001B[0m, in \u001B[0;36mevaluate_cadd_score\u001B[1;34m(row, cadd_version)\u001B[0m\n\u001B[0;32m     65\u001B[0m     pos_end \u001B[38;5;241m=\u001B[39m row[\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mPosition_gnomad\u001B[39m\u001B[38;5;124m\"\u001B[39m]\n\u001B[0;32m     67\u001B[0m \u001B[38;5;66;03m# Fetch CADD scores using the fetch_cadd_scores function\u001B[39;00m\n\u001B[1;32m---> 68\u001B[0m score \u001B[38;5;241m=\u001B[39m \u001B[43mfetch_cadd_scores\u001B[49m\u001B[43m(\u001B[49m\u001B[43mcadd_version\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mchromosome\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mpos_start\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mpos_end\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m     70\u001B[0m \u001B[38;5;66;03m# Check if the score is None or not a valid list\u001B[39;00m\n\u001B[0;32m     71\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m score \u001B[38;5;129;01mis\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m \u001B[38;5;129;01mor\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m \u001B[38;5;28misinstance\u001B[39m(score, \u001B[38;5;28mlist\u001B[39m) \u001B[38;5;129;01mor\u001B[39;00m \u001B[38;5;28mlen\u001B[39m(score) \u001B[38;5;241m<\u001B[39m \u001B[38;5;241m2\u001B[39m:\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\tests\\tools\\cadd\\cadd.py:35\u001B[0m, in \u001B[0;36mfetch_cadd_scores\u001B[1;34m(cadd_version, chromosome, start, end)\u001B[0m\n\u001B[0;32m     32\u001B[0m     url \u001B[38;5;241m=\u001B[39m \u001B[38;5;124mf\u001B[39m\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mhttps://cadd.gs.washington.edu/api/v1.0/\u001B[39m\u001B[38;5;132;01m{\u001B[39;00mcadd_version\u001B[38;5;132;01m}\u001B[39;00m\u001B[38;5;124m/\u001B[39m\u001B[38;5;132;01m{\u001B[39;00mchromosome\u001B[38;5;132;01m}\u001B[39;00m\u001B[38;5;124m:\u001B[39m\u001B[38;5;132;01m{\u001B[39;00mstart\u001B[38;5;132;01m}\u001B[39;00m\u001B[38;5;124m\"\u001B[39m\n\u001B[0;32m     34\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m---> 35\u001B[0m     response \u001B[38;5;241m=\u001B[39m \u001B[43mrequests\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mget\u001B[49m\u001B[43m(\u001B[49m\u001B[43murl\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mtimeout\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;241;43m30\u001B[39;49m\u001B[43m)\u001B[49m\n\u001B[0;32m     36\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m response\u001B[38;5;241m.\u001B[39mstatus_code \u001B[38;5;241m==\u001B[39m \u001B[38;5;241m200\u001B[39m:\n\u001B[0;32m     37\u001B[0m         data \u001B[38;5;241m=\u001B[39m response\u001B[38;5;241m.\u001B[39mjson()\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\requests\\api.py:73\u001B[0m, in \u001B[0;36mget\u001B[1;34m(url, params, **kwargs)\u001B[0m\n\u001B[0;32m     62\u001B[0m \u001B[38;5;28;01mdef\u001B[39;00m \u001B[38;5;21mget\u001B[39m(url, params\u001B[38;5;241m=\u001B[39m\u001B[38;5;28;01mNone\u001B[39;00m, \u001B[38;5;241m*\u001B[39m\u001B[38;5;241m*\u001B[39mkwargs):\n\u001B[0;32m     63\u001B[0m \u001B[38;5;250m    \u001B[39m\u001B[38;5;124mr\u001B[39m\u001B[38;5;124;03m\"\"\"Sends a GET request.\u001B[39;00m\n\u001B[0;32m     64\u001B[0m \n\u001B[0;32m     65\u001B[0m \u001B[38;5;124;03m    :param url: URL for the new :class:`Request` object.\u001B[39;00m\n\u001B[1;32m   (...)\u001B[0m\n\u001B[0;32m     70\u001B[0m \u001B[38;5;124;03m    :rtype: requests.Response\u001B[39;00m\n\u001B[0;32m     71\u001B[0m \u001B[38;5;124;03m    \"\"\"\u001B[39;00m\n\u001B[1;32m---> 73\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[43mrequest\u001B[49m\u001B[43m(\u001B[49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[38;5;124;43mget\u001B[39;49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43murl\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mparams\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mparams\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[43mkwargs\u001B[49m\u001B[43m)\u001B[49m\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\requests\\api.py:59\u001B[0m, in \u001B[0;36mrequest\u001B[1;34m(method, url, **kwargs)\u001B[0m\n\u001B[0;32m     55\u001B[0m \u001B[38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we\u001B[39;00m\n\u001B[0;32m     56\u001B[0m \u001B[38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001B[39;00m\n\u001B[0;32m     57\u001B[0m \u001B[38;5;66;03m# cases, and look like a memory leak in others.\u001B[39;00m\n\u001B[0;32m     58\u001B[0m \u001B[38;5;28;01mwith\u001B[39;00m sessions\u001B[38;5;241m.\u001B[39mSession() \u001B[38;5;28;01mas\u001B[39;00m session:\n\u001B[1;32m---> 59\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[43msession\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mrequest\u001B[49m\u001B[43m(\u001B[49m\u001B[43mmethod\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mmethod\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43murl\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43murl\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[43mkwargs\u001B[49m\u001B[43m)\u001B[49m\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\requests\\sessions.py:589\u001B[0m, in \u001B[0;36mSession.request\u001B[1;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001B[0m\n\u001B[0;32m    584\u001B[0m send_kwargs \u001B[38;5;241m=\u001B[39m {\n\u001B[0;32m    585\u001B[0m     \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mtimeout\u001B[39m\u001B[38;5;124m\"\u001B[39m: timeout,\n\u001B[0;32m    586\u001B[0m     \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mallow_redirects\u001B[39m\u001B[38;5;124m\"\u001B[39m: allow_redirects,\n\u001B[0;32m    587\u001B[0m }\n\u001B[0;32m    588\u001B[0m send_kwargs\u001B[38;5;241m.\u001B[39mupdate(settings)\n\u001B[1;32m--> 589\u001B[0m resp \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43msend\u001B[49m\u001B[43m(\u001B[49m\u001B[43mprep\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[43msend_kwargs\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    591\u001B[0m \u001B[38;5;28;01mreturn\u001B[39;00m resp\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\requests\\sessions.py:703\u001B[0m, in \u001B[0;36mSession.send\u001B[1;34m(self, request, **kwargs)\u001B[0m\n\u001B[0;32m    700\u001B[0m start \u001B[38;5;241m=\u001B[39m preferred_clock()\n\u001B[0;32m    702\u001B[0m \u001B[38;5;66;03m# Send the request\u001B[39;00m\n\u001B[1;32m--> 703\u001B[0m r \u001B[38;5;241m=\u001B[39m \u001B[43madapter\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43msend\u001B[49m\u001B[43m(\u001B[49m\u001B[43mrequest\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[43mkwargs\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    705\u001B[0m \u001B[38;5;66;03m# Total elapsed time of the request (approximately)\u001B[39;00m\n\u001B[0;32m    706\u001B[0m elapsed \u001B[38;5;241m=\u001B[39m preferred_clock() \u001B[38;5;241m-\u001B[39m start\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\requests\\adapters.py:667\u001B[0m, in \u001B[0;36mHTTPAdapter.send\u001B[1;34m(self, request, stream, timeout, verify, cert, proxies)\u001B[0m\n\u001B[0;32m    664\u001B[0m     timeout \u001B[38;5;241m=\u001B[39m TimeoutSauce(connect\u001B[38;5;241m=\u001B[39mtimeout, read\u001B[38;5;241m=\u001B[39mtimeout)\n\u001B[0;32m    666\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m--> 667\u001B[0m     resp \u001B[38;5;241m=\u001B[39m \u001B[43mconn\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43murlopen\u001B[49m\u001B[43m(\u001B[49m\n\u001B[0;32m    668\u001B[0m \u001B[43m        \u001B[49m\u001B[43mmethod\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mrequest\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mmethod\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    669\u001B[0m \u001B[43m        \u001B[49m\u001B[43murl\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43murl\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    670\u001B[0m \u001B[43m        \u001B[49m\u001B[43mbody\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mrequest\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mbody\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    671\u001B[0m \u001B[43m        \u001B[49m\u001B[43mheaders\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mrequest\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mheaders\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    672\u001B[0m \u001B[43m        \u001B[49m\u001B[43mredirect\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43;01mFalse\u001B[39;49;00m\u001B[43m,\u001B[49m\n\u001B[0;32m    673\u001B[0m \u001B[43m        \u001B[49m\u001B[43massert_same_host\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43;01mFalse\u001B[39;49;00m\u001B[43m,\u001B[49m\n\u001B[0;32m    674\u001B[0m \u001B[43m        \u001B[49m\u001B[43mpreload_content\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43;01mFalse\u001B[39;49;00m\u001B[43m,\u001B[49m\n\u001B[0;32m    675\u001B[0m \u001B[43m        \u001B[49m\u001B[43mdecode_content\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43;01mFalse\u001B[39;49;00m\u001B[43m,\u001B[49m\n\u001B[0;32m    676\u001B[0m \u001B[43m        \u001B[49m\u001B[43mretries\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mmax_retries\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    677\u001B[0m \u001B[43m        \u001B[49m\u001B[43mtimeout\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mtimeout\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    678\u001B[0m \u001B[43m        \u001B[49m\u001B[43mchunked\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mchunked\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    679\u001B[0m \u001B[43m    \u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    681\u001B[0m \u001B[38;5;28;01mexcept\u001B[39;00m (ProtocolError, \u001B[38;5;167;01mOSError\u001B[39;00m) \u001B[38;5;28;01mas\u001B[39;00m err:\n\u001B[0;32m    682\u001B[0m     \u001B[38;5;28;01mraise\u001B[39;00m \u001B[38;5;167;01mConnectionError\u001B[39;00m(err, request\u001B[38;5;241m=\u001B[39mrequest)\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\urllib3\\connectionpool.py:789\u001B[0m, in \u001B[0;36mHTTPConnectionPool.urlopen\u001B[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001B[0m\n\u001B[0;32m    786\u001B[0m response_conn \u001B[38;5;241m=\u001B[39m conn \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m release_conn \u001B[38;5;28;01melse\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m\n\u001B[0;32m    788\u001B[0m \u001B[38;5;66;03m# Make the request on the HTTPConnection object\u001B[39;00m\n\u001B[1;32m--> 789\u001B[0m response \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_make_request\u001B[49m\u001B[43m(\u001B[49m\n\u001B[0;32m    790\u001B[0m \u001B[43m    \u001B[49m\u001B[43mconn\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    791\u001B[0m \u001B[43m    \u001B[49m\u001B[43mmethod\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    792\u001B[0m \u001B[43m    \u001B[49m\u001B[43murl\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    793\u001B[0m \u001B[43m    \u001B[49m\u001B[43mtimeout\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mtimeout_obj\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    794\u001B[0m \u001B[43m    \u001B[49m\u001B[43mbody\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mbody\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    795\u001B[0m \u001B[43m    \u001B[49m\u001B[43mheaders\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mheaders\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    796\u001B[0m \u001B[43m    \u001B[49m\u001B[43mchunked\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mchunked\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    797\u001B[0m \u001B[43m    \u001B[49m\u001B[43mretries\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mretries\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    798\u001B[0m \u001B[43m    \u001B[49m\u001B[43mresponse_conn\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mresponse_conn\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    799\u001B[0m \u001B[43m    \u001B[49m\u001B[43mpreload_content\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mpreload_content\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    800\u001B[0m \u001B[43m    \u001B[49m\u001B[43mdecode_content\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mdecode_content\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    801\u001B[0m \u001B[43m    \u001B[49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[38;5;241;43m*\u001B[39;49m\u001B[43mresponse_kw\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    802\u001B[0m \u001B[43m\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    804\u001B[0m \u001B[38;5;66;03m# Everything went great!\u001B[39;00m\n\u001B[0;32m    805\u001B[0m clean_exit \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;01mTrue\u001B[39;00m\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\urllib3\\connectionpool.py:466\u001B[0m, in \u001B[0;36mHTTPConnectionPool._make_request\u001B[1;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001B[0m\n\u001B[0;32m    463\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[0;32m    464\u001B[0m     \u001B[38;5;66;03m# Trigger any extra validation we need to do.\u001B[39;00m\n\u001B[0;32m    465\u001B[0m     \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m--> 466\u001B[0m         \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_validate_conn\u001B[49m\u001B[43m(\u001B[49m\u001B[43mconn\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    467\u001B[0m     \u001B[38;5;28;01mexcept\u001B[39;00m (SocketTimeout, BaseSSLError) \u001B[38;5;28;01mas\u001B[39;00m e:\n\u001B[0;32m    468\u001B[0m         \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m_raise_timeout(err\u001B[38;5;241m=\u001B[39me, url\u001B[38;5;241m=\u001B[39murl, timeout_value\u001B[38;5;241m=\u001B[39mconn\u001B[38;5;241m.\u001B[39mtimeout)\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\urllib3\\connectionpool.py:1095\u001B[0m, in \u001B[0;36mHTTPSConnectionPool._validate_conn\u001B[1;34m(self, conn)\u001B[0m\n\u001B[0;32m   1093\u001B[0m \u001B[38;5;66;03m# Force connect early to allow us to validate the connection.\u001B[39;00m\n\u001B[0;32m   1094\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m conn\u001B[38;5;241m.\u001B[39mis_closed:\n\u001B[1;32m-> 1095\u001B[0m     \u001B[43mconn\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mconnect\u001B[49m\u001B[43m(\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m   1097\u001B[0m \u001B[38;5;66;03m# TODO revise this, see https://github.com/urllib3/urllib3/issues/2791\u001B[39;00m\n\u001B[0;32m   1098\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m conn\u001B[38;5;241m.\u001B[39mis_verified \u001B[38;5;129;01mand\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m conn\u001B[38;5;241m.\u001B[39mproxy_is_verified:\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\urllib3\\connection.py:615\u001B[0m, in \u001B[0;36mHTTPSConnection.connect\u001B[1;34m(self)\u001B[0m\n\u001B[0;32m    613\u001B[0m \u001B[38;5;28;01mdef\u001B[39;00m \u001B[38;5;21mconnect\u001B[39m(\u001B[38;5;28mself\u001B[39m) \u001B[38;5;241m-\u001B[39m\u001B[38;5;241m>\u001B[39m \u001B[38;5;28;01mNone\u001B[39;00m:\n\u001B[0;32m    614\u001B[0m     sock: socket\u001B[38;5;241m.\u001B[39msocket \u001B[38;5;241m|\u001B[39m ssl\u001B[38;5;241m.\u001B[39mSSLSocket\n\u001B[1;32m--> 615\u001B[0m     \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39msock \u001B[38;5;241m=\u001B[39m sock \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_new_conn\u001B[49m\u001B[43m(\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    616\u001B[0m     server_hostname: \u001B[38;5;28mstr\u001B[39m \u001B[38;5;241m=\u001B[39m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mhost\n\u001B[0;32m    617\u001B[0m     tls_in_tls \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;01mFalse\u001B[39;00m\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\urllib3\\connection.py:196\u001B[0m, in \u001B[0;36mHTTPConnection._new_conn\u001B[1;34m(self)\u001B[0m\n\u001B[0;32m    191\u001B[0m \u001B[38;5;250m\u001B[39m\u001B[38;5;124;03m\"\"\"Establish a socket connection and set nodelay settings on it.\u001B[39;00m\n\u001B[0;32m    192\u001B[0m \n\u001B[0;32m    193\u001B[0m \u001B[38;5;124;03m:return: New socket connection.\u001B[39;00m\n\u001B[0;32m    194\u001B[0m \u001B[38;5;124;03m\"\"\"\u001B[39;00m\n\u001B[0;32m    195\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m--> 196\u001B[0m     sock \u001B[38;5;241m=\u001B[39m \u001B[43mconnection\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mcreate_connection\u001B[49m\u001B[43m(\u001B[49m\n\u001B[0;32m    197\u001B[0m \u001B[43m        \u001B[49m\u001B[43m(\u001B[49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_dns_host\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mport\u001B[49m\u001B[43m)\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    198\u001B[0m \u001B[43m        \u001B[49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mtimeout\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    199\u001B[0m \u001B[43m        \u001B[49m\u001B[43msource_address\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43msource_address\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    200\u001B[0m \u001B[43m        \u001B[49m\u001B[43msocket_options\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43msocket_options\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m    201\u001B[0m \u001B[43m    \u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    202\u001B[0m \u001B[38;5;28;01mexcept\u001B[39;00m socket\u001B[38;5;241m.\u001B[39mgaierror \u001B[38;5;28;01mas\u001B[39;00m e:\n\u001B[0;32m    203\u001B[0m     \u001B[38;5;28;01mraise\u001B[39;00m NameResolutionError(\u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mhost, \u001B[38;5;28mself\u001B[39m, e) \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01me\u001B[39;00m\n",
      "File \u001B[1;32m~\\PycharmProjects\\kath\\.venv\\Lib\\site-packages\\urllib3\\util\\connection.py:73\u001B[0m, in \u001B[0;36mcreate_connection\u001B[1;34m(address, timeout, source_address, socket_options)\u001B[0m\n\u001B[0;32m     71\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m source_address:\n\u001B[0;32m     72\u001B[0m     sock\u001B[38;5;241m.\u001B[39mbind(source_address)\n\u001B[1;32m---> 73\u001B[0m \u001B[43msock\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mconnect\u001B[49m\u001B[43m(\u001B[49m\u001B[43msa\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m     74\u001B[0m \u001B[38;5;66;03m# Break explicitly a reference cycle\u001B[39;00m\n\u001B[0;32m     75\u001B[0m err \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;01mNone\u001B[39;00m\n",
      "\u001B[1;31mKeyboardInterrupt\u001B[0m: "
     ]
    }
   ],
   "source": [
    "from tests.tools.cadd.cadd import add_cadd_eval_column\n",
    "from tools.cadd.cadd import fetch_cadd_scores\n",
    "from api.data.refactoring import merge_gnomad_lovd, parse_gnomad, set_gnomad_dtypes\n",
    "from api import (store_database_for_eys_gene,\n",
    "                 parse_lovd,\n",
    "                 set_lovd_dtypes,\n",
    "                 LOVD_PATH,\n",
    "                 GNOMAD_PATH)\n",
    "import pandas as pd\n",
    "\n",
    "store_database_for_eys_gene('lovd', False)\n",
    "store_database_for_eys_gene('gnomad', False)\n",
    "\n",
    "lovd_data = parse_lovd(LOVD_PATH + \"/lovd_data.txt\")\n",
    "gnomad_data = parse_gnomad(GNOMAD_PATH+'/gnomad_data.csv')\n",
    "\n",
    "set_lovd_dtypes(lovd_data)\n",
    "set_gnomad_dtypes(gnomad_data)\n",
    "\n",
    "variants_on_genome = lovd_data[\"Variants_On_Genome\"].copy()\n",
    "\n",
    "lovd_data = pd.merge(lovd_data[\"Variants_On_Transcripts\"],\n",
    "                       variants_on_genome[['id','VariantOnGenome/DNA','VariantOnGenome/DNA/hg38','chromosome','position_g_start','position_g_end']],\n",
    "                       on='id',\n",
    "                       how='left')\n",
    "\n",
    "gnomad_data = gnomad_data.copy()\n",
    "data = merge_gnomad_lovd(lovd_data, gnomad_data)\n",
    "first_100_rows = data.head(100).copy()\n",
    "result_data = add_cadd_eval_column(first_100_rows)\n",
    "\n",
    "# # Display the updated DataFrame with the new 'cadd_eval' column\n",
    "# display(result_data)\n",
    "# row_idx=612\n",
    "# selected_row = data.iloc[row_idx]\n",
    "# selected_row_df = pd.DataFrame([selected_row])\n",
    "# headers = pd.DataFrame(columns=selected_row.index)\n",
    "# result_df = pd.concat([headers, selected_row_df])\n",
    "# display(result_df)\n",
    "# \n",
    "# chromosome = selected_row[\"chromosome\"]\n",
    "# pos_start = selected_row[\"position_g_start\"]\n",
    "# pos_end = selected_row[\"position_g_end\"]\n",
    "# if pd.isna(chromosome) and pd.isna(pos_start):\n",
    "#     chromosome = selected_row[\"Chromosome_gnomad\"]\n",
    "#     pos_start = selected_row[\"Position_gnomad\"]\n",
    "#     pos_end = selected_row[\"Position_gnomad\"]\n",
    "# \n",
    "# display(\"Variant scoring:\")\n",
    "# score = fetch_cadd_scores(\"GRCh38-v1.7\", chromosome, pos_start, pos_end)\n",
    "# score_df = pd.DataFrame(score[1:], columns=score[0])\n",
    "# display(score_df)\n",
    "# sorted_df = score_df.sort_values(by=\"PHRED\", ascending=False)\n",
    "# highest_score_row = sorted_df.iloc[0]\n",
    "# display(highest_score_row)\n",
    "# result_df"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-08-28T17:06:10.368404Z",
     "start_time": "2024-08-28T17:05:47.727332Z"
    }
   },
   "id": "8b7a39aad0c2e944",
   "execution_count": 2
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false
   },
   "id": "e8c070e2a54b120"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
