{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "initial_id",
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from data_collection import (store_database_for_eys_gene,\n",
    "                             parse_lovd,\n",
    "                             LOVD_PATH,\n",
    "                             set_lovd_dtypes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f49f7691a27aa7b4",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The file at ../data/lovd/lovd_data.txt already exists.\n"
     ]
    }
   ],
   "source": [
    "store_database_for_eys_gene(\"lovd\", override=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "cf5c45c0f7b9de0f",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chrom_band</th>\n",
       "      <th>imprinting</th>\n",
       "      <th>refseq_genomic</th>\n",
       "      <th>refseq_UD</th>\n",
       "      <th>reference</th>\n",
       "      <th>url_homepage</th>\n",
       "      <th>url_external</th>\n",
       "      <th>...</th>\n",
       "      <th>header</th>\n",
       "      <th>header_align</th>\n",
       "      <th>footer</th>\n",
       "      <th>footer_align</th>\n",
       "      <th>created_by</th>\n",
       "      <th>created_date</th>\n",
       "      <th>edited_by</th>\n",
       "      <th>edited_date</th>\n",
       "      <th>updated_by</th>\n",
       "      <th>updated_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EYS</td>\n",
       "      <td>eyes shut homolog (Drosophila)</td>\n",
       "      <td>6</td>\n",
       "      <td>q12</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NG_023443.2</td>\n",
       "      <td>UD_132085377375</td>\n",
       "      <td></td>\n",
       "      <td>http://www.LOVD.nl/EYS</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>&lt;font color=\\\"#FF0000\\\"&gt;This database is one o...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>00001</td>\n",
       "      <td>2012-02-13 00:00:00</td>\n",
       "      <td>00006</td>\n",
       "      <td>2023-08-30 13:08:19</td>\n",
       "      <td>00000</td>\n",
       "      <td>2024-04-19 20:27:30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    id                            name chromosome chrom_band imprinting   \n",
       "0  EYS  eyes shut homolog (Drosophila)          6        q12    unknown  \\\n",
       "\n",
       "  refseq_genomic        refseq_UD reference            url_homepage   \n",
       "0    NG_023443.2  UD_132085377375            http://www.LOVD.nl/EYS  \\\n",
       "\n",
       "  url_external  ...                                             header   \n",
       "0               ...  <font color=\\\"#FF0000\\\">This database is one o...  \\\n",
       "\n",
       "  header_align footer footer_align created_by         created_date edited_by   \n",
       "0           -1                  -1      00001  2012-02-13 00:00:00     00006  \\\n",
       "\n",
       "           edited_date updated_by         updated_date  \n",
       "0  2023-08-30 13:08:19      00000  2024-04-19 20:27:30  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transcripts\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>geneid</th>\n",
       "      <th>name</th>\n",
       "      <th>id_mutalyzer</th>\n",
       "      <th>id_ncbi</th>\n",
       "      <th>id_ensembl</th>\n",
       "      <th>id_protein_ncbi</th>\n",
       "      <th>id_protein_ensembl</th>\n",
       "      <th>id_protein_uniprot</th>\n",
       "      <th>remarks</th>\n",
       "      <th>position_c_mrna_start</th>\n",
       "      <th>position_c_mrna_end</th>\n",
       "      <th>position_c_cds_end</th>\n",
       "      <th>position_g_mrna_start</th>\n",
       "      <th>position_g_mrna_end</th>\n",
       "      <th>created_by</th>\n",
       "      <th>created_date</th>\n",
       "      <th>edited_by</th>\n",
       "      <th>edited_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00007329</td>\n",
       "      <td>EYS</td>\n",
       "      <td>transcript variant 1</td>\n",
       "      <td>001</td>\n",
       "      <td>NM_001142800.1</td>\n",
       "      <td></td>\n",
       "      <td>NP_001136272.1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-538</td>\n",
       "      <td>10051</td>\n",
       "      <td>9435</td>\n",
       "      <td>66417118</td>\n",
       "      <td>64429876</td>\n",
       "      <td></td>\n",
       "      <td>0000-00-00 00:00:00</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         id geneid                  name id_mutalyzer         id_ncbi   \n",
       "0  00007329    EYS  transcript variant 1          001  NM_001142800.1  \\\n",
       "\n",
       "  id_ensembl id_protein_ncbi id_protein_ensembl id_protein_uniprot remarks   \n",
       "0             NP_001136272.1                                                \\\n",
       "\n",
       "  position_c_mrna_start position_c_mrna_end position_c_cds_end   \n",
       "0                  -538               10051               9435  \\\n",
       "\n",
       "  position_g_mrna_start position_g_mrna_end created_by         created_date   \n",
       "0              66417118            64429876             0000-00-00 00:00:00  \\\n",
       "\n",
       "  edited_by edited_date  \n",
       "0                        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Diseases\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>symbol</th>\n",
       "      <th>name</th>\n",
       "      <th>inheritance</th>\n",
       "      <th>id_omim</th>\n",
       "      <th>tissues</th>\n",
       "      <th>features</th>\n",
       "      <th>remarks</th>\n",
       "      <th>created_by</th>\n",
       "      <th>created_date</th>\n",
       "      <th>edited_by</th>\n",
       "      <th>edited_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00012</td>\n",
       "      <td>PSORS</td>\n",
       "      <td>psoriasis, pustular, generalized (PSORS)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2012-07-06 21:50:32</td>\n",
       "      <td>00006</td>\n",
       "      <td>2019-08-12 13:38:21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>00058</td>\n",
       "      <td>CORD</td>\n",
       "      <td>dystrophy, cone-rod (CORD)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2012-09-22 11:31:25</td>\n",
       "      <td>00006</td>\n",
       "      <td>2020-08-30 09:43:59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00112</td>\n",
       "      <td>RP</td>\n",
       "      <td>retinitis pigmentosa (RP)</td>\n",
       "      <td></td>\n",
       "      <td>268000</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00001</td>\n",
       "      <td>2013-02-21 17:12:36</td>\n",
       "      <td>00006</td>\n",
       "      <td>2021-01-18 09:53:26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>00139</td>\n",
       "      <td>ID</td>\n",
       "      <td>intellectual disability (ID)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00084</td>\n",
       "      <td>2013-06-04 18:18:07</td>\n",
       "      <td>00006</td>\n",
       "      <td>2015-02-09 10:02:49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>00173</td>\n",
       "      <td>SLOS</td>\n",
       "      <td>Smith-Lemli-Opitz syndrome (SLOS)</td>\n",
       "      <td>AR</td>\n",
       "      <td>270400</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2013-08-01 11:16:14</td>\n",
       "      <td>00006</td>\n",
       "      <td>2021-12-10 21:51:32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>00198</td>\n",
       "      <td>?</td>\n",
       "      <td>unclassified / mixed</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2013-09-13 14:21:47</td>\n",
       "      <td>00006</td>\n",
       "      <td>2016-10-22 17:54:40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>02156</td>\n",
       "      <td>-</td>\n",
       "      <td>retinitis pigmentosa, X-linked, and sinorespir...</td>\n",
       "      <td></td>\n",
       "      <td>300455</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2014-09-25 23:29:40</td>\n",
       "      <td>00006</td>\n",
       "      <td>2021-12-10 21:51:32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>02440</td>\n",
       "      <td>RP25</td>\n",
       "      <td>retinitis pigmentosa, type 25 (RP25)</td>\n",
       "      <td>AR</td>\n",
       "      <td>602772</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2014-09-25 23:29:40</td>\n",
       "      <td>00006</td>\n",
       "      <td>2021-12-10 21:51:32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>04211</td>\n",
       "      <td>RPar</td>\n",
       "      <td>retinitis pigmentosa, autosomal recessive (RPar)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2015-02-27 18:58:57</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>04214</td>\n",
       "      <td>-</td>\n",
       "      <td>retinal disease</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2015-02-27 19:48:07</td>\n",
       "      <td>00001</td>\n",
       "      <td>2023-03-09 14:26:26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>04249</td>\n",
       "      <td>macular dystrophy</td>\n",
       "      <td>dystrophy, macular</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2015-05-04 22:10:58</td>\n",
       "      <td>00006</td>\n",
       "      <td>2024-02-15 21:18:39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>05086</td>\n",
       "      <td>HL</td>\n",
       "      <td>hearing loss (HL)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2015-10-23 11:41:05</td>\n",
       "      <td>00006</td>\n",
       "      <td>2015-10-23 11:43:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>05415</td>\n",
       "      <td>USH</td>\n",
       "      <td>Usher syndrome (USH)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2018-04-02 16:40:44</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>05468</td>\n",
       "      <td>uveitis</td>\n",
       "      <td>uveitis</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2018-08-22 09:47:04</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>06906</td>\n",
       "      <td>DEE</td>\n",
       "      <td>encephalopathy, developmental and epileptic</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>2022-04-07 09:24:23</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       id             symbol   \n",
       "0   00012              PSORS  \\\n",
       "1   00058               CORD   \n",
       "2   00112                 RP   \n",
       "3   00139                 ID   \n",
       "4   00173               SLOS   \n",
       "5   00198                  ?   \n",
       "6   02156                  -   \n",
       "7   02440               RP25   \n",
       "8   04211               RPar   \n",
       "9   04214                  -   \n",
       "10  04249  macular dystrophy   \n",
       "11  05086                 HL   \n",
       "12  05415                USH   \n",
       "13  05468            uveitis   \n",
       "14  06906                DEE   \n",
       "\n",
       "                                                 name inheritance id_omim   \n",
       "0            psoriasis, pustular, generalized (PSORS)                      \\\n",
       "1                          dystrophy, cone-rod (CORD)                       \n",
       "2                           retinitis pigmentosa (RP)              268000   \n",
       "3                        intellectual disability (ID)                       \n",
       "4                   Smith-Lemli-Opitz syndrome (SLOS)          AR  270400   \n",
       "5                                unclassified / mixed                       \n",
       "6   retinitis pigmentosa, X-linked, and sinorespir...              300455   \n",
       "7                retinitis pigmentosa, type 25 (RP25)          AR  602772   \n",
       "8    retinitis pigmentosa, autosomal recessive (RPar)                       \n",
       "9                                     retinal disease                       \n",
       "10                                 dystrophy, macular                       \n",
       "11                                  hearing loss (HL)                       \n",
       "12                               Usher syndrome (USH)                       \n",
       "13                                            uveitis                       \n",
       "14        encephalopathy, developmental and epileptic                       \n",
       "\n",
       "   tissues features remarks created_by         created_date edited_by   \n",
       "0                                00006  2012-07-06 21:50:32     00006  \\\n",
       "1                                00006  2012-09-22 11:31:25     00006   \n",
       "2                                00001  2013-02-21 17:12:36     00006   \n",
       "3                                00084  2013-06-04 18:18:07     00006   \n",
       "4                                00006  2013-08-01 11:16:14     00006   \n",
       "5                                00006  2013-09-13 14:21:47     00006   \n",
       "6                                00006  2014-09-25 23:29:40     00006   \n",
       "7                                00006  2014-09-25 23:29:40     00006   \n",
       "8                                00006  2015-02-27 18:58:57             \n",
       "9                                00006  2015-02-27 19:48:07     00001   \n",
       "10                               00006  2015-05-04 22:10:58     00006   \n",
       "11                               00006  2015-10-23 11:41:05     00006   \n",
       "12                               00006  2018-04-02 16:40:44             \n",
       "13                               00006  2018-08-22 09:47:04             \n",
       "14                               00006  2022-04-07 09:24:23             \n",
       "\n",
       "            edited_date  \n",
       "0   2019-08-12 13:38:21  \n",
       "1   2020-08-30 09:43:59  \n",
       "2   2021-01-18 09:53:26  \n",
       "3   2015-02-09 10:02:49  \n",
       "4   2021-12-10 21:51:32  \n",
       "5   2016-10-22 17:54:40  \n",
       "6   2021-12-10 21:51:32  \n",
       "7   2021-12-10 21:51:32  \n",
       "8                        \n",
       "9   2023-03-09 14:26:26  \n",
       "10  2024-02-15 21:18:39  \n",
       "11  2015-10-23 11:43:00  \n",
       "12                       \n",
       "13                       \n",
       "14                       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genes_To_Diseases\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneid</th>\n",
       "      <th>diseaseid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EYS</td>\n",
       "      <td>00112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EYS</td>\n",
       "      <td>02440</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  geneid diseaseid\n",
       "0    EYS     00112\n",
       "1    EYS     02440"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>fatherid</th>\n",
       "      <th>motherid</th>\n",
       "      <th>panelid</th>\n",
       "      <th>panel_size</th>\n",
       "      <th>license</th>\n",
       "      <th>owned_by</th>\n",
       "      <th>Individual/Reference</th>\n",
       "      <th>Individual/Remarks</th>\n",
       "      <th>Individual/Gender</th>\n",
       "      <th>Individual/Consanguinity</th>\n",
       "      <th>Individual/Origin/Geographic</th>\n",
       "      <th>Individual/Age_of_death</th>\n",
       "      <th>Individual/VIP</th>\n",
       "      <th>Individual/Data_av</th>\n",
       "      <th>Individual/Treatment</th>\n",
       "      <th>Individual/Origin/Population</th>\n",
       "      <th>Individual/Individual_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00000135</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>{PMID:Marrakchi 2011:21848462}</td>\n",
       "      <td>5-generation family, 3 affecteds (M)</td>\n",
       "      <td>M</td>\n",
       "      <td>yes</td>\n",
       "      <td>Tunisia</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>00000210</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>00039</td>\n",
       "      <td>{PMID:Abu-Safieh-2013:23105016}</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>(Saudi Arabia)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00001962</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>00025</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>M</td>\n",
       "      <td>?</td>\n",
       "      <td>Germany</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>white</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>00016605</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>00552</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>00033096</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>00229</td>\n",
       "      <td>{PMID:Neveling 2012:22334370}</td>\n",
       "      <td></td>\n",
       "      <td>M</td>\n",
       "      <td>no</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1445</th>\n",
       "      <td>00447702</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>{PMID:Weisschuh 2024:37734845}</td>\n",
       "      <td>patient, no family history</td>\n",
       "      <td>F</td>\n",
       "      <td></td>\n",
       "      <td>Germany</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SRP-1105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1446</th>\n",
       "      <td>00447707</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>{PMID:Weisschuh 2024:37734845}</td>\n",
       "      <td>patient, no family history</td>\n",
       "      <td>M</td>\n",
       "      <td></td>\n",
       "      <td>Germany</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SRP-1167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1447</th>\n",
       "      <td>00447716</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>{PMID:Weisschuh 2024:37734845}</td>\n",
       "      <td>patient, no family history</td>\n",
       "      <td>F</td>\n",
       "      <td></td>\n",
       "      <td>Germany</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SRP-1249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1448</th>\n",
       "      <td>00447718</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>{PMID:Weisschuh 2024:37734845}</td>\n",
       "      <td>patient, no family history</td>\n",
       "      <td>M</td>\n",
       "      <td></td>\n",
       "      <td>Germany</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SRP-1274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1449</th>\n",
       "      <td>00447720</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>00006</td>\n",
       "      <td>{PMID:Weisschuh 2024:37734845}</td>\n",
       "      <td>patient</td>\n",
       "      <td>M</td>\n",
       "      <td></td>\n",
       "      <td>Germany</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SRP-1299</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1450 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            id fatherid motherid panelid panel_size license owned_by   \n",
       "0     00000135                                    3            00006  \\\n",
       "1     00000210                                    1            00039   \n",
       "2     00001962                                    1            00025   \n",
       "3     00016605                                    1            00552   \n",
       "4     00033096                                    1            00229   \n",
       "...        ...      ...      ...     ...        ...     ...      ...   \n",
       "1445  00447702                                    1            00006   \n",
       "1446  00447707                                    1            00006   \n",
       "1447  00447716                                    1            00006   \n",
       "1448  00447718                                    1            00006   \n",
       "1449  00447720                                    1            00006   \n",
       "\n",
       "                 Individual/Reference                    Individual/Remarks   \n",
       "0      {PMID:Marrakchi 2011:21848462}  5-generation family, 3 affecteds (M)  \\\n",
       "1     {PMID:Abu-Safieh-2013:23105016}                                         \n",
       "2                                                                             \n",
       "3                                                                             \n",
       "4       {PMID:Neveling 2012:22334370}                                         \n",
       "...                               ...                                   ...   \n",
       "1445   {PMID:Weisschuh 2024:37734845}            patient, no family history   \n",
       "1446   {PMID:Weisschuh 2024:37734845}            patient, no family history   \n",
       "1447   {PMID:Weisschuh 2024:37734845}            patient, no family history   \n",
       "1448   {PMID:Weisschuh 2024:37734845}            patient, no family history   \n",
       "1449   {PMID:Weisschuh 2024:37734845}                               patient   \n",
       "\n",
       "     Individual/Gender Individual/Consanguinity Individual/Origin/Geographic   \n",
       "0                    M                      yes                      Tunisia  \\\n",
       "1                                                             (Saudi Arabia)   \n",
       "2                    M                        ?                      Germany   \n",
       "3                                                                              \n",
       "4                    M                       no                                \n",
       "...                ...                      ...                          ...   \n",
       "1445                 F                                               Germany   \n",
       "1446                 M                                               Germany   \n",
       "1447                 F                                               Germany   \n",
       "1448                 M                                               Germany   \n",
       "1449                 M                                               Germany   \n",
       "\n",
       "     Individual/Age_of_death Individual/VIP Individual/Data_av   \n",
       "0                                                               \\\n",
       "1                                                                \n",
       "2                                                                \n",
       "3                                         0                      \n",
       "4                                         0                      \n",
       "...                      ...            ...                ...   \n",
       "1445                                      0                      \n",
       "1446                                      0                      \n",
       "1447                                      0                      \n",
       "1448                                      0                      \n",
       "1449                                      0                      \n",
       "\n",
       "     Individual/Treatment Individual/Origin/Population   \n",
       "0                                                       \\\n",
       "1                                                        \n",
       "2                                                white   \n",
       "3                                                        \n",
       "4                                                        \n",
       "...                   ...                          ...   \n",
       "1445                                                     \n",
       "1446                                                     \n",
       "1447                                                     \n",
       "1448                                                     \n",
       "1449                                                     \n",
       "\n",
       "     Individual/Individual_ID  \n",
       "0                              \n",
       "1                              \n",
       "2                              \n",
       "3                              \n",
       "4                              \n",
       "...                       ...  \n",
       "1445                 SRP-1105  \n",
       "1446                 SRP-1167  \n",
       "1447                 SRP-1249  \n",
       "1448                 SRP-1274  \n",
       "1449                 SRP-1299  \n",
       "\n",
       "[1450 rows x 18 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals_To_Diseases\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualid</th>\n",
       "      <th>diseaseid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>00000135</td>\n",
       "      <td>00012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>00000210</td>\n",
       "      <td>00058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00001962</td>\n",
       "      <td>00173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>00033096</td>\n",
       "      <td>04214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>00033109</td>\n",
       "      <td>04214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1444</th>\n",
       "      <td>00447702</td>\n",
       "      <td>00198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1445</th>\n",
       "      <td>00447707</td>\n",
       "      <td>00198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1446</th>\n",
       "      <td>00447716</td>\n",
       "      <td>00198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1447</th>\n",
       "      <td>00447718</td>\n",
       "      <td>00198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1448</th>\n",
       "      <td>00447720</td>\n",
       "      <td>00198</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1449 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     individualid diseaseid\n",
       "0        00000135     00012\n",
       "1        00000210     00058\n",
       "2        00001962     00173\n",
       "3        00033096     04214\n",
       "4        00033109     04214\n",
       "...           ...       ...\n",
       "1444     00447702     00198\n",
       "1445     00447707     00198\n",
       "1446     00447716     00198\n",
       "1447     00447718     00198\n",
       "1448     00447720     00198\n",
       "\n",
       "[1449 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Phenotypes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>diseaseid</th>\n",
       "      <th>individualid</th>\n",
       "      <th>owned_by</th>\n",
       "      <th>Phenotype/Inheritance</th>\n",
       "      <th>Phenotype/Age</th>\n",
       "      <th>Phenotype/Additional</th>\n",
       "      <th>Phenotype/Biochem_param</th>\n",
       "      <th>Phenotype/Age/Onset</th>\n",
       "      <th>Phenotype/Age/Diagnosis</th>\n",
       "      <th>Phenotype/Severity_score</th>\n",
       "      <th>Phenotype/Onset</th>\n",
       "      <th>Phenotype/Protein</th>\n",
       "      <th>Phenotype/Tumor/MSI</th>\n",
       "      <th>Phenotype/Enzyme/CPK</th>\n",
       "      <th>Phenotype/Heart/Myocardium</th>\n",
       "      <th>Phenotype/Lung</th>\n",
       "      <th>Phenotype/Diagnosis/Definite</th>\n",
       "      <th>Phenotype/Diagnosis/Initial</th>\n",
       "      <th>Phenotype/Diagnosis/Criteria</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0000000008</td>\n",
       "      <td>00012</td>\n",
       "      <td>00000135</td>\n",
       "      <td>00006</td>\n",
       "      <td>Familial, autosomal recessive</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0000000026</td>\n",
       "      <td>00058</td>\n",
       "      <td>00000210</td>\n",
       "      <td>00039</td>\n",
       "      <td>Familial, autosomal recessive</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0000000941</td>\n",
       "      <td>00173</td>\n",
       "      <td>00001962</td>\n",
       "      <td>00025</td>\n",
       "      <td>Familial</td>\n",
       "      <td></td>\n",
       "      <td>2-3 toe syndactyly</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0000026525</td>\n",
       "      <td>04214</td>\n",
       "      <td>00033096</td>\n",
       "      <td>00229</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0000026538</td>\n",
       "      <td>04214</td>\n",
       "      <td>00033109</td>\n",
       "      <td>00229</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1266</th>\n",
       "      <td>0000336901</td>\n",
       "      <td>00198</td>\n",
       "      <td>00447702</td>\n",
       "      <td>00006</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa, simplex</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1267</th>\n",
       "      <td>0000336906</td>\n",
       "      <td>00198</td>\n",
       "      <td>00447707</td>\n",
       "      <td>00006</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa, simplex</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1268</th>\n",
       "      <td>0000336915</td>\n",
       "      <td>00198</td>\n",
       "      <td>00447716</td>\n",
       "      <td>00006</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa, simplex</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269</th>\n",
       "      <td>0000336917</td>\n",
       "      <td>00198</td>\n",
       "      <td>00447718</td>\n",
       "      <td>00006</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa, simplex</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1270</th>\n",
       "      <td>0000336919</td>\n",
       "      <td>00198</td>\n",
       "      <td>00447720</td>\n",
       "      <td>00006</td>\n",
       "      <td>Unknown</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retinitis pigmentosa, simplex</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1271 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              id diseaseid individualid owned_by   \n",
       "0     0000000008     00012     00000135    00006  \\\n",
       "1     0000000026     00058     00000210    00039   \n",
       "2     0000000941     00173     00001962    00025   \n",
       "3     0000026525     04214     00033096    00229   \n",
       "4     0000026538     04214     00033109    00229   \n",
       "...          ...       ...          ...      ...   \n",
       "1266  0000336901     00198     00447702    00006   \n",
       "1267  0000336906     00198     00447707    00006   \n",
       "1268  0000336915     00198     00447716    00006   \n",
       "1269  0000336917     00198     00447718    00006   \n",
       "1270  0000336919     00198     00447720    00006   \n",
       "\n",
       "              Phenotype/Inheritance Phenotype/Age Phenotype/Additional   \n",
       "0     Familial, autosomal recessive                                     \\\n",
       "1     Familial, autosomal recessive                                      \n",
       "2                          Familial                 2-3 toe syndactyly   \n",
       "3                           Unknown                                      \n",
       "4                           Unknown                                      \n",
       "...                             ...           ...                  ...   \n",
       "1266                        Unknown                                      \n",
       "1267                        Unknown                                      \n",
       "1268                        Unknown                                      \n",
       "1269                        Unknown                                      \n",
       "1270                        Unknown                                      \n",
       "\n",
       "     Phenotype/Biochem_param Phenotype/Age/Onset Phenotype/Age/Diagnosis   \n",
       "0                                                                         \\\n",
       "1                                                                          \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...                      ...                 ...                     ...   \n",
       "1266                                                                       \n",
       "1267                                                                       \n",
       "1268                                                                       \n",
       "1269                                                                       \n",
       "1270                                                                       \n",
       "\n",
       "     Phenotype/Severity_score Phenotype/Onset Phenotype/Protein   \n",
       "0                                                                \\\n",
       "1                                                                 \n",
       "2                           5                                     \n",
       "3                                                                 \n",
       "4                                                                 \n",
       "...                       ...             ...               ...   \n",
       "1266                                                              \n",
       "1267                                                              \n",
       "1268                                                              \n",
       "1269                                                              \n",
       "1270                                                              \n",
       "\n",
       "     Phenotype/Tumor/MSI Phenotype/Enzyme/CPK Phenotype/Heart/Myocardium   \n",
       "0                                                                         \\\n",
       "1                                                                          \n",
       "2                                                                          \n",
       "3                                                                          \n",
       "4                                                                          \n",
       "...                  ...                  ...                        ...   \n",
       "1266                                                                       \n",
       "1267                                                                       \n",
       "1268                                                                       \n",
       "1269                                                                       \n",
       "1270                                                                       \n",
       "\n",
       "     Phenotype/Lung Phenotype/Diagnosis/Definite   \n",
       "0                                                 \\\n",
       "1                                                  \n",
       "2                                                  \n",
       "3                                                  \n",
       "4                                                  \n",
       "...             ...                          ...   \n",
       "1266                                               \n",
       "1267                                               \n",
       "1268                                               \n",
       "1269                                               \n",
       "1270                                               \n",
       "\n",
       "        Phenotype/Diagnosis/Initial Phenotype/Diagnosis/Criteria  \n",
       "0                                                                 \n",
       "1                                                                 \n",
       "2                                                                 \n",
       "3              retinitis pigmentosa                               \n",
       "4              retinitis pigmentosa                               \n",
       "...                             ...                          ...  \n",
       "1266  retinitis pigmentosa, simplex                               \n",
       "1267  retinitis pigmentosa, simplex                               \n",
       "1268  retinitis pigmentosa, simplex                               \n",
       "1269  retinitis pigmentosa, simplex                               \n",
       "1270  retinitis pigmentosa, simplex                               \n",
       "\n",
       "[1271 rows x 20 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screenings\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>individualid</th>\n",
       "      <th>variants_found</th>\n",
       "      <th>owned_by</th>\n",
       "      <th>created_by</th>\n",
       "      <th>created_date</th>\n",
       "      <th>edited_by</th>\n",
       "      <th>edited_date</th>\n",
       "      <th>Screening/Technique</th>\n",
       "      <th>Screening/Template</th>\n",
       "      <th>Screening/Tissue</th>\n",
       "      <th>Screening/Remarks</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0000000126</td>\n",
       "      <td>00000135</td>\n",
       "      <td>1</td>\n",
       "      <td>00006</td>\n",
       "      <td>00006</td>\n",
       "      <td>2012-07-07 19:04:19</td>\n",
       "      <td>00006</td>\n",
       "      <td>2012-07-07 19:12:08</td>\n",
       "      <td>RT-PCR;SEQ</td>\n",
       "      <td>DNA;RNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0000000211</td>\n",
       "      <td>00000210</td>\n",
       "      <td>1</td>\n",
       "      <td>00039</td>\n",
       "      <td>00006</td>\n",
       "      <td>2012-09-22 11:36:24</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SEQ</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0000001640</td>\n",
       "      <td>00001962</td>\n",
       "      <td>1</td>\n",
       "      <td>00025</td>\n",
       "      <td>00006</td>\n",
       "      <td>2010-03-11 16:36:41</td>\n",
       "      <td>00025</td>\n",
       "      <td>2012-04-13 15:18:00</td>\n",
       "      <td>SEQ</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0000016557</td>\n",
       "      <td>00016605</td>\n",
       "      <td>1</td>\n",
       "      <td>00552</td>\n",
       "      <td>00552</td>\n",
       "      <td>2014-05-23 13:12:43</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SEQ-NG-I</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0000033164</td>\n",
       "      <td>00033096</td>\n",
       "      <td>1</td>\n",
       "      <td>00229</td>\n",
       "      <td>00229</td>\n",
       "      <td>2012-02-04 15:20:01</td>\n",
       "      <td>00006</td>\n",
       "      <td>2012-05-18 13:59:33</td>\n",
       "      <td>SEQ;SEQ-NG-S</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1445</th>\n",
       "      <td>0000449279</td>\n",
       "      <td>00447702</td>\n",
       "      <td>1</td>\n",
       "      <td>00006</td>\n",
       "      <td>00006</td>\n",
       "      <td>2024-01-26 10:23:59</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SEQ-NG</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td>WGS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1446</th>\n",
       "      <td>0000449284</td>\n",
       "      <td>00447707</td>\n",
       "      <td>1</td>\n",
       "      <td>00006</td>\n",
       "      <td>00006</td>\n",
       "      <td>2024-01-26 10:23:59</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SEQ-NG</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td>WGS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1447</th>\n",
       "      <td>0000449293</td>\n",
       "      <td>00447716</td>\n",
       "      <td>1</td>\n",
       "      <td>00006</td>\n",
       "      <td>00006</td>\n",
       "      <td>2024-01-26 10:23:59</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SEQ-NG</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td>WGS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1448</th>\n",
       "      <td>0000449295</td>\n",
       "      <td>00447718</td>\n",
       "      <td>1</td>\n",
       "      <td>00006</td>\n",
       "      <td>00006</td>\n",
       "      <td>2024-01-26 10:23:59</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SEQ-NG</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td>WGS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1449</th>\n",
       "      <td>0000449297</td>\n",
       "      <td>00447720</td>\n",
       "      <td>1</td>\n",
       "      <td>00006</td>\n",
       "      <td>00006</td>\n",
       "      <td>2024-01-26 10:23:59</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SEQ-NG</td>\n",
       "      <td>DNA</td>\n",
       "      <td></td>\n",
       "      <td>WGS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1450 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              id individualid variants_found owned_by created_by   \n",
       "0     0000000126     00000135              1    00006      00006  \\\n",
       "1     0000000211     00000210              1    00039      00006   \n",
       "2     0000001640     00001962              1    00025      00006   \n",
       "3     0000016557     00016605              1    00552      00552   \n",
       "4     0000033164     00033096              1    00229      00229   \n",
       "...          ...          ...            ...      ...        ...   \n",
       "1445  0000449279     00447702              1    00006      00006   \n",
       "1446  0000449284     00447707              1    00006      00006   \n",
       "1447  0000449293     00447716              1    00006      00006   \n",
       "1448  0000449295     00447718              1    00006      00006   \n",
       "1449  0000449297     00447720              1    00006      00006   \n",
       "\n",
       "             created_date edited_by          edited_date Screening/Technique   \n",
       "0     2012-07-07 19:04:19     00006  2012-07-07 19:12:08          RT-PCR;SEQ  \\\n",
       "1     2012-09-22 11:36:24                                                SEQ   \n",
       "2     2010-03-11 16:36:41     00025  2012-04-13 15:18:00                 SEQ   \n",
       "3     2014-05-23 13:12:43                                           SEQ-NG-I   \n",
       "4     2012-02-04 15:20:01     00006  2012-05-18 13:59:33        SEQ;SEQ-NG-S   \n",
       "...                   ...       ...                  ...                 ...   \n",
       "1445  2024-01-26 10:23:59                                             SEQ-NG   \n",
       "1446  2024-01-26 10:23:59                                             SEQ-NG   \n",
       "1447  2024-01-26 10:23:59                                             SEQ-NG   \n",
       "1448  2024-01-26 10:23:59                                             SEQ-NG   \n",
       "1449  2024-01-26 10:23:59                                             SEQ-NG   \n",
       "\n",
       "     Screening/Template Screening/Tissue Screening/Remarks  \n",
       "0               DNA;RNA                                     \n",
       "1                   DNA                                     \n",
       "2                   DNA                                     \n",
       "3                   DNA                                     \n",
       "4                   DNA                                     \n",
       "...                 ...              ...               ...  \n",
       "1445                DNA                                WGS  \n",
       "1446                DNA                                WGS  \n",
       "1447                DNA                                WGS  \n",
       "1448                DNA                                WGS  \n",
       "1449                DNA                                WGS  \n",
       "\n",
       "[1450 rows x 12 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screenings_To_Genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>screeningid</th>\n",
       "      <th>geneid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0000000126</td>\n",
       "      <td>IL36RN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0000000211</td>\n",
       "      <td>MKS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0000001640</td>\n",
       "      <td>DHCR7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0000033164</td>\n",
       "      <td>AHI1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0000033164</td>\n",
       "      <td>EYS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1311</th>\n",
       "      <td>0000437646</td>\n",
       "      <td>EYS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1312</th>\n",
       "      <td>0000437902</td>\n",
       "      <td>EYS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1313</th>\n",
       "      <td>0000437922</td>\n",
       "      <td>EYS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1314</th>\n",
       "      <td>0000443144</td>\n",
       "      <td>EYS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1315</th>\n",
       "      <td>0000443145</td>\n",
       "      <td>EYS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1316 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     screeningid  geneid\n",
       "0     0000000126  IL36RN\n",
       "1     0000000211    MKS1\n",
       "2     0000001640   DHCR7\n",
       "3     0000033164    AHI1\n",
       "4     0000033164     EYS\n",
       "...          ...     ...\n",
       "1311  0000437646     EYS\n",
       "1312  0000437902     EYS\n",
       "1313  0000437922     EYS\n",
       "1314  0000443144     EYS\n",
       "1315  0000443145     EYS\n",
       "\n",
       "[1316 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants_On_Genome\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>allele</th>\n",
       "      <th>effectid</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position_g_start</th>\n",
       "      <th>position_g_end</th>\n",
       "      <th>type</th>\n",
       "      <th>average_frequency</th>\n",
       "      <th>owned_by</th>\n",
       "      <th>VariantOnGenome/DBID</th>\n",
       "      <th>...</th>\n",
       "      <th>VariantOnGenome/Genetic_origin</th>\n",
       "      <th>VariantOnGenome/Segregation</th>\n",
       "      <th>VariantOnGenome/dbSNP</th>\n",
       "      <th>VariantOnGenome/VIP</th>\n",
       "      <th>VariantOnGenome/Methylation</th>\n",
       "      <th>VariantOnGenome/ISCN</th>\n",
       "      <th>VariantOnGenome/DNA/hg38</th>\n",
       "      <th>VariantOnGenome/ClinVar</th>\n",
       "      <th>VariantOnGenome/ClinicalClassification</th>\n",
       "      <th>VariantOnGenome/ClinicalClassification/Method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0000036426</td>\n",
       "      <td>3</td>\n",
       "      <td>50</td>\n",
       "      <td>6</td>\n",
       "      <td>64498971</td>\n",
       "      <td>64498971</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.000742922</td>\n",
       "      <td>00552</td>\n",
       "      <td>EYS_000007</td>\n",
       "      <td>...</td>\n",
       "      <td>Germline</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>g.63789078A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>VUS</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0000059881</td>\n",
       "      <td>3</td>\n",
       "      <td>55</td>\n",
       "      <td>6</td>\n",
       "      <td>65655758</td>\n",
       "      <td>65655758</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.00115297</td>\n",
       "      <td>00229</td>\n",
       "      <td>EYS_000001</td>\n",
       "      <td>...</td>\n",
       "      <td>Germline</td>\n",
       "      <td>yes</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>g.64945865T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>VUS</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0000059883</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>65336143</td>\n",
       "      <td>65336143</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.224189</td>\n",
       "      <td>00229</td>\n",
       "      <td>EYS_000002</td>\n",
       "      <td>...</td>\n",
       "      <td>Germline</td>\n",
       "      <td>no</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>g.64626250G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0000059884</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>6</td>\n",
       "      <td>65300869</td>\n",
       "      <td>65300869</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.000837928</td>\n",
       "      <td>00229</td>\n",
       "      <td>EYS_000003</td>\n",
       "      <td>...</td>\n",
       "      <td>Germline</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>g.64590976G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0000059885</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>65016998</td>\n",
       "      <td>65016999</td>\n",
       "      <td>del</td>\n",
       "      <td>0</td>\n",
       "      <td>00229</td>\n",
       "      <td>EYS_000004</td>\n",
       "      <td>...</td>\n",
       "      <td>Germline</td>\n",
       "      <td>yes</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>g.64307105_64307106del</td>\n",
       "      <td></td>\n",
       "      <td>benign</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2536</th>\n",
       "      <td>0000964211</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "      <td>65767634</td>\n",
       "      <td>65767634</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.243022</td>\n",
       "      <td>02330</td>\n",
       "      <td>EYS_000248</td>\n",
       "      <td>...</td>\n",
       "      <td>CLASSIFICATION record</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>likely benign</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2537</th>\n",
       "      <td>0000964212</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "      <td>65767643</td>\n",
       "      <td>65767643</td>\n",
       "      <td>del</td>\n",
       "      <td>0</td>\n",
       "      <td>02330</td>\n",
       "      <td>EYS_000926</td>\n",
       "      <td>...</td>\n",
       "      <td>CLASSIFICATION record</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>likely benign</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2538</th>\n",
       "      <td>0000964215</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>6</td>\n",
       "      <td>66005927</td>\n",
       "      <td>66005927</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.000112112</td>\n",
       "      <td>02327</td>\n",
       "      <td>EYS_000253</td>\n",
       "      <td>...</td>\n",
       "      <td>CLASSIFICATION record</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VUS</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2539</th>\n",
       "      <td>0000964216</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>6</td>\n",
       "      <td>66044874</td>\n",
       "      <td>66044874</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.0000818974</td>\n",
       "      <td>02327</td>\n",
       "      <td>EYS_000256</td>\n",
       "      <td>...</td>\n",
       "      <td>CLASSIFICATION record</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VUS</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2540</th>\n",
       "      <td>0000977314</td>\n",
       "      <td>0</td>\n",
       "      <td>90</td>\n",
       "      <td>6</td>\n",
       "      <td>64430943</td>\n",
       "      <td>64430943</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.00000659822</td>\n",
       "      <td>01804</td>\n",
       "      <td>EYS_000060</td>\n",
       "      <td>...</td>\n",
       "      <td>CLASSIFICATION record</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>pathogenic</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2541 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              id allele effectid chromosome position_g_start position_g_end   \n",
       "0     0000036426      3       50          6         64498971       64498971  \\\n",
       "1     0000059881      3       55          6         65655758       65655758   \n",
       "2     0000059883      1       11          6         65336143       65336143   \n",
       "3     0000059884      1       15          6         65300869       65300869   \n",
       "4     0000059885      1       11          6         65016998       65016999   \n",
       "...          ...    ...      ...        ...              ...            ...   \n",
       "2536  0000964211      0       30          6         65767634       65767634   \n",
       "2537  0000964212      0       30          6         65767643       65767643   \n",
       "2538  0000964215      0       50          6         66005927       66005927   \n",
       "2539  0000964216      0       50          6         66044874       66044874   \n",
       "2540  0000977314      0       90          6         64430943       64430943   \n",
       "\n",
       "       type average_frequency owned_by VariantOnGenome/DBID  ...   \n",
       "0     subst       0.000742922    00552           EYS_000007  ...  \\\n",
       "1     subst        0.00115297    00229           EYS_000001  ...   \n",
       "2     subst          0.224189    00229           EYS_000002  ...   \n",
       "3     subst       0.000837928    00229           EYS_000003  ...   \n",
       "4       del                 0    00229           EYS_000004  ...   \n",
       "...     ...               ...      ...                  ...  ...   \n",
       "2536  subst          0.243022    02330           EYS_000248  ...   \n",
       "2537    del                 0    02330           EYS_000926  ...   \n",
       "2538  subst       0.000112112    02327           EYS_000253  ...   \n",
       "2539  subst      0.0000818974    02327           EYS_000256  ...   \n",
       "2540  subst     0.00000659822    01804           EYS_000060  ...   \n",
       "\n",
       "     VariantOnGenome/Genetic_origin VariantOnGenome/Segregation   \n",
       "0                          Germline                              \\\n",
       "1                          Germline                         yes   \n",
       "2                          Germline                          no   \n",
       "3                          Germline                               \n",
       "4                          Germline                         yes   \n",
       "...                             ...                         ...   \n",
       "2536          CLASSIFICATION record                               \n",
       "2537          CLASSIFICATION record                               \n",
       "2538          CLASSIFICATION record                               \n",
       "2539          CLASSIFICATION record                               \n",
       "2540          CLASSIFICATION record                               \n",
       "\n",
       "     VariantOnGenome/dbSNP VariantOnGenome/VIP VariantOnGenome/Methylation   \n",
       "0                                            0                              \\\n",
       "1                                            0                               \n",
       "2                                            0                               \n",
       "3                                            0                               \n",
       "4                                            0                               \n",
       "...                    ...                 ...                         ...   \n",
       "2536                                                                         \n",
       "2537                                                                         \n",
       "2538                                                                         \n",
       "2539                                                                         \n",
       "2540                                                                         \n",
       "\n",
       "     VariantOnGenome/ISCN VariantOnGenome/DNA/hg38 VariantOnGenome/ClinVar   \n",
       "0                                    g.63789078A>G                          \\\n",
       "1                                    g.64945865T>G                           \n",
       "2                                    g.64626250G>A                           \n",
       "3                                    g.64590976G>A                           \n",
       "4                           g.64307105_64307106del                           \n",
       "...                   ...                      ...                     ...   \n",
       "2536                                                                         \n",
       "2537                                                                         \n",
       "2538                                                                         \n",
       "2539                                                                         \n",
       "2540                                                                         \n",
       "\n",
       "     VariantOnGenome/ClinicalClassification   \n",
       "0                                       VUS  \\\n",
       "1                                       VUS   \n",
       "2                                    benign   \n",
       "3                                    benign   \n",
       "4                                    benign   \n",
       "...                                     ...   \n",
       "2536                          likely benign   \n",
       "2537                          likely benign   \n",
       "2538                                    VUS   \n",
       "2539                                    VUS   \n",
       "2540                             pathogenic   \n",
       "\n",
       "     VariantOnGenome/ClinicalClassification/Method  \n",
       "0                                                   \n",
       "1                                                   \n",
       "2                                                   \n",
       "3                                                   \n",
       "4                                                   \n",
       "...                                            ...  \n",
       "2536                                                \n",
       "2537                                                \n",
       "2538                                                \n",
       "2539                                                \n",
       "2540                                                \n",
       "\n",
       "[2541 rows x 26 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants_On_Transcripts\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>transcriptid</th>\n",
       "      <th>effectid</th>\n",
       "      <th>position_c_start</th>\n",
       "      <th>position_c_start_intron</th>\n",
       "      <th>position_c_end</th>\n",
       "      <th>position_c_end_intron</th>\n",
       "      <th>VariantOnTranscript/DNA</th>\n",
       "      <th>VariantOnTranscript/RNA</th>\n",
       "      <th>VariantOnTranscript/Protein</th>\n",
       "      <th>VariantOnTranscript/Exon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0000036426</td>\n",
       "      <td>00007329</td>\n",
       "      <td>50</td>\n",
       "      <td>7558</td>\n",
       "      <td>0</td>\n",
       "      <td>7558</td>\n",
       "      <td>0</td>\n",
       "      <td>c.7558T&gt;C</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Phe2520Leu)</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0000059881</td>\n",
       "      <td>00007329</td>\n",
       "      <td>55</td>\n",
       "      <td>2309</td>\n",
       "      <td>0</td>\n",
       "      <td>2309</td>\n",
       "      <td>0</td>\n",
       "      <td>c.2309A&gt;C</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Gln770Pro)</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0000059883</td>\n",
       "      <td>00007329</td>\n",
       "      <td>11</td>\n",
       "      <td>3444</td>\n",
       "      <td>-5</td>\n",
       "      <td>3444</td>\n",
       "      <td>-5</td>\n",
       "      <td>c.3444-5C&gt;T</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>22i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0000059884</td>\n",
       "      <td>00007329</td>\n",
       "      <td>15</td>\n",
       "      <td>4891</td>\n",
       "      <td>0</td>\n",
       "      <td>4891</td>\n",
       "      <td>0</td>\n",
       "      <td>c.4891C&gt;T</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Pro1631Ser)</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0000059885</td>\n",
       "      <td>00007329</td>\n",
       "      <td>11</td>\n",
       "      <td>6079</td>\n",
       "      <td>-4</td>\n",
       "      <td>6079</td>\n",
       "      <td>-3</td>\n",
       "      <td>c.6079-4_6079-3del</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>29i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2536</th>\n",
       "      <td>0000964211</td>\n",
       "      <td>00007329</td>\n",
       "      <td>30</td>\n",
       "      <td>2024</td>\n",
       "      <td>-14</td>\n",
       "      <td>2024</td>\n",
       "      <td>-14</td>\n",
       "      <td>c.2024-14C&gt;T</td>\n",
       "      <td>r.(=)</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2537</th>\n",
       "      <td>0000964212</td>\n",
       "      <td>00007329</td>\n",
       "      <td>30</td>\n",
       "      <td>2024</td>\n",
       "      <td>-15</td>\n",
       "      <td>2024</td>\n",
       "      <td>-15</td>\n",
       "      <td>c.2024-15del</td>\n",
       "      <td>r.(=)</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2538</th>\n",
       "      <td>0000964215</td>\n",
       "      <td>00007329</td>\n",
       "      <td>50</td>\n",
       "      <td>1852</td>\n",
       "      <td>0</td>\n",
       "      <td>1852</td>\n",
       "      <td>0</td>\n",
       "      <td>c.1852G&gt;A</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Gly618Ser)</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2539</th>\n",
       "      <td>0000964216</td>\n",
       "      <td>00007329</td>\n",
       "      <td>50</td>\n",
       "      <td>1765</td>\n",
       "      <td>0</td>\n",
       "      <td>1765</td>\n",
       "      <td>0</td>\n",
       "      <td>c.1765A&gt;G</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Arg589Gly)</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2540</th>\n",
       "      <td>0000977314</td>\n",
       "      <td>00007329</td>\n",
       "      <td>90</td>\n",
       "      <td>8984</td>\n",
       "      <td>0</td>\n",
       "      <td>8984</td>\n",
       "      <td>0</td>\n",
       "      <td>c.8984T&gt;A</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Ile2995Asn)</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2541 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              id transcriptid effectid position_c_start   \n",
       "0     0000036426     00007329       50             7558  \\\n",
       "1     0000059881     00007329       55             2309   \n",
       "2     0000059883     00007329       11             3444   \n",
       "3     0000059884     00007329       15             4891   \n",
       "4     0000059885     00007329       11             6079   \n",
       "...          ...          ...      ...              ...   \n",
       "2536  0000964211     00007329       30             2024   \n",
       "2537  0000964212     00007329       30             2024   \n",
       "2538  0000964215     00007329       50             1852   \n",
       "2539  0000964216     00007329       50             1765   \n",
       "2540  0000977314     00007329       90             8984   \n",
       "\n",
       "     position_c_start_intron position_c_end position_c_end_intron   \n",
       "0                          0           7558                     0  \\\n",
       "1                          0           2309                     0   \n",
       "2                         -5           3444                    -5   \n",
       "3                          0           4891                     0   \n",
       "4                         -4           6079                    -3   \n",
       "...                      ...            ...                   ...   \n",
       "2536                     -14           2024                   -14   \n",
       "2537                     -15           2024                   -15   \n",
       "2538                       0           1852                     0   \n",
       "2539                       0           1765                     0   \n",
       "2540                       0           8984                     0   \n",
       "\n",
       "     VariantOnTranscript/DNA VariantOnTranscript/RNA   \n",
       "0                  c.7558T>C                   r.(?)  \\\n",
       "1                  c.2309A>C                   r.(?)   \n",
       "2                c.3444-5C>T                   r.(?)   \n",
       "3                  c.4891C>T                   r.(?)   \n",
       "4         c.6079-4_6079-3del                   r.(?)   \n",
       "...                      ...                     ...   \n",
       "2536            c.2024-14C>T                   r.(=)   \n",
       "2537            c.2024-15del                   r.(=)   \n",
       "2538               c.1852G>A                   r.(?)   \n",
       "2539               c.1765A>G                   r.(?)   \n",
       "2540               c.8984T>A                   r.(?)   \n",
       "\n",
       "     VariantOnTranscript/Protein VariantOnTranscript/Exon  \n",
       "0                 p.(Phe2520Leu)                       38  \n",
       "1                  p.(Gln770Pro)                       15  \n",
       "2                          p.(=)                      22i  \n",
       "3                 p.(Pro1631Ser)                       26  \n",
       "4                          p.(=)                      29i  \n",
       "...                          ...                      ...  \n",
       "2536                       p.(=)                           \n",
       "2537                       p.(=)                           \n",
       "2538               p.(Gly618Ser)                           \n",
       "2539               p.(Arg589Gly)                           \n",
       "2540              p.(Ile2995Asn)                           \n",
       "\n",
       "[2541 rows x 11 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screenings_To_Variants\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>screeningid</th>\n",
       "      <th>variantid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0000000126</td>\n",
       "      <td>0000783293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0000000211</td>\n",
       "      <td>0000790459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0000001640</td>\n",
       "      <td>0000235838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0000016557</td>\n",
       "      <td>0000036426</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0000033164</td>\n",
       "      <td>0000059884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2144</th>\n",
       "      <td>0000449279</td>\n",
       "      <td>0000959046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2145</th>\n",
       "      <td>0000449284</td>\n",
       "      <td>0000959051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2146</th>\n",
       "      <td>0000449293</td>\n",
       "      <td>0000959060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2147</th>\n",
       "      <td>0000449295</td>\n",
       "      <td>0000959474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2148</th>\n",
       "      <td>0000449297</td>\n",
       "      <td>0000959064</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2149 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     screeningid   variantid\n",
       "0     0000000126  0000783293\n",
       "1     0000000211  0000790459\n",
       "2     0000001640  0000235838\n",
       "3     0000016557  0000036426\n",
       "4     0000033164  0000059884\n",
       "...          ...         ...\n",
       "2144  0000449279  0000959046\n",
       "2145  0000449284  0000959051\n",
       "2146  0000449293  0000959060\n",
       "2147  0000449295  0000959474\n",
       "2148  0000449297  0000959064\n",
       "\n",
       "[2149 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data = parse_lovd(LOVD_PATH + \"/lovd_data.txt\")\n",
    "for i in data:\n",
    "    print(i)\n",
    "    display(data[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ef07740b2fa63e42",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genes\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1 entries, 0 to 0\n",
      "Data columns (total 34 columns):\n",
      " #   Column           Non-Null Count  Dtype         \n",
      "---  ------           --------------  -----         \n",
      " 0   id               1 non-null      string        \n",
      " 1   name             1 non-null      string        \n",
      " 2   chromosome       1 non-null      Int64         \n",
      " 3   chrom_band       1 non-null      string        \n",
      " 4   imprinting       1 non-null      string        \n",
      " 5   refseq_genomic   1 non-null      string        \n",
      " 6   refseq_UD        1 non-null      string        \n",
      " 7   reference        1 non-null      string        \n",
      " 8   url_homepage     1 non-null      string        \n",
      " 9   url_external     1 non-null      string        \n",
      " 10  allow_download   1 non-null      bool          \n",
      " 11  id_hgnc          1 non-null      Int64         \n",
      " 12  id_entrez        1 non-null      Int64         \n",
      " 13  id_omim          1 non-null      Int64         \n",
      " 14  show_hgmd        1 non-null      bool          \n",
      " 15  show_genecards   1 non-null      bool          \n",
      " 16  show_genetests   1 non-null      bool          \n",
      " 17  show_orphanet    1 non-null      bool          \n",
      " 18  note_index       1 non-null      string        \n",
      " 19  note_listing     1 non-null      string        \n",
      " 20  refseq           1 non-null      string        \n",
      " 21  refseq_url       1 non-null      string        \n",
      " 22  disclaimer       1 non-null      bool          \n",
      " 23  disclaimer_text  1 non-null      string        \n",
      " 24  header           1 non-null      string        \n",
      " 25  header_align     1 non-null      Int64         \n",
      " 26  footer           1 non-null      string        \n",
      " 27  footer_align     1 non-null      Int64         \n",
      " 28  created_by       1 non-null      Int64         \n",
      " 29  created_date     1 non-null      datetime64[ns]\n",
      " 30  edited_by        1 non-null      Int64         \n",
      " 31  edited_date      1 non-null      datetime64[ns]\n",
      " 32  updated_by       1 non-null      Int64         \n",
      " 33  updated_date     1 non-null      datetime64[ns]\n",
      "dtypes: Int64(9), bool(6), datetime64[ns](3), string(16)\n",
      "memory usage: 367.0 bytes\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transcripts\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1 entries, 0 to 0\n",
      "Data columns (total 19 columns):\n",
      " #   Column                 Non-Null Count  Dtype         \n",
      "---  ------                 --------------  -----         \n",
      " 0   id                     1 non-null      Int64         \n",
      " 1   geneid                 1 non-null      string        \n",
      " 2   name                   1 non-null      string        \n",
      " 3   id_mutalyzer           1 non-null      Int64         \n",
      " 4   id_ncbi                1 non-null      string        \n",
      " 5   id_ensembl             1 non-null      string        \n",
      " 6   id_protein_ncbi        1 non-null      string        \n",
      " 7   id_protein_ensembl     1 non-null      string        \n",
      " 8   id_protein_uniprot     1 non-null      string        \n",
      " 9   remarks                1 non-null      string        \n",
      " 10  position_c_mrna_start  1 non-null      Int64         \n",
      " 11  position_c_mrna_end    1 non-null      Int64         \n",
      " 12  position_c_cds_end     1 non-null      Int64         \n",
      " 13  position_g_mrna_start  1 non-null      Int64         \n",
      " 14  position_g_mrna_end    1 non-null      Int64         \n",
      " 15  created_by             0 non-null      Int64         \n",
      " 16  created_date           0 non-null      datetime64[ns]\n",
      " 17  edited_by              0 non-null      Int64         \n",
      " 18  edited_date            0 non-null      datetime64[ns]\n",
      "dtypes: Int64(9), datetime64[ns](2), string(8)\n",
      "memory usage: 289.0 bytes\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Diseases\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 15 entries, 0 to 14\n",
      "Data columns (total 12 columns):\n",
      " #   Column        Non-Null Count  Dtype         \n",
      "---  ------        --------------  -----         \n",
      " 0   id            15 non-null     Int64         \n",
      " 1   symbol        15 non-null     string        \n",
      " 2   name          15 non-null     string        \n",
      " 3   inheritance   15 non-null     string        \n",
      " 4   id_omim       4 non-null      Int64         \n",
      " 5   tissues       15 non-null     string        \n",
      " 6   features      15 non-null     string        \n",
      " 7   remarks       15 non-null     string        \n",
      " 8   created_by    15 non-null     Int64         \n",
      " 9   created_date  15 non-null     datetime64[ns]\n",
      " 10  edited_by     11 non-null     Int64         \n",
      " 11  edited_date   11 non-null     datetime64[ns]\n",
      "dtypes: Int64(4), datetime64[ns](2), string(6)\n",
      "memory usage: 1.6 KB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genes_To_Diseases\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 2 entries, 0 to 1\n",
      "Data columns (total 2 columns):\n",
      " #   Column     Non-Null Count  Dtype \n",
      "---  ------     --------------  ----- \n",
      " 0   geneid     2 non-null      string\n",
      " 1   diseaseid  2 non-null      Int64 \n",
      "dtypes: Int64(1), string(1)\n",
      "memory usage: 162.0 bytes\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1450 entries, 0 to 1449\n",
      "Data columns (total 18 columns):\n",
      " #   Column                        Non-Null Count  Dtype \n",
      "---  ------                        --------------  ----- \n",
      " 0   id                            1450 non-null   Int64 \n",
      " 1   fatherid                      1450 non-null   string\n",
      " 2   motherid                      1450 non-null   string\n",
      " 3   panelid                       6 non-null      Int64 \n",
      " 4   panel_size                    1450 non-null   Int64 \n",
      " 5   license                       1450 non-null   string\n",
      " 6   owned_by                      1450 non-null   Int64 \n",
      " 7   Individual/Reference          1450 non-null   string\n",
      " 8   Individual/Remarks            1450 non-null   string\n",
      " 9   Individual/Gender             1450 non-null   string\n",
      " 10  Individual/Consanguinity      1450 non-null   string\n",
      " 11  Individual/Origin/Geographic  1450 non-null   string\n",
      " 12  Individual/Age_of_death       1450 non-null   string\n",
      " 13  Individual/VIP                1450 non-null   string\n",
      " 14  Individual/Data_av            1450 non-null   string\n",
      " 15  Individual/Treatment          1450 non-null   string\n",
      " 16  Individual/Origin/Population  1450 non-null   string\n",
      " 17  Individual/Individual_ID      1450 non-null   string\n",
      "dtypes: Int64(4), string(14)\n",
      "memory usage: 209.7 KB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals_To_Diseases\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1449 entries, 0 to 1448\n",
      "Data columns (total 2 columns):\n",
      " #   Column        Non-Null Count  Dtype\n",
      "---  ------        --------------  -----\n",
      " 0   individualid  1449 non-null   Int64\n",
      " 1   diseaseid     1449 non-null   Int64\n",
      "dtypes: Int64(2)\n",
      "memory usage: 25.6 KB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Phenotypes\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1271 entries, 0 to 1270\n",
      "Data columns (total 20 columns):\n",
      " #   Column                        Non-Null Count  Dtype \n",
      "---  ------                        --------------  ----- \n",
      " 0   id                            1271 non-null   Int64 \n",
      " 1   diseaseid                     1271 non-null   Int64 \n",
      " 2   individualid                  1271 non-null   Int64 \n",
      " 3   owned_by                      1271 non-null   Int64 \n",
      " 4   Phenotype/Inheritance         1271 non-null   string\n",
      " 5   Phenotype/Age                 1271 non-null   string\n",
      " 6   Phenotype/Additional          1271 non-null   string\n",
      " 7   Phenotype/Biochem_param       1271 non-null   string\n",
      " 8   Phenotype/Age/Onset           1271 non-null   string\n",
      " 9   Phenotype/Age/Diagnosis       1271 non-null   string\n",
      " 10  Phenotype/Severity_score      1271 non-null   string\n",
      " 11  Phenotype/Onset               1271 non-null   string\n",
      " 12  Phenotype/Protein             1271 non-null   string\n",
      " 13  Phenotype/Tumor/MSI           1271 non-null   string\n",
      " 14  Phenotype/Enzyme/CPK          1271 non-null   string\n",
      " 15  Phenotype/Heart/Myocardium    1271 non-null   string\n",
      " 16  Phenotype/Lung                1271 non-null   string\n",
      " 17  Phenotype/Diagnosis/Definite  1271 non-null   string\n",
      " 18  Phenotype/Diagnosis/Initial   1271 non-null   string\n",
      " 19  Phenotype/Diagnosis/Criteria  1271 non-null   string\n",
      "dtypes: Int64(4), string(16)\n",
      "memory usage: 203.7 KB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screenings\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1450 entries, 0 to 1449\n",
      "Data columns (total 12 columns):\n",
      " #   Column               Non-Null Count  Dtype         \n",
      "---  ------               --------------  -----         \n",
      " 0   id                   1450 non-null   Int64         \n",
      " 1   individualid         1450 non-null   Int64         \n",
      " 2   variants_found       1450 non-null   Int64         \n",
      " 3   owned_by             1450 non-null   Int64         \n",
      " 4   created_by           1450 non-null   Int64         \n",
      " 5   created_date         1450 non-null   datetime64[ns]\n",
      " 6   edited_by            15 non-null     Int64         \n",
      " 7   edited_date          15 non-null     datetime64[ns]\n",
      " 8   Screening/Technique  1450 non-null   string        \n",
      " 9   Screening/Template   1450 non-null   string        \n",
      " 10  Screening/Tissue     1450 non-null   string        \n",
      " 11  Screening/Remarks    1450 non-null   string        \n",
      "dtypes: Int64(6), datetime64[ns](2), string(4)\n",
      "memory usage: 144.6 KB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screenings_To_Genes\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1316 entries, 0 to 1315\n",
      "Data columns (total 2 columns):\n",
      " #   Column       Non-Null Count  Dtype \n",
      "---  ------       --------------  ----- \n",
      " 0   screeningid  1316 non-null   Int64 \n",
      " 1   geneid       1316 non-null   string\n",
      "dtypes: Int64(1), string(1)\n",
      "memory usage: 22.0 KB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants_On_Genome\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 2541 entries, 0 to 2540\n",
      "Data columns (total 26 columns):\n",
      " #   Column                                         Non-Null Count  Dtype  \n",
      "---  ------                                         --------------  -----  \n",
      " 0   id                                             2541 non-null   Int64  \n",
      " 1   allele                                         2541 non-null   Int64  \n",
      " 2   effectid                                       2541 non-null   Int64  \n",
      " 3   chromosome                                     2541 non-null   Int64  \n",
      " 4   position_g_start                               2540 non-null   Int64  \n",
      " 5   position_g_end                                 2540 non-null   Int64  \n",
      " 6   type                                           2541 non-null   string \n",
      " 7   average_frequency                              2540 non-null   float64\n",
      " 8   owned_by                                       2541 non-null   Int64  \n",
      " 9   VariantOnGenome/DBID                           2541 non-null   string \n",
      " 10  VariantOnGenome/DNA                            2541 non-null   string \n",
      " 11  VariantOnGenome/Frequency                      2541 non-null   string \n",
      " 12  VariantOnGenome/Reference                      2541 non-null   string \n",
      " 13  VariantOnGenome/Restriction_site               2541 non-null   string \n",
      " 14  VariantOnGenome/Published_as                   2541 non-null   string \n",
      " 15  VariantOnGenome/Remarks                        2541 non-null   string \n",
      " 16  VariantOnGenome/Genetic_origin                 2541 non-null   string \n",
      " 17  VariantOnGenome/Segregation                    2541 non-null   string \n",
      " 18  VariantOnGenome/dbSNP                          2541 non-null   string \n",
      " 19  VariantOnGenome/VIP                            2541 non-null   string \n",
      " 20  VariantOnGenome/Methylation                    2541 non-null   string \n",
      " 21  VariantOnGenome/ISCN                           2541 non-null   string \n",
      " 22  VariantOnGenome/DNA/hg38                       2541 non-null   string \n",
      " 23  VariantOnGenome/ClinVar                        2541 non-null   string \n",
      " 24  VariantOnGenome/ClinicalClassification         2541 non-null   string \n",
      " 25  VariantOnGenome/ClinicalClassification/Method  2541 non-null   string \n",
      "dtypes: Int64(7), float64(1), string(18)\n",
      "memory usage: 533.6 KB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants_On_Transcripts\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 2541 entries, 0 to 2540\n",
      "Data columns (total 11 columns):\n",
      " #   Column                       Non-Null Count  Dtype \n",
      "---  ------                       --------------  ----- \n",
      " 0   id                           2541 non-null   Int64 \n",
      " 1   transcriptid                 2541 non-null   Int64 \n",
      " 2   effectid                     2541 non-null   Int64 \n",
      " 3   position_c_start             2540 non-null   Int64 \n",
      " 4   position_c_start_intron      2541 non-null   Int64 \n",
      " 5   position_c_end               2540 non-null   Int64 \n",
      " 6   position_c_end_intron        2541 non-null   Int64 \n",
      " 7   VariantOnTranscript/DNA      2541 non-null   string\n",
      " 8   VariantOnTranscript/RNA      2541 non-null   string\n",
      " 9   VariantOnTranscript/Protein  2541 non-null   string\n",
      " 10  VariantOnTranscript/Exon     2541 non-null   string\n",
      "dtypes: Int64(7), string(4)\n",
      "memory usage: 235.9 KB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screenings_To_Variants\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 2149 entries, 0 to 2148\n",
      "Data columns (total 2 columns):\n",
      " #   Column       Non-Null Count  Dtype\n",
      "---  ------       --------------  -----\n",
      " 0   screeningid  2149 non-null   Int64\n",
      " 1   variantid    2149 non-null   Int64\n",
      "dtypes: Int64(2)\n",
      "memory usage: 37.9 KB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "set_lovd_dtypes(data)\n",
    "for i in data:\n",
    "    print(i)\n",
    "    display(data[i].info())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d5ee0724",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>allele</th>\n",
       "      <th>effectid</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position_g_start</th>\n",
       "      <th>position_g_end</th>\n",
       "      <th>type</th>\n",
       "      <th>average_frequency</th>\n",
       "      <th>owned_by</th>\n",
       "      <th>VariantOnGenome/DBID</th>\n",
       "      <th>...</th>\n",
       "      <th>VariantOnGenome/Genetic_origin</th>\n",
       "      <th>VariantOnGenome/Segregation</th>\n",
       "      <th>VariantOnGenome/dbSNP</th>\n",
       "      <th>VariantOnGenome/VIP</th>\n",
       "      <th>VariantOnGenome/Methylation</th>\n",
       "      <th>VariantOnGenome/ISCN</th>\n",
       "      <th>VariantOnGenome/DNA/hg38</th>\n",
       "      <th>VariantOnGenome/ClinVar</th>\n",
       "      <th>VariantOnGenome/ClinicalClassification</th>\n",
       "      <th>VariantOnGenome/ClinicalClassification/Method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>59881</td>\n",
       "      <td>3</td>\n",
       "      <td>55</td>\n",
       "      <td>6</td>\n",
       "      <td>65655758</td>\n",
       "      <td>65655758</td>\n",
       "      <td>subst</td>\n",
       "      <td>0.001153</td>\n",
       "      <td>229</td>\n",
       "      <td>EYS_000001</td>\n",
       "      <td>...</td>\n",
       "      <td>Germline</td>\n",
       "      <td>yes</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>g.64945865T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>VUS</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      id allele effectid chromosome position_g_start position_g_end   type   \n",
       "1  59881      3       55          6         65655758       65655758  subst  \\\n",
       "\n",
       "   average_frequency owned_by VariantOnGenome/DBID  ...   \n",
       "1           0.001153      229           EYS_000001  ...  \\\n",
       "\n",
       "  VariantOnGenome/Genetic_origin VariantOnGenome/Segregation   \n",
       "1                       Germline                         yes  \\\n",
       "\n",
       "  VariantOnGenome/dbSNP VariantOnGenome/VIP VariantOnGenome/Methylation   \n",
       "1                                         0                              \\\n",
       "\n",
       "  VariantOnGenome/ISCN VariantOnGenome/DNA/hg38 VariantOnGenome/ClinVar   \n",
       "1                                 g.64945865T>G                          \\\n",
       "\n",
       "  VariantOnGenome/ClinicalClassification   \n",
       "1                                    VUS  \\\n",
       "\n",
       "  VariantOnGenome/ClinicalClassification/Method  \n",
       "1                                                \n",
       "\n",
       "[1 rows x 26 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "'Variant scoring:'"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6</td>\n",
       "      <td>65655758</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.917516</td>\n",
       "      <td>9.431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>65655758</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.934118</td>\n",
       "      <td>9.597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6</td>\n",
       "      <td>65655758</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.609154</td>\n",
       "      <td>6.542</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chrom       Pos Ref Alt  RawScore  PHRED\n",
       "0     6  65655758   A   C  0.917516  9.431\n",
       "1     6  65655758   A   G  0.934118  9.597\n",
       "2     6  65655758   A   T  0.609154  6.542"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from tools import (fetch_cadd_score, fetch_cadd_scores)\n",
    "import pandas as pd\n",
    "\n",
    "row_idx = 1\n",
    "selected_row = data[\"Variants_On_Genome\"].iloc[row_idx]\n",
    "\n",
    "selected_row_df = pd.DataFrame([selected_row])\n",
    "headers = pd.DataFrame(columns=selected_row.index)\n",
    "result_df = pd.concat([headers, selected_row_df])\n",
    "display(result_df)\n",
    "\n",
    "chromosome = selected_row[\"chromosome\"]\n",
    "pos_start = selected_row[\"position_g_start\"]\n",
    "pos_end = selected_row[\"position_g_end\"]\n",
    "\n",
    "display(\"Variant scoring:\")\n",
    "score = fetch_cadd_scores(\"GRCh38-v1.7\", chromosome, pos_start, pos_end)\n",
    "score_df = pd.DataFrame(score[1:], columns=score[0])\n",
    "display(score_df)\n",
    "      "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c968af1617be40db",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-13T15:38:25.149624Z",
     "start_time": "2024-05-13T15:38:24.807199Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:root:Skipping variant g.64307105_64307106del\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64902422_64902438del\n",
      "WARNING:root:Skipping variant g.64902422_64902438del\n",
      "WARNING:root:Skipping variant g.64840707_64997105del\n",
      "WARNING:root:Skipping variant g.64840707_64997105del\n",
      "WARNING:root:Skipping variant g.64840707_64997105del\n",
      "WARNING:root:Skipping variant g.65295915del\n",
      "WARNING:root:Skipping variant g.65295915del\n",
      "WARNING:root:Skipping variant g.65295915del\n",
      "WARNING:root:Skipping variant g.65057728_65320715del\n",
      "WARNING:root:Skipping variant g.65057728_65320715del\n",
      "WARNING:root:Skipping variant g.65057728_65320715del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65384425del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64439165del\n",
      "WARNING:root:Skipping variant g.64439165del\n",
      "WARNING:root:Skipping variant g.64626122del\n",
      "WARNING:root:Skipping variant g.65494867del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65494867del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.63720919_63720920del\n",
      "WARNING:root:Skipping variant g.63720668dup\n",
      "WARNING:root:Skipping variant g.63720919_63720920del\n",
      "WARNING:root:Skipping variant g.63720668dup\n",
      "WARNING:root:Skipping variant g.63999116del\n",
      "WARNING:root:Skipping variant g.64591039_64591042del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64822643dup\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.64591505_64591506delinsCT\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65384480dup\n",
      "WARNING:root:Skipping variant g.63726599_63726600del\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.63726599_63726600del\n",
      "WARNING:root:Skipping variant g.65495005_65495008delinsAAG\n",
      "WARNING:root:Skipping variant g.63726599_63726600del\n",
      "WARNING:root:Skipping variant g.65335102del\n",
      "WARNING:root:Skipping variant g.65335102del\n",
      "WARNING:root:Skipping variant g.65335102del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63999116del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63999110_63999111del\n",
      "WARNING:root:Skipping variant g.63999110_63999111del\n",
      "WARNING:root:Skipping variant g.65321830_65370656del\n",
      "WARNING:root:Skipping variant g.63720850_63720853del\n",
      "WARNING:root:Skipping variant g.65321830_65370656del\n",
      "WARNING:root:Skipping variant g.63720850_63720853del\n",
      "WARNING:root:Skipping variant g.65321830_65370656del\n",
      "WARNING:root:Skipping variant g.63720850_63720853del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63806240del\n",
      "WARNING:root:Skipping variant g.63806240del\n",
      "WARNING:root:Skipping variant g.65495332_65495333dup\n",
      "WARNING:root:Skipping variant g.65324960_65416038del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65295915del\n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.65494988_65495003del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.65384480dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.65344144_65344152delinsCTTTTCG\n",
      "WARNING:root:Skipping variant g.63984409_63984410delinsACGAT\n",
      "WARNING:root:Skipping variant g.63788163_63788164del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64912603dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590700dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590700dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64591845del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63984390del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63721651_63721652insCA\n",
      "WARNING:root:Skipping variant g.64590665_64590666del\n",
      "WARNING:root:Skipping variant g.63721651_63721652insCA\n",
      "WARNING:root:Skipping variant g.64590665_64590666del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64591466dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64591480del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63788136del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63720730_63720733del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64912603dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590665_64590666del\n",
      "WARNING:root:Skipping variant g.63721652_63721655dup\n",
      "WARNING:root:Skipping variant g.64590665_64590666del\n",
      "WARNING:root:Skipping variant g.63721652_63721655dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64912603dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64591313del\n",
      "WARNING:root:Skipping variant g.64439355_64439356insA\n",
      "WARNING:root:Skipping variant g.63984543_63984570del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63984537_63984542dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64614795_64939832del\n",
      "WARNING:root:Skipping variant g.64614795_64939832del\n",
      "WARNING:root:Skipping variant g.64614795_64939832del\n",
      "WARNING:root:Skipping variant g.63721576del\n",
      "WARNING:root:Skipping variant g.64902132_64902133del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720889dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63721625dup\n",
      "WARNING:root:Skipping variant g.65405300dup\n",
      "WARNING:root:Skipping variant g.64813506del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63720753_63720754dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.65405325dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.65707136_65707226del\n",
      "WARNING:root:Skipping variant g.65495348del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63720845_63720846del\n",
      "WARNING:root:Skipping variant g.63726607del\n",
      "WARNING:root:Skipping variant g.63721786dup\n",
      "WARNING:root:Skipping variant g.63721640del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65344138dup\n",
      "WARNING:root:Skipping variant g.65402503del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590699_64590700del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590699_64590700del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65384384_65384387del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65384480dup\n",
      "WARNING:root:Skipping variant g.63806228dup\n",
      "WARNING:root:Skipping variant g.63999116del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.65384384_65384387del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63999110_63999111del\n",
      "WARNING:root:Skipping variant g.63999110_63999111del\n",
      "WARNING:root:Skipping variant g.63999110_63999111del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64439331del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65494957dup\n",
      "WARNING:root:Skipping variant g.63720995del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63721771_63721776del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.65494957dup\n",
      "WARNING:root:Skipping variant g.64591977del\n",
      "WARNING:root:Skipping variant g.64886728_64886736del\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant g.63721138del\n",
      "WARNING:root:Skipping variant g.65353541del\n",
      "WARNING:root:Skipping variant g.65353541del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65494988_65495002del\n",
      "WARNING:root:Skipping variant g.65295857dup\n",
      "WARNING:root:Skipping variant g.65295856_65295857dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64307105_64307106del\n",
      "WARNING:root:Skipping variant g.64307103_64307106del\n",
      "WARNING:root:Skipping variant g.63726618_63726622del\n",
      "WARNING:root:Skipping variant g.63720642_63720644del\n",
      "WARNING:root:Skipping variant g.65295857dup\n",
      "WARNING:root:Skipping variant g.64307105_64307106del\n",
      "WARNING:root:Skipping variant g.65353541del\n",
      "WARNING:root:Skipping variant g.64591918_64591919del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64590525_64590548del\n",
      "WARNING:root:Skipping variant g.64388841_64388843del\n",
      "WARNING:root:Skipping variant g.64307084_64307085del\n",
      "WARNING:root:Skipping variant g.64307105_64307106del\n",
      "WARNING:root:Skipping variant g.64307103_64307106del\n",
      "WARNING:root:Skipping variant g.64307105_64307106del\n",
      "WARNING:root:Skipping variant g.64593097_64593101del\n",
      "WARNING:root:Skipping variant g.65384384_65384387del\n",
      "WARNING:root:Skipping variant g.63999110_63999111del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64439213del\n",
      "WARNING:root:Skipping variant g.63721771_63721776del\n",
      "WARNING:root:Skipping variant g.65494957dup\n",
      "WARNING:root:Skipping variant g.65494988_65495008delinsAAAAG\n",
      "WARNING:root:Skipping variant g.63720799_63720808del\n",
      "WARNING:root:Skipping variant g.63726584del\n",
      "WARNING:root:Skipping variant g.63726599_63726600del\n",
      "WARNING:root:Skipping variant g.63726648del\n",
      "WARNING:root:Skipping variant g.64349976_64426764del\n",
      "WARNING:root:Skipping variant g.64591505_64591506delinsCT\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720728_63720729del\n",
      "WARNING:root:Skipping variant g.63720845_63720846del\n",
      "WARNING:root:Skipping variant g.63721162del\n",
      "WARNING:root:Skipping variant g.63721237_63721240del\n",
      "WARNING:root:Skipping variant g.63721343del\n",
      "WARNING:root:Skipping variant g.63721730_63721733del\n",
      "WARNING:root:Skipping variant g.63721763_63721767del\n",
      "WARNING:root:Skipping variant g.63721787dup\n",
      "WARNING:root:Skipping variant g.63726618_63726622del\n",
      "WARNING:root:Skipping variant g.63788163_63788164del\n",
      "WARNING:root:Skipping variant g.63984389del\n",
      "WARNING:root:Skipping variant g.64066348del\n",
      "WARNING:root:Skipping variant g.64590909dup\n",
      "WARNING:root:Skipping variant g.64591309del\n",
      "WARNING:root:Skipping variant g.64591858_64591859del\n",
      "WARNING:root:Skipping variant g.64886728_64886736del\n",
      "WARNING:root:Skipping variant g.64945814del\n",
      "WARNING:root:Skipping variant g.65344143_65344144insCTTT\n",
      "WARNING:root:Skipping variant g.65344146_65344151del\n",
      "WARNING:root:Skipping variant g.65344181dup\n",
      "WARNING:root:Skipping variant g.65384473dup\n",
      "WARNING:root:Skipping variant g.65405287del\n",
      "WARNING:root:Skipping variant g.65494885_65494887del\n",
      "WARNING:root:Skipping variant g.65494961del\n",
      "WARNING:root:Skipping variant g.65495205del\n",
      "WARNING:root:Skipping variant g.65495296_65495297del\n",
      "WARNING:root:Skipping variant g.64590909dup\n",
      "WARNING:root:Skipping variant g.65384473dup\n",
      "WARNING:root:Skipping variant g.65405287del\n",
      "WARNING:root:Skipping variant g.65494885_65494887del\n",
      "WARNING:root:Skipping variant g.63762589del\n",
      "WARNING:root:Skipping variant g.63720874del\n",
      "WARNING:root:Skipping variant g.63721619_63721620insGT\n",
      "WARNING:root:Skipping variant g.63788268dup\n",
      "WARNING:root:Skipping variant g.63984369_63984392del\n",
      "WARNING:root:Skipping variant g.65057740_65057741insA\n",
      "WARNING:root:Skipping variant g.65057740_65057741insAA\n",
      "WARNING:root:Skipping variant g.65057750dup\n",
      "WARNING:root:Skipping variant g.65384384_65384387del\n",
      "WARNING:root:Skipping variant g.65402624dup\n",
      "WARNING:root:Skipping variant g.65405377dup\n",
      "WARNING:root:Skipping variant g.65494957dup\n",
      "WARNING:root:Skipping variant g.65494957dup\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant g.63721625dup\n",
      "WARNING:root:Skipping variant g.63721704dup\n",
      "WARNING:root:Skipping variant g.65353537dup\n",
      "WARNING:root:Skipping variant g.65335105_65335108del\n",
      "WARNING:root:Skipping variant g.63721385_63721386del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64439195del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63720867_63720868del\n",
      "WARNING:root:Skipping variant g.63721314_63721321del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63765706_63791377del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63781919_63803805del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64892926_64948294del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64591845del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64945857dup\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63721557_63721558insA\n",
      "WARNING:root:Skipping variant g.64439319_64439323del\n",
      "WARNING:root:Skipping variant g.65296021del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65384425del\n",
      "WARNING:root:Skipping variant g.64591039_64591042del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63721599_63721604del\n",
      "WARNING:root:Skipping variant g.64902132_64902133del\n",
      "WARNING:root:Skipping variant g.65494988_65495008delinsAAAAG\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65495236del\n",
      "WARNING:root:Skipping variant g.64590908_64590909insT\n",
      "WARNING:root:Skipping variant g.64617409_64617411dup\n",
      "WARNING:root:Skipping variant g.65295857dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63720845_63720846del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720668dup\n",
      "WARNING:root:Skipping variant g.(63741975_63762461)_(63778180_63788105)dup\n",
      "WARNING:root:Skipping variant g.65296051_65296052del\n",
      "WARNING:root:Skipping variant g.65296051_65296052del\n",
      "WARNING:root:Skipping variant g.64439200dup\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.65494885_65494887del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63984390del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.65295897del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65494885_65494887del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590615dup\n",
      "WARNING:root:Skipping variant g.63721705del\n",
      "WARNING:root:Skipping variant g.64081884del\n",
      "WARNING:root:Skipping variant g.65405342_65405355delinsAAA\n",
      "WARNING:root:Skipping variant g.65490640_65490643dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65296058del\n",
      "WARNING:root:Skipping variant g.65405345del\n",
      "WARNING:root:Skipping variant g.(65405368_65490593)_(65495411_?)del\n",
      "WARNING:root:Skipping variant g.(65353618_65384385)_(65405368_65490593)del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64081885_64081888del\n",
      "WARNING:root:Skipping variant g.64591401_64591408dup\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63721237_63721240del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64591256_64591272del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.63721436del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64591039_64591042del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.63720695_63720714del\n",
      "WARNING:root:Skipping variant g.65384480dup\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.65495064del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64066470_64066473del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64066470_64066473del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.63720695_63720714del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720668del\n",
      "WARNING:root:Skipping variant g.64912603dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63720730_63720733del\n",
      "WARNING:root:Skipping variant g.64902132_64902133del\n",
      "WARNING:root:Skipping variant g.63720954dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64066335del\n",
      "WARNING:root:Skipping variant g.64886711del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.63778166_63778172del\n",
      "WARNING:root:Skipping variant g.64590914dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63726618_63726622del\n",
      "WARNING:root:Skipping variant g.64886728_64886736del\n",
      "WARNING:root:Skipping variant g.64893157_64947352del\n",
      "WARNING:root:Skipping variant g.64797009_64846087del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.63726519del\n",
      "WARNING:root:Skipping variant g.64997275_64998015del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64081605_64082252del\n",
      "WARNING:root:Skipping variant g.63777755_63789474dup\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.64081605_64082252del\n",
      "WARNING:root:Skipping variant g.65353208_65353867del\n",
      "WARNING:root:Skipping variant g.63984134_63984854del\n",
      "WARNING:root:Skipping variant g.63743557_63907234del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.64662532_64979780del\n",
      "WARNING:root:Skipping variant g.64839119_64970113del\n",
      "WARNING:root:Skipping variant g.63720872_63720873del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64081885_64081888del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.63720682dup\n",
      "WARNING:root:Skipping variant g.63721619dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64388841del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63720649_63720653del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63720695_63720714del\n",
      "WARNING:root:Skipping variant g.65384384_65384387del\n",
      "WARNING:root:Skipping variant g.63720730_63720733del\n",
      "WARNING:root:Skipping variant g.65335105_65335108del\n",
      "WARNING:root:Skipping variant g.64439165del\n",
      "WARNING:root:Skipping variant g.63999116del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63998527_64002156del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.65495379dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63720730_63720733del\n",
      "WARNING:root:Skipping variant g.63720730_63720733del\n",
      "WARNING:root:Skipping variant g.63721237_63721240del\n",
      "WARNING:root:Skipping variant g.63721237_63721240del\n",
      "WARNING:root:Skipping variant g.63721237_63721240del\n",
      "WARNING:root:Skipping variant g.64081885_64081888del\n",
      "WARNING:root:Skipping variant g.65495348dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64439331del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64912705del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64439200dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.65495181del\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.63721625dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.65384480del\n",
      "WARNING:root:Skipping variant g.64388841_64388843del\n",
      "WARNING:root:Skipping variant g.63998527_64002156del\n",
      "WARNING:root:Skipping variant g.64590525_64590550delinsTA\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64081884del\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.216327637C>T\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.215879068C>T\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.215878931del\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.65384480dup\n",
      "WARNING:root:Skipping variant g.216073265G>A\n",
      "WARNING:root:Skipping variant g.65495206_65495207insTGCCAGTTTA\n",
      "WARNING:root:Skipping variant g.63721227dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.63720990_63720991insATAT\n",
      "WARNING:root:Skipping variant g.63720728_63720729del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720947_63720948insT\n",
      "WARNING:root:Skipping variant g.64590875_64590876insTCTT\n",
      "WARNING:root:Skipping variant g.63721432del\n",
      "WARNING:root:Skipping variant g.64591501_64591502insAGAA\n",
      "WARNING:root:Skipping variant g.64590556_64590566del\n",
      "WARNING:root:Skipping variant g.63999116del\n",
      "WARNING:root:Skipping variant g.63720753_63720754dup\n",
      "WARNING:root:Skipping variant g.(?_64945792)_(64945915_64997581)del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63720730_63720733del\n",
      "WARNING:root:Skipping variant g.63721377_63721384del\n",
      "WARNING:root:Skipping variant g.63726524del\n",
      "WARNING:root:Skipping variant g.64423168_64798957delinsATGA\n",
      "WARNING:root:Skipping variant g.63942752_64337822delinsATTATG\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant g.?\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64423168_64798957delinsATGA\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63942752_64337822delinsATTATG\n",
      "WARNING:root:Skipping variant g.63957115_63958454del\n",
      "WARNING:root:Skipping variant g.65001113_65005820del\n",
      "WARNING:root:Skipping variant g.65550144_65552138del\n",
      "WARNING:root:Skipping variant g.65689153_65694794del\n",
      "WARNING:root:Skipping variant g.65454073_65454074insN[305]\n",
      "WARNING:root:Skipping variant g.64296539_64296632del\n",
      "WARNING:root:Skipping variant g.65204982_65205044del\n",
      "WARNING:root:Skipping variant g.65564961_65565284del\n",
      "WARNING:root:Skipping variant :g.64295412_64295413insN[118]\n",
      "WARNING:root:Skipping variant g.65278328_65278329insN[59]\n",
      "WARNING:root:Skipping variant g.64590911dup\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64590875_64590876insTCTT\n",
      "WARNING:root:Skipping variant g.63720947_63720948insT\n",
      "WARNING:root:Skipping variant g.64591501_64591502insAGAA\n",
      "WARNING:root:Skipping variant g.63721432del\n",
      "WARNING:root:Skipping variant g.64590556_64590566del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.65658176_65718924del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant NM_001142800.2:c.6079-2A>G\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.63999116del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.64986218_65013355del\n",
      "WARNING:root:Skipping variant g.64388690_64388840del\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant g.(?_63719980)_(63726681_63762460)dup\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.65274506_65316845delinsAGATCA\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.65494100_65508832del\n",
      "WARNING:root:Skipping variant g.65213025_65296862delinsGTTTTCTTTTTA\n",
      "WARNING:root:Skipping variant g.64066349del\n",
      "WARNING:root:Skipping variant g.64602159_64657461dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63762589del\n",
      "WARNING:root:Skipping variant g.65383303_65441305delinsAACTTTTACT\n",
      "WARNING:root:Skipping variant g.63720737_63720746del\n",
      "WARNING:root:Skipping variant g.65284957_66872862delinsT\n",
      "WARNING:root:Skipping variant g.64122444_64129159delins64204448_64235506inv\n",
      "WARNING:root:Skipping variant g.64937848_64948401delins[64944099_64944163inv;CAATTTTGTAT]\n",
      "WARNING:root:Skipping variant g.63721385_63721386del\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63720654_63720657del\n",
      "WARNING:root:Skipping variant g.64591069_64591081del\n",
      "WARNING:root:Skipping variant g.64886841del\n",
      "WARNING:root:Skipping variant g.64790603_64977512del\n",
      "WARNING:root:Skipping variant g.63721625dup\n",
      "WARNING:root:Skipping variant g.64591514_64591520del\n",
      "WARNING:root:Skipping variant g.63720649_63720653del\n",
      "WARNING:root:Skipping variant g.65479942_67131267inv\n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n",
      "WARNING:root:Skipping variant \n"
     ]
    }
   ],
   "source": [
    "save_lovd_as_vcf(data[\"Variants_On_Genome\"], \"./lovd.vcf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c7ff16903e0c52bd",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-13T15:58:47.868055Z",
     "start_time": "2024-05-13T15:58:41.380466Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-05-13 18:58:41.794056: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.\n",
      "2024-05-13 18:58:41.794769: I external/local_tsl/tsl/cuda/cudart_stub.cc:32] Could not find cuda drivers on your machine, GPU will not be used.\n",
      "2024-05-13 18:58:41.797917: I external/local_tsl/tsl/cuda/cudart_stub.cc:32] Could not find cuda drivers on your machine, GPU will not be used.\n",
      "2024-05-13 18:58:41.857361: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: AVX2 AVX_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2024-05-13 18:58:42.410244: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT\n",
      "2024-05-13 18:58:42.957291: I external/local_xla/xla/stream_executor/cuda/cuda_executor.cc:998] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-05-13 18:58:42.957684: W tensorflow/core/common_runtime/gpu/gpu_device.cc:2251] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.\n",
      "Skipping registering GPU devices...\n",
      "WARNING:absl:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.\n",
      "WARNING:absl:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.\n",
      "WARNING:absl:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.\n",
      "WARNING:absl:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.\n",
      "WARNING:absl:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1m1/1\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m1s\u001b[0m 595ms/step\n",
      "\u001b[1m1/1\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m1s\u001b[0m 554ms/step\n",
      "\u001b[1m1/1\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m1s\u001b[0m 553ms/step\n",
      "\u001b[1m1/1\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m1s\u001b[0m 548ms/step\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:5 out of the last 5 calls to <function TensorFlowTrainer.make_predict_function.<locals>.one_step_on_data_distributed at 0x7dee981abf40> triggered tf.function retracing. Tracing is expensive and the excessive number of tracings could be due to (1) creating @tf.function repeatedly in a loop, (2) passing tensors with different shapes, (3) passing Python objects instead of tensors. For (1), please define your @tf.function outside of the loop. For (2), @tf.function has reduce_retracing=True option that can avoid unnecessary retracing. For (3), please refer to https://www.tensorflow.org/guide/function#controlling_retracing and https://www.tensorflow.org/api_docs/python/tf/function for  more details.\n",
      "WARNING:tensorflow:5 out of the last 5 calls to <function TensorFlowTrainer.make_predict_function.<locals>.one_step_on_data_distributed at 0x7dee981abf40> triggered tf.function retracing. Tracing is expensive and the excessive number of tracings could be due to (1) creating @tf.function repeatedly in a loop, (2) passing tensors with different shapes, (3) passing Python objects instead of tensors. For (1), please define your @tf.function outside of the loop. For (2), @tf.function has reduce_retracing=True option that can avoid unnecessary retracing. For (3), please refer to https://www.tensorflow.org/guide/function#controlling_retracing and https://www.tensorflow.org/api_docs/python/tf/function for  more details.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1m1/1\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m1s\u001b[0m 548ms/step\n",
      "\u001b[1m1/1\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m0s\u001b[0m 49ms/step\n",
      "\u001b[1m1/1\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m0s\u001b[0m 51ms/step\n",
      "\u001b[1m1/1\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m0s\u001b[0m 51ms/step\n",
      "\u001b[1m1/1\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m0s\u001b[0m 59ms/step\n",
      "\u001b[1m1/1\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m0s\u001b[0m 53ms/step\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:root:dede['G|EYS|0.00|0.00|0.00|0.00|3|9|-20|9']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from subprocess import Popen\n",
    "\n",
    "\n",
    "process = Popen(\"spliceai -I ./lovd.vcf -O ./lovd_output.vcf -R ../tools/spliceai/hg38.fa -A grch38\".split())\n",
    "process.wait()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6a424607361ab528",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
