{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import importlib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>transcriptid</th>\n",
       "      <th>effectid</th>\n",
       "      <th>position_c_start</th>\n",
       "      <th>position_c_start_intron</th>\n",
       "      <th>position_c_end</th>\n",
       "      <th>position_c_end_intron</th>\n",
       "      <th>VariantOnTranscript/DNA</th>\n",
       "      <th>VariantOnTranscript/RNA</th>\n",
       "      <th>VariantOnTranscript/Protein</th>\n",
       "      <th>VariantOnTranscript/Exon</th>\n",
       "      <th>VariantOnGenome/DNA</th>\n",
       "      <th>VariantOnGenome/DNA/hg38</th>\n",
       "      <th>hg38_gnomad_format</th>\n",
       "      <th>Popmax_gnomad</th>\n",
       "      <th>Popmax population_gnomad</th>\n",
       "      <th>Homozygote Count_gnomad</th>\n",
       "      <th>Allele Frequency_gnomad</th>\n",
       "      <th>variant_id_gnomad</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>822823</td>\n",
       "      <td>7329</td>\n",
       "      <td>70</td>\n",
       "      <td>632</td>\n",
       "      <td>0</td>\n",
       "      <td>632</td>\n",
       "      <td>0</td>\n",
       "      <td>c.632G&gt;A</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Cys211Tyr)</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>g.216052273del</td>\n",
       "      <td>g.215878931del</td>\n",
       "      <td>6-215878931-del</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>822787</td>\n",
       "      <td>7329</td>\n",
       "      <td>70</td>\n",
       "      <td>8391</td>\n",
       "      <td>0</td>\n",
       "      <td>8391</td>\n",
       "      <td>0</td>\n",
       "      <td>c.8391del</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Gly2799Valfs*31)</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>g.216052410C&gt;T</td>\n",
       "      <td>g.215879068C&gt;T</td>\n",
       "      <td>6-215879068-C-T</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>822843</td>\n",
       "      <td>7329</td>\n",
       "      <td>70</td>\n",
       "      <td>5608</td>\n",
       "      <td>0</td>\n",
       "      <td>5608</td>\n",
       "      <td>0</td>\n",
       "      <td>c.5608C&gt;T</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Arg1870Trp)</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>g.216246607G&gt;A</td>\n",
       "      <td>g.216073265G&gt;A</td>\n",
       "      <td>6-216073265-G-A</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>822771</td>\n",
       "      <td>7329</td>\n",
       "      <td>70</td>\n",
       "      <td>8206</td>\n",
       "      <td>0</td>\n",
       "      <td>8206</td>\n",
       "      <td>0</td>\n",
       "      <td>c.8206G&gt;C</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Ala2736Pro)</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>g.216500979C&gt;T</td>\n",
       "      <td>g.216327637C&gt;T</td>\n",
       "      <td>6-216327637-C-T</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>958489</td>\n",
       "      <td>7329</td>\n",
       "      <td>70</td>\n",
       "      <td>9418</td>\n",
       "      <td>0</td>\n",
       "      <td>9418</td>\n",
       "      <td>0</td>\n",
       "      <td>c.9418C&gt;T</td>\n",
       "      <td>r.(?)</td>\n",
       "      <td>p.(Gln3140Ter)</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>g.64430509G&gt;A</td>\n",
       "      <td>g.63720613G&gt;A</td>\n",
       "      <td>6-63720613-G-A</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15930</th>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>African/African American</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>6-65495479-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15931</th>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6-65495479-G-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15932</th>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0.00007</td>\n",
       "      <td>Admixed American</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>6-65495482-A-G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15933</th>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0.00006</td>\n",
       "      <td>South Asian</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>6-65495484-T-G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15934</th>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>South Asian</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>6-65495485-T-C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16415 rows Ã— 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           id  transcriptid  effectid  position_c_start  \\\n",
       "0      822823          7329        70               632   \n",
       "1      822787          7329        70              8391   \n",
       "2      822843          7329        70              5608   \n",
       "3      822771          7329        70              8206   \n",
       "91     958489          7329        70              9418   \n",
       "...       ...           ...       ...               ...   \n",
       "15930    <NA>          <NA>      <NA>              <NA>   \n",
       "15931    <NA>          <NA>      <NA>              <NA>   \n",
       "15932    <NA>          <NA>      <NA>              <NA>   \n",
       "15933    <NA>          <NA>      <NA>              <NA>   \n",
       "15934    <NA>          <NA>      <NA>              <NA>   \n",
       "\n",
       "       position_c_start_intron  position_c_end  position_c_end_intron  \\\n",
       "0                            0             632                      0   \n",
       "1                            0            8391                      0   \n",
       "2                            0            5608                      0   \n",
       "3                            0            8206                      0   \n",
       "91                           0            9418                      0   \n",
       "...                        ...             ...                    ...   \n",
       "15930                     <NA>            <NA>                   <NA>   \n",
       "15931                     <NA>            <NA>                   <NA>   \n",
       "15932                     <NA>            <NA>                   <NA>   \n",
       "15933                     <NA>            <NA>                   <NA>   \n",
       "15934                     <NA>            <NA>                   <NA>   \n",
       "\n",
       "      VariantOnTranscript/DNA VariantOnTranscript/RNA  \\\n",
       "0                    c.632G>A                   r.(?)   \n",
       "1                   c.8391del                   r.(?)   \n",
       "2                   c.5608C>T                   r.(?)   \n",
       "3                   c.8206G>C                   r.(?)   \n",
       "91                  c.9418C>T                   r.(?)   \n",
       "...                       ...                     ...   \n",
       "15930                    <NA>                    <NA>   \n",
       "15931                    <NA>                    <NA>   \n",
       "15932                    <NA>                    <NA>   \n",
       "15933                    <NA>                    <NA>   \n",
       "15934                    <NA>                    <NA>   \n",
       "\n",
       "      VariantOnTranscript/Protein VariantOnTranscript/Exon  \\\n",
       "0                   p.(Cys211Tyr)                     <NA>   \n",
       "1             p.(Gly2799Valfs*31)                     <NA>   \n",
       "2                  p.(Arg1870Trp)                     <NA>   \n",
       "3                  p.(Ala2736Pro)                     <NA>   \n",
       "91                 p.(Gln3140Ter)                     <NA>   \n",
       "...                           ...                      ...   \n",
       "15930                        <NA>                     <NA>   \n",
       "15931                        <NA>                     <NA>   \n",
       "15932                        <NA>                     <NA>   \n",
       "15933                        <NA>                     <NA>   \n",
       "15934                        <NA>                     <NA>   \n",
       "\n",
       "      VariantOnGenome/DNA VariantOnGenome/DNA/hg38 hg38_gnomad_format  \\\n",
       "0          g.216052273del           g.215878931del    6-215878931-del   \n",
       "1          g.216052410C>T           g.215879068C>T    6-215879068-C-T   \n",
       "2          g.216246607G>A           g.216073265G>A    6-216073265-G-A   \n",
       "3          g.216500979C>T           g.216327637C>T    6-216327637-C-T   \n",
       "91          g.64430509G>A            g.63720613G>A     6-63720613-G-A   \n",
       "...                   ...                      ...                ...   \n",
       "15930                <NA>                     <NA>               <NA>   \n",
       "15931                <NA>                     <NA>               <NA>   \n",
       "15932                <NA>                     <NA>               <NA>   \n",
       "15933                <NA>                     <NA>               <NA>   \n",
       "15934                <NA>                     <NA>               <NA>   \n",
       "\n",
       "       Popmax_gnomad  Popmax population_gnomad  Homozygote Count_gnomad  \\\n",
       "0               <NA>                      <NA>                     <NA>   \n",
       "1               <NA>                      <NA>                     <NA>   \n",
       "2               <NA>                      <NA>                     <NA>   \n",
       "3               <NA>                      <NA>                     <NA>   \n",
       "91              <NA>                      <NA>                     <NA>   \n",
       "...              ...                       ...                      ...   \n",
       "15930       0.000031  African/African American                        0   \n",
       "15931            0.0                      <NA>                        0   \n",
       "15932        0.00007          Admixed American                        0   \n",
       "15933        0.00006               South Asian                        0   \n",
       "15934       0.000012               South Asian                        0   \n",
       "\n",
       "       Allele Frequency_gnomad variant_id_gnomad  \n",
       "0                         <NA>              <NA>  \n",
       "1                         <NA>              <NA>  \n",
       "2                         <NA>              <NA>  \n",
       "3                         <NA>              <NA>  \n",
       "91                        <NA>              <NA>  \n",
       "...                        ...               ...  \n",
       "15930                 0.000001    6-65495479-G-A  \n",
       "15931                      0.0    6-65495479-G-T  \n",
       "15932                 0.000003    6-65495482-A-G  \n",
       "15933                 0.000004    6-65495484-T-G  \n",
       "15934                 0.000001    6-65495485-T-C  \n",
       "\n",
       "[16415 rows x 19 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data = pd.read_csv('lovd_gnomad.csv')\n",
    "data = data.convert_dtypes()\n",
    "data.sort_values(by=\"hg38_gnomad_format\", inplace=True)\n",
    "\n",
    "display(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/__init__.py:42: FutureWarning: \n",
      "Dask dataframe query planning is disabled because dask-expr is not installed.\n",
      "\n",
      "You can install it with `pip install dask[dataframe]` or `conda install dask`.\n",
      "This will raise in a future version.\n",
      "\n",
      "  warnings.warn(msg, FutureWarning)\n"
     ]
    }
   ],
   "source": [
    "from revel import get_single_revel_score\n",
    "\n",
    "def assign_revel_scores(data:pd.DataFrame):\n",
    "    \"\"\"\n",
    "    Assigns REVEL scores to the data based on the genomic positions.\n",
    "\n",
    "    Parameters:\n",
    "    data : pd.DataFrame\n",
    "        DataFrame containing the data to which REVEL scores need to be assigned.\n",
    "    revel : pd.DataFrame\n",
    "        DataFrame containing REVEL scores.\n",
    "\n",
    "    Returns:\n",
    "    pd.DataFrame\n",
    "        DataFrame with REVEL scores assigned to the data.\n",
    "    \"\"\"\n",
    "    \n",
    "    for i in range(data.shape[0]):\n",
    "        hg38_gnomad_format = data.loc[i, 'hg38_gnomad_format']\n",
    "        print(f\"getting REVEL score for {hg38_gnomad_format}\")\n",
    "        if pd.isna(hg38_gnomad_format):\n",
    "            data.loc[i, 'REVEL'] = pd.NA\n",
    "        else:\n",
    "            try:\n",
    "                chromosome, position, ref, alt = hg38_gnomad_format.split('-')\n",
    "                revel_score = get_single_revel_score(chromosome, position, ref, alt)\n",
    "                data.loc[i, 'REVEL'] = revel_score\n",
    "            except:\n",
    "                data.loc[i, 'REVEL'] = pd.NA\n",
    "                print(f\"Error in getting REVEL score for {hg38_gnomad_format}\")\n",
    "    return data\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/io/csv.py:195: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = reader(bio, **kwargs)\n",
      "/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/io/csv.py:195: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = reader(bio, **kwargs)\n",
      "/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/io/csv.py:195: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = reader(bio, **kwargs)\n",
      "/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/io/csv.py:195: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = reader(bio, **kwargs)\n",
      "/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/io/csv.py:195: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = reader(bio, **kwargs)\n",
      "/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/io/csv.py:195: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = reader(bio, **kwargs)\n",
      "/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/io/csv.py:195: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = reader(bio, **kwargs)\n",
      "/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/io/csv.py:195: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = reader(bio, **kwargs)\n",
      "/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/io/csv.py:195: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = reader(bio, **kwargs)\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "Mismatched dtypes found in `pd.read_csv`/`pd.read_table`.\n\n+------------+--------+----------+\n| Column     | Found  | Expected |\n+------------+--------+----------+\n| grch38_pos | object | int64    |\n+------------+--------+----------+\n\nThe following columns also raised exceptions on conversion:\n\n- grch38_pos\n  ValueError(\"invalid literal for int() with base 10: '.'\")\n\nUsually this is due to dask's dtype inference failing, and\n*may* be fixed by specifying dtypes manually by adding:\n\ndtype={'grch38_pos': 'object'}\n\nto the call to `read_csv`/`read_table`.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[4], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m name \u001b[38;5;241m=\u001b[39m \u001b[43mget_single_revel_score\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m35142\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mG\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mA\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28mprint\u001b[39m(name)\n",
      "File \u001b[0;32m/mnt/c/Users/Kajus/PycharmProjects/kath/app/back_end/tools/revel/revel.py:81\u001b[0m, in \u001b[0;36mget_single_revel_score\u001b[0;34m(chromosome, grch38_position, ref, alt, chunk_size, revel_file)\u001b[0m\n\u001b[1;32m     73\u001b[0m filtered_ddf \u001b[38;5;241m=\u001b[39m ddf[\n\u001b[1;32m     74\u001b[0m     (ddf[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mchr\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mstr\u001b[38;5;241m.\u001b[39mstrip() \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mstr\u001b[39m(chromosome)) \u001b[38;5;241m&\u001b[39m\n\u001b[1;32m     75\u001b[0m     (ddf[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgrch38_pos\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m==\u001b[39m grch38_position) \u001b[38;5;241m&\u001b[39m\n\u001b[1;32m     76\u001b[0m     (ddf[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mref\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mstr\u001b[38;5;241m.\u001b[39mstrip() \u001b[38;5;241m==\u001b[39m ref) \u001b[38;5;241m&\u001b[39m\n\u001b[1;32m     77\u001b[0m     (ddf[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124malt\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mstr\u001b[38;5;241m.\u001b[39mstrip() \u001b[38;5;241m==\u001b[39m alt)\n\u001b[1;32m     78\u001b[0m ]\n\u001b[1;32m     80\u001b[0m \u001b[38;5;66;03m# Compute and bring the filtered result into memory\u001b[39;00m\n\u001b[0;32m---> 81\u001b[0m result \u001b[38;5;241m=\u001b[39m \u001b[43mfiltered_ddf\u001b[49m\u001b[43m[\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mREVEL\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m]\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcompute\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m  \u001b[38;5;66;03m# Brings the result into a Pandas DataFrame\u001b[39;00m\n\u001b[1;32m     83\u001b[0m \u001b[38;5;66;03m# Return the first matching REVEL score, if any\u001b[39;00m\n\u001b[1;32m     84\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m result[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mREVEL\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39miloc[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m result\u001b[38;5;241m.\u001b[39mempty \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "File \u001b[0;32m/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/base.py:372\u001b[0m, in \u001b[0;36mDaskMethodsMixin.compute\u001b[0;34m(self, **kwargs)\u001b[0m\n\u001b[1;32m    348\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mcompute\u001b[39m(\u001b[38;5;28mself\u001b[39m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m    349\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"Compute this dask collection\u001b[39;00m\n\u001b[1;32m    350\u001b[0m \n\u001b[1;32m    351\u001b[0m \u001b[38;5;124;03m    This turns a lazy Dask collection into its in-memory equivalent.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    370\u001b[0m \u001b[38;5;124;03m    dask.compute\u001b[39;00m\n\u001b[1;32m    371\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m--> 372\u001b[0m     (result,) \u001b[38;5;241m=\u001b[39m \u001b[43mcompute\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtraverse\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    373\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m result\n",
      "File \u001b[0;32m/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/base.py:660\u001b[0m, in \u001b[0;36mcompute\u001b[0;34m(traverse, optimize_graph, scheduler, get, *args, **kwargs)\u001b[0m\n\u001b[1;32m    657\u001b[0m     postcomputes\u001b[38;5;241m.\u001b[39mappend(x\u001b[38;5;241m.\u001b[39m__dask_postcompute__())\n\u001b[1;32m    659\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m shorten_traceback():\n\u001b[0;32m--> 660\u001b[0m     results \u001b[38;5;241m=\u001b[39m \u001b[43mschedule\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdsk\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    662\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m repack([f(r, \u001b[38;5;241m*\u001b[39ma) \u001b[38;5;28;01mfor\u001b[39;00m r, (f, a) \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mzip\u001b[39m(results, postcomputes)])\n",
      "File \u001b[0;32m/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/io/csv.py:142\u001b[0m, in \u001b[0;36mCSVFunctionWrapper.__call__\u001b[0;34m(self, part)\u001b[0m\n\u001b[1;32m    139\u001b[0m         rest_kwargs[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124musecols\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m columns\n\u001b[1;32m    141\u001b[0m \u001b[38;5;66;03m# Call `pandas_read_text`\u001b[39;00m\n\u001b[0;32m--> 142\u001b[0m df \u001b[38;5;241m=\u001b[39m \u001b[43mpandas_read_text\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    143\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mreader\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    144\u001b[0m \u001b[43m    \u001b[49m\u001b[43mblock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    145\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheader\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    146\u001b[0m \u001b[43m    \u001b[49m\u001b[43mrest_kwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    147\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdtypes\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    148\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    149\u001b[0m \u001b[43m    \u001b[49m\u001b[43mwrite_header\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    150\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43menforce\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    151\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpath_info\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    152\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    153\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m project_after_read:\n\u001b[1;32m    154\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m df[\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns]\n",
      "File \u001b[0;32m/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/io/csv.py:197\u001b[0m, in \u001b[0;36mpandas_read_text\u001b[0;34m(reader, b, header, kwargs, dtypes, columns, write_header, enforce, path)\u001b[0m\n\u001b[1;32m    195\u001b[0m df \u001b[38;5;241m=\u001b[39m reader(bio, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[1;32m    196\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m dtypes:\n\u001b[0;32m--> 197\u001b[0m     \u001b[43mcoerce_dtypes\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtypes\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    199\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m enforce \u001b[38;5;129;01mand\u001b[39;00m columns \u001b[38;5;129;01mand\u001b[39;00m (\u001b[38;5;28mlist\u001b[39m(df\u001b[38;5;241m.\u001b[39mcolumns) \u001b[38;5;241m!=\u001b[39m \u001b[38;5;28mlist\u001b[39m(columns)):\n\u001b[1;32m    200\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mColumns do not match\u001b[39m\u001b[38;5;124m\"\u001b[39m, df\u001b[38;5;241m.\u001b[39mcolumns, columns)\n",
      "File \u001b[0;32m/mnt/c/Users/Kajus/PycharmProjects/kath/.venv/lib/python3.12/site-packages/dask/dataframe/io/csv.py:298\u001b[0m, in \u001b[0;36mcoerce_dtypes\u001b[0;34m(df, dtypes)\u001b[0m\n\u001b[1;32m    294\u001b[0m rule \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m (\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m-\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m*\u001b[39m \u001b[38;5;241m61\u001b[39m)\n\u001b[1;32m    295\u001b[0m msg \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mMismatched dtypes found in `pd.read_csv`/`pd.read_table`.\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m (\n\u001b[1;32m    296\u001b[0m     rule\u001b[38;5;241m.\u001b[39mjoin(\u001b[38;5;28mfilter\u001b[39m(\u001b[38;5;28;01mNone\u001b[39;00m, [dtype_msg, date_msg]))\n\u001b[1;32m    297\u001b[0m )\n\u001b[0;32m--> 298\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(msg)\n",
      "\u001b[0;31mValueError\u001b[0m: Mismatched dtypes found in `pd.read_csv`/`pd.read_table`.\n\n+------------+--------+----------+\n| Column     | Found  | Expected |\n+------------+--------+----------+\n| grch38_pos | object | int64    |\n+------------+--------+----------+\n\nThe following columns also raised exceptions on conversion:\n\n- grch38_pos\n  ValueError(\"invalid literal for int() with base 10: '.'\")\n\nUsually this is due to dask's dtype inference failing, and\n*may* be fixed by specifying dtypes manually by adding:\n\ndtype={'grch38_pos': 'object'}\n\nto the call to `read_csv`/`read_table`."
     ]
    }
   ],
   "source": [
    "name = get_single_revel_score(1, 35142, 'G', 'A')\n",
    "\n",
    "print(name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "getting REVEL score for 6-215878931-del\n",
      "Error in getting REVEL score for 6-215878931-del\n",
      "getting REVEL score for 6-215879068-C-T\n",
      "Error in getting REVEL score for 6-215879068-C-T\n",
      "getting REVEL score for 6-216073265-G-A\n",
      "Error in getting REVEL score for 6-216073265-G-A\n",
      "getting REVEL score for 6-216327637-C-T\n",
      "Error in getting REVEL score for 6-216327637-C-T\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for 6-63720613-G-A\n",
      "Error in getting REVEL score for 6-63720613-G-A\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for 6-63720617-A-C\n",
      "Error in getting REVEL score for 6-63720617-A-C\n",
      "getting REVEL score for 6-63720617-A-C\n",
      "Error in getting REVEL score for 6-63720617-A-C\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for 6-63720621-C-T\n",
      "Error in getting REVEL score for 6-63720621-C-T\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for 6-63720622-C-T\n",
      "Error in getting REVEL score for 6-63720622-C-T\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for <NA>\n",
      "getting REVEL score for 6-63720626-A-T\n",
      "Error in getting REVEL score for 6-63720626-A-T\n",
      "getting REVEL score for 6-63720626-A-T\n",
      "Error in getting REVEL score for 6-63720626-A-T\n",
      "getting REVEL score for 6-63720626-A-T\n",
      "Error in getting REVEL score for 6-63720626-A-T\n",
      "getting REVEL score for 6-63720626-A-T\n",
      "Error in getting REVEL score for 6-63720626-A-T\n",
      "getting REVEL score for 6-63720626-A-T\n",
      "Error in getting REVEL score for 6-63720626-A-T\n",
      "getting REVEL score for 6-63720626-A-T\n",
      "Error in getting REVEL score for 6-63720626-A-T\n",
      "getting REVEL score for 6-63720626-A-T\n",
      "Error in getting REVEL score for 6-63720626-A-T\n",
      "getting REVEL score for 6-63720626-A-T\n",
      "Error in getting REVEL score for 6-63720626-A-T\n",
      "getting REVEL score for 6-63720626-A-T\n",
      "Error in getting REVEL score for 6-63720626-A-T\n",
      "getting REVEL score for 6-63720626-A-T\n",
      "Error in getting REVEL score for 6-63720626-A-T\n",
      "getting REVEL score for 6-63720626-A-T\n"
     ]
    },
    {
     "ename": "",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31mThe Kernel crashed while executing code in the current cell or a previous cell. \n",
      "\u001b[1;31mPlease review the code in the cell(s) to identify a possible cause of the failure. \n",
      "\u001b[1;31mClick <a href='https://aka.ms/vscodeJupyterKernelCrash'>here</a> for more info. \n",
      "\u001b[1;31mView Jupyter <a href='command:jupyter.viewOutput'>log</a> for further details."
     ]
    }
   ],
   "source": [
    "\n",
    "data = assign_revel_scores(data)\n",
    "\n",
    "display(data)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
